// @atlaskit/smart-card/dist/cjs/view/BlockCard/views/styledOld.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.titleBlockCss = exports.metadataBlockCss = exports.footerBlockCss = exports.flexibleBlockCardStyle = void 0;
var _react = require("@emotion/react");
var _colors = require("@atlaskit/theme/colors");
// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766

// in editor prosemirror adds padding-left so we need to overwrite it
// renderer overwrites the margin-right so we need to add it here
// eslint-disable-next-line @atlaskit/ui-styling-standard/no-exported-styles -- Ignored via go/DSP-18766
var metadataBlockCss = exports.metadataBlockCss = (0, _react.css)({
  // eslint-disable-next-line @atlaskit/ui-styling-standard/no-nested-selectors -- Ignored via go/DSP-18766
  'span[data-smart-element-avatar-group]': {
    // eslint-disable-next-line @atlaskit/ui-styling-standard/no-nested-selectors -- Ignored via go/DSP-18766
    '> ul': {
      paddingLeft: '0px',
      marginRight: "var(--ds-space-100, 0.5rem)"
    }
  },
  // eslint-disable-next-line @atlaskit/ui-styling-standard/no-nested-selectors -- Ignored via go/DSP-18766
  '[data-smart-element-group]': {
    // eslint-disable-next-line @atlaskit/design-system/use-tokens-typography
    lineHeight: '1rem'
  }
});
// eslint-disable-next-line @atlaskit/ui-styling-standard/no-exported-styles -- Ignored via go/DSP-18766
var titleBlockCss = exports.titleBlockCss = (0, _react.css)({
  gap: "var(--ds-space-100, 0.5em)",
  // eslint-disable-next-line @atlaskit/ui-styling-standard/no-nested-selectors -- Ignored via go/DSP-18766
  "[data-smart-element='Title']": {
    fontWeight: "var(--ds-font-weight-semibold, 600)"
  }
});

// eslint-disable-next-line @atlaskit/ui-styling-standard/no-exported-styles -- Ignored via go/DSP-18766
var footerBlockCss = exports.footerBlockCss = (0, _react.css)({
  height: '1.5rem',
  // eslint-disable-next-line @atlaskit/ui-styling-standard/no-nested-selectors -- Ignored via go/DSP-18766
  '.actions-button-group': {
    // eslint-disable-next-line @atlaskit/ui-styling-standard/no-nested-selectors -- Ignored via go/DSP-18766
    'button, button:hover, button:focus, button:active': {
      // eslint-disable-next-line @atlaskit/design-system/use-tokens-typography
      fontSize: '0.875rem'
    }
  }
});

// eslint-disable-next-line @atlaskit/ui-styling-standard/no-exported-styles -- Ignored via go/DSP-18766
var flexibleBlockCardStyle = exports.flexibleBlockCardStyle = (0, _react.css)({
  // eslint-disable-next-line @atlaskit/ui-styling-standard/no-nested-selectors -- Ignored via go/DSP-18766
  '& > div': {
    borderRadius: "var(--ds-border-radius-200, 8px)",
    border: "1px solid ".concat("var(--ds-border, ".concat(_colors.N40, ")"))
  }
});
