// @atlaskit/smart-card/dist/cjs/view/FlexibleCard/components/actions/action/action-stack-item/action-buttonOld.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
var _typeof = require("@babel/runtime/helpers/typeof");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _react = _interopRequireWildcard(require("react"));
var _primitives = require("@atlaskit/primitives");
var _spinner = _interopRequireDefault(require("@atlaskit/spinner"));
var _utils = require("../../../utils");
var _actionIcon = _interopRequireDefault(require("../action-icon"));
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
var buttonStyles = (0, _primitives.xcss)({
  all: 'unset',
  padding: 'space.050',
  width: '100%',
  ':hover': {
    backgroundColor: 'color.background.neutral.subtle.hovered'
  },
  ':active': {
    backgroundColor: 'color.background.neutral.subtle.pressed'
  }
});
var contentStyles = (0, _primitives.xcss)({
  color: 'color.text',
  font: "var(--ds-font-body-UNSAFE_small, normal 400 12px/16px ui-sans-serif, -apple-system, BlinkMacSystemFont, \"Segoe UI\", Ubuntu, \"Helvetica Neue\", sans-serif)"
});
var ActionButtonOld = function ActionButtonOld(_ref) {
  var content = _ref.content,
    iconOption = _ref.icon,
    isDisabled = _ref.isDisabled,
    isLoading = _ref.isLoading,
    onClickCallback = _ref.onClick,
    size = _ref.size,
    spaceOption = _ref.space,
    testId = _ref.testId,
    tooltipProps = _ref.tooltipProps,
    xcss = _ref.xcss;
  var space = spaceOption !== null && spaceOption !== void 0 ? spaceOption : (0, _utils.getPrimitivesInlineSpaceBySize)(size);
  var onClick = (0, _react.useCallback)(function () {
    if (!isDisabled && !isLoading && onClickCallback) {
      onClickCallback();
    }
  }, [isDisabled, isLoading, onClickCallback]);
  var icon = iconOption && isLoading ? /*#__PURE__*/_react.default.createElement(_actionIcon.default, {
    asStackItemIcon: true,
    icon: /*#__PURE__*/_react.default.createElement(_spinner.default, {
      testId: "".concat(testId, "-loading")
    }),
    size: size
  }) : iconOption;
  return (
    /*#__PURE__*/
    // eslint-disable-next-line @atlaskit/design-system/consistent-css-prop-usage
    _react.default.createElement(_primitives.Pressable, (0, _extends2.default)({
      xcss: [buttonStyles, xcss]
    }, tooltipProps, {
      onClick: onClick,
      testId: testId
    }), /*#__PURE__*/_react.default.createElement(_primitives.Inline, {
      alignBlock: "center",
      grow: "fill",
      space: space
    }, icon, /*#__PURE__*/_react.default.createElement(_primitives.Box, {
      xcss: contentStyles
    }, content)))
  );
};
var _default = exports.default = ActionButtonOld;
