// @atlaskit/link-datasource/dist/cjs/ui/common/modal/content-container/content-container-old.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ContentContainerOld = void 0;
var _react = _interopRequireDefault(require("react"));
var _primitives = require("@atlaskit/primitives");
var _colors = require("@atlaskit/theme/colors");
var _issueLikeTableOld = require("../../../issue-like-table/issue-like-table-old");
var contentContainerStyles = (0, _primitives.xcss)({
  display: 'grid',
  maxHeight: '420px',
  overflow: 'auto',
  borderBottom: "2px solid ".concat("var(--ds-background-accent-gray-subtler, ".concat(_colors.N40, ")")),
  // eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values, @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
  backgroundImage: _issueLikeTableOld.scrollableContainerShadowsCssComponents.backgroundImage,
  // eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values, @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
  backgroundPosition: _issueLikeTableOld.scrollableContainerShadowsCssComponents.backgroundPosition,
  // eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values, @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
  backgroundRepeat: _issueLikeTableOld.scrollableContainerShadowsCssComponents.backgroundRepeat,
  // eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values, @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
  backgroundSize: _issueLikeTableOld.scrollableContainerShadowsCssComponents.backgroundSize,
  // eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values, @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
  backgroundAttachment: _issueLikeTableOld.scrollableContainerShadowsCssComponents.backgroundAttachment
});
var tableContainerStyles = (0, _primitives.xcss)({
  borderTopLeftRadius: "var(--ds-border-radius-200, 8px)",
  borderTopRightRadius: "var(--ds-border-radius-200, 8px)",
  border: "1px solid ".concat("var(--ds-border, ".concat(_colors.N40, ")"))
});
var ContentContainerOld = exports.ContentContainerOld = function ContentContainerOld(_ref) {
  var children = _ref.children,
    withTableBorder = _ref.withTableBorder;
  return /*#__PURE__*/_react.default.createElement(_primitives.Box, {
    xcss: [contentContainerStyles, withTableBorder && tableContainerStyles]
  }, children);
};
