// @atlaskit/link-datasource/dist/cjs/ui/assets-modal/modal/render-assets-content/initial-state-view/initial-state-view-old.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.InitialStateViewOld = void 0;
var _react = require("@emotion/react");
var _reactIntlNext = require("react-intl-next");
var _typography = require("@atlaskit/theme/typography");
var _crystalBallSvgOld = require("./assets/crystal-ball-svg-old");
var _messages = require("./messages");
/**
 * @jsxRuntime classic
 * @jsx jsx
 */

// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766

var initialStateViewContainerStyles = (0, _react.css)({
  display: 'flex',
  justifyContent: 'center',
  height: '100%'
});
var svgAndTextsWrapperStyles = (0, _react.css)({
  textAlign: 'center',
  alignSelf: 'center'
});
var searchTitleStyles = (0, _react.css)({
  // eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values -- Ignored via go/DSP-18766
  font: "var(--ds-font-heading-small, ".concat(_typography.fontFallback.heading.small, ")"),
  paddingTop: "var(--ds-space-200, 16px)",
  paddingBottom: "var(--ds-space-100, 8px)"
});
var AQLSupportDocumentLink = 'https://support.atlassian.com/jira-service-management-cloud/docs/use-assets-query-language-aql/';
var InitialStateViewOld = exports.InitialStateViewOld = function InitialStateViewOld() {
  var _useIntl = (0, _reactIntlNext.useIntl)(),
    formatMessage = _useIntl.formatMessage;
  return (0, _react.jsx)("div", {
    css: initialStateViewContainerStyles,
    "data-testid": "assets-aql-datasource-modal--initial-state-view"
  }, (0, _react.jsx)("div", {
    css: svgAndTextsWrapperStyles
  }, (0, _react.jsx)(_crystalBallSvgOld.CrystalBallSVGOld, null), (0, _react.jsx)("div", {
    css: searchTitleStyles
  }, formatMessage(_messages.initialStateViewMessages.searchTitle)), (0, _react.jsx)("div", null, formatMessage(_messages.initialStateViewMessages.searchDescription)), (0, _react.jsx)("a", {
    href: AQLSupportDocumentLink,
    target: "_blank"
  }, formatMessage(_messages.initialStateViewMessages.learnMoreLink))));
};
