// @atlaskit/smart-card/dist/cjs/view/HoverCard/components/views/forbidden/HoverCardForbiddenViewOld.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _react = _interopRequireDefault(require("react"));
var _reactIntlNext = require("react-intl-next");
var _analyticsNext = require("@atlaskit/analytics-next");
var _button = _interopRequireDefault(require("@atlaskit/button"));
var _linkExtractors = require("@atlaskit/link-extractors");
var _extractAccessContext = require("../../../../../extractors/common/context/extractAccessContext");
var _extractHostname = _interopRequireDefault(require("../../../../../extractors/common/hostname/extractHostname"));
var _messages = require("../../../../../messages");
var _FlexibleCard = _interopRequireDefault(require("../../../../FlexibleCard"));
var _blocks = require("../../../../FlexibleCard/components/blocks");
var _styled = require("../../../styled");
var _styled2 = require("./styled");
var HoverCardForbiddenViewOld = function HoverCardForbiddenViewOld(_ref) {
  var _cardState$details, _cardState$details2, _extractProvider$text, _extractProvider, _extractRequestAccess;
  var flexibleCardProps = _ref.flexibleCardProps,
    _ref$testId = _ref.testId,
    testId = _ref$testId === void 0 ? 'hover-card-forbidden-view' : _ref$testId;
  var _useAnalyticsEvents = (0, _analyticsNext.useAnalyticsEvents)(),
    createAnalyticsEvent = _useAnalyticsEvents.createAnalyticsEvent;
  var cardState = flexibleCardProps.cardState,
    url = flexibleCardProps.url;
  var data = (_cardState$details = cardState.details) === null || _cardState$details === void 0 ? void 0 : _cardState$details.data;
  var meta = (_cardState$details2 = cardState.details) === null || _cardState$details2 === void 0 ? void 0 : _cardState$details2.meta;
  var product = (_extractProvider$text = (_extractProvider = (0, _linkExtractors.extractProvider)(data)) === null || _extractProvider === void 0 ? void 0 : _extractProvider.text) !== null && _extractProvider$text !== void 0 ? _extractProvider$text : '';
  var hostname = /*#__PURE__*/_react.default.createElement("b", null, (0, _extractHostname.default)(url));
  var _ref2 = (_extractRequestAccess = (0, _extractAccessContext.extractRequestAccessContextImproved)({
      jsonLd: meta,
      url: url,
      product: product,
      createAnalyticsEvent: createAnalyticsEvent
    })) !== null && _extractRequestAccess !== void 0 ? _extractRequestAccess : {},
    action = _ref2.action,
    descriptiveMessageKey = _ref2.descriptiveMessageKey,
    titleMessageKey = _ref2.titleMessageKey,
    buttonDisabled = _ref2.buttonDisabled;
  if (!titleMessageKey || !descriptiveMessageKey) {
    return null;
  }
  return /*#__PURE__*/_react.default.createElement(_FlexibleCard.default, (0, _extends2.default)({}, flexibleCardProps, {
    testId: testId
  }), /*#__PURE__*/_react.default.createElement(_blocks.PreviewBlock, {
    ignoreContainerPadding: true,
    overrideCss: (0, _styled.getPreviewBlockStyles)(),
    testId: testId
  }), /*#__PURE__*/_react.default.createElement(_blocks.CustomBlock, {
    overrideCss: _styled2.titleBlockStyles,
    testId: "".concat(testId, "-title")
  }, /*#__PURE__*/_react.default.createElement(_reactIntlNext.FormattedMessage, (0, _extends2.default)({}, _messages.messages[titleMessageKey], {
    values: {
      product: product
    }
  }))), /*#__PURE__*/_react.default.createElement(_blocks.CustomBlock, {
    overrideCss: _styled2.mainTextStyles,
    testId: "".concat(testId, "-content")
  }, /*#__PURE__*/_react.default.createElement(_reactIntlNext.FormattedMessage, (0, _extends2.default)({}, _messages.messages[descriptiveMessageKey], {
    values: {
      product: product,
      hostname: hostname
    }
  }))), action && /*#__PURE__*/_react.default.createElement(_blocks.CustomBlock, {
    overrideCss: _styled2.connectButtonStyles
  }, /*#__PURE__*/_react.default.createElement(_button.default, {
    testId: "".concat(testId, "-button"),
    onClick: action === null || action === void 0 ? void 0 : action.promise,
    appearance: "primary",
    isDisabled: buttonDisabled
  }, action === null || action === void 0 ? void 0 : action.text)));
};
var _default = exports.default = HoverCardForbiddenViewOld;
