// @atlaskit/smart-card/dist/cjs/view/HoverCard/components/views/forbidden/styled.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.titleBlockStyles = exports.mainTextStyles = exports.connectButtonStyles = void 0;
var _react = require("@emotion/react");
// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766

/**
 * @deprecated remove on FF clean up bandicoots-compiled-migration-smartcard
 */
// eslint-disable-next-line @atlaskit/ui-styling-standard/no-exported-styles -- Ignored via go/DSP-18766
var titleBlockStyles = exports.titleBlockStyles = (0, _react.css)({
  justifyContent: 'center',
  fontWeight: "var(--ds-font-weight-semibold, 600)",
  marginTop: "var(--ds-space-100, 8px)"
});

/**
 * @deprecated remove on FF clean up bandicoots-compiled-migration-smartcard
 */
// eslint-disable-next-line @atlaskit/ui-styling-standard/no-exported-styles -- Ignored via go/DSP-18766
var mainTextStyles = exports.mainTextStyles = (0, _react.css)({
  display: 'inline',
  justifyContent: 'center',
  marginTop: "var(--ds-space-0, 0px)",
  font: "var(--ds-font-body-UNSAFE_small, normal 400 12px/16px ui-sans-serif, -apple-system, BlinkMacSystemFont, \"Segoe UI\", Ubuntu, \"Helvetica Neue\", sans-serif)",
  textAlign: 'center'
});

/**
 * @deprecated remove on FF clean up bandicoots-compiled-migration-smartcard
 */
// eslint-disable-next-line @atlaskit/ui-styling-standard/no-exported-styles -- Ignored via go/DSP-18766
var connectButtonStyles = exports.connectButtonStyles = (0, _react.css)({
  justifyContent: 'center',
  marginTop: "var(--ds-space-100, 8px)"
});
