// @atlaskit/renderer/dist/es2019/react/nodes/codeBlock/components/codeBlockButtonContainer.js
/**
 * @jsxRuntime classic
 * @jsx jsx
 */
// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766
import { jsx, css } from '@emotion/react';
import CopyButton from './codeBlockCopyButton';
import CodeWrapButton from './codeBlockWrapButton';
import { N0, N20, N30, N700 } from '@atlaskit/theme/colors';
const codeBlockButtonsWrapper = css({
  position: 'sticky',
  top: '0px',
  background: `${`var(--ds-surface, ${N20})`}`
});
const codeBlockButtonsStyle = css({
  display: 'flex',
  justifyContent: 'flex-end',
  position: 'absolute',
  height: '0',
  width: '100%',
  right: "var(--ds-space-075, 6px)",
  top: "var(--ds-space-050, 4px)",
  padding: "var(--ds-space-025, 2px)",
  // eslint-disable-next-line @atlaskit/ui-styling-standard/no-nested-selectors -- Ignored via go/DSP-18766
  button: {
    height: '32px',
    width: '32px',
    border: `2px solid ${`var(--ds-border, ${N0})`}`,
    borderRadius: '4px',
    marginLeft: "var(--ds-space-050, 4px)",
    padding: "var(--ds-space-025, 2px)",
    background: `${`var(--ds-surface-overlay, ${N20})`}`,
    color: `${"var(--ds-icon, rgb(66, 82, 110))"}`,
    '&:hover': {
      borderWidth: '2px',
      backgroundColor: `${`var(--ds-surface-overlay-hovered, ${N30})`}`,
      height: '32px',
      width: '32px'
    },
    // eslint-disable-next-line @atlaskit/ui-styling-standard/no-nested-selectors -- Ignored via go/DSP-18766
    '&.clicked': {
      backgroundColor: `${`var(--ds-background-neutral-bold-pressed, ${N700})`}`,
      borderRadius: '4px',
      // eslint-disable-next-line @atlaskit/ui-styling-standard/no-important-styles -- Ignored via go/DSP-18766
      color: `${`var(--ds-icon-inverse, ${N0})`} !important`
    }
  }
});
const CodeBlockButtonContainer = ({
  allowCopyToClipboard,
  allowWrapCodeBlock,
  setWrapLongLines,
  text,
  wrapLongLines
}) => {
  return jsx("div", {
    css: codeBlockButtonsWrapper
  }, jsx("div", {
    css: codeBlockButtonsStyle
  }, allowWrapCodeBlock && jsx(CodeWrapButton, {
    setWrapLongLines: setWrapLongLines,
    wrapLongLines: wrapLongLines
  }), allowCopyToClipboard && jsx(CopyButton, {
    content: text
  })));
};
export default CodeBlockButtonContainer;
