// @atlaskit/link-picker/dist/cjs/ui/link-picker/search-results/link-search-list/list-item/old/styled.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.listItemContextStyles = exports.listItemContainerStyles = exports.listItemContainerInnerStyles = exports.itemNameStyles = exports.itemIconStyles = exports.imgStyles = exports.composeListItemStyles = void 0;
var _react = require("@emotion/react");
var _colors = require("@atlaskit/theme/colors");
// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766

var listItemBaseStyles = (0, _react.css)({
  display: 'flex',
  paddingTop: "var(--ds-space-100, 8px)",
  paddingBottom: "var(--ds-space-100, 8px)",
  paddingLeft: "clamp( ".concat("var(--ds-space-100, 8px)", ", var(--link-picker-padding-left), 100% )"),
  paddingRight: "clamp( ".concat("var(--ds-space-100, 8px)", ", var(--link-picker-padding-right), 100% )"),
  margin: 0,
  cursor: 'pointer'
});
var listItemFocusStyles = (0, _react.css)({
  '&:focus': {
    outline: 'none',
    boxShadow: "0 0 0 2px ".concat("var(--ds-border-focused, ".concat(_colors.B100, ")"), " inset"),
    textDecoration: 'none'
  }
});
var listItemBoxShadow = (0, _react.css)({
  boxShadow: "inset 2px 0px 0px ".concat("var(--ds-border-selected, ".concat(_colors.B400, ")"))
});
var listItemActive = (0, _react.css)({
  '&:hover': {
    backgroundColor: "var(--ds-background-neutral-subtle-hovered, ".concat(_colors.N20, ")")
  },
  // eslint-disable-next-line @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
  listItemBoxShadow: listItemBoxShadow
});
var listItemSelected = (0, _react.css)({
  backgroundColor: "var(--ds-background-selected, ".concat(_colors.B50, ")")
},
// eslint-disable-next-line @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
listItemBoxShadow);
var composeListItemStyles = exports.composeListItemStyles = function composeListItemStyles() {
  var selected = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : false;
  return (0, _react.css)(
  // eslint-disable-next-line @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
  listItemBaseStyles,
  // eslint-disable-next-line @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
  !selected && listItemActive,
  // eslint-disable-next-line @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
  selected && listItemSelected,
  // eslint-disable-next-line @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
  listItemFocusStyles);
};

// eslint-disable-next-line @atlaskit/ui-styling-standard/no-exported-styles -- Ignored via go/DSP-18766
var itemNameStyles = exports.itemNameStyles = (0, _react.css)({
  overflow: 'hidden',
  alignContent: 'center',
  width: '100%'
});

// eslint-disable-next-line @atlaskit/ui-styling-standard/no-exported-styles -- Ignored via go/DSP-18766
var listItemContextStyles = exports.listItemContextStyles = (0, _react.css)({
  color: "var(--ds-text, ".concat(_colors.N300, ")"),
  font: "var(--ds-font-body-small, normal 400 11px/16px ui-sans-serif, -apple-system, BlinkMacSystemFont, \"Segoe UI\", Ubuntu, system-ui, \"Helvetica Neue\", sans-serif)",
  display: 'flex'
});

// eslint-disable-next-line @atlaskit/ui-styling-standard/no-exported-styles -- Ignored via go/DSP-18766
var listItemContainerStyles = exports.listItemContainerStyles = (0, _react.css)({
  overflow: 'hidden',
  textOverflow: 'ellipsis'
});

// eslint-disable-next-line @atlaskit/ui-styling-standard/no-exported-styles -- Ignored via go/DSP-18766
var listItemContainerInnerStyles = exports.listItemContainerInnerStyles = (0, _react.css)({
  color: "var(--ds-text-subtlest, ".concat(_colors.N200, ")"),
  whiteSpace: 'nowrap'
});

// eslint-disable-next-line @atlaskit/design-system/ensure-design-token-usage, @atlaskit/ui-styling-standard/no-exported-styles -- Ignored via go/DSP-18766
var itemIconStyles = exports.itemIconStyles = (0, _react.css)({
  minWidth: "var(--ds-space-200, 16px)",
  marginTop: "var(--ds-space-050, 4px)",
  marginRight: "var(--ds-space-150, 12px)"
});

// eslint-disable-next-line @atlaskit/ui-styling-standard/no-exported-styles -- Ignored via go/DSP-18766
var imgStyles = exports.imgStyles = (0, _react.css)({
  maxWidth: "var(--ds-space-200, 16px)"
});
