// @atlaskit/tmp-editor-statsig/dist/cjs/experiments-config.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.editorExperimentsConfig = void 0;
var _experimentBuilders = require("./experiment-builders");
/* eslint-disable @atlaskit/editor/no-re-export */
// Entry file in package.json

/**
 * When adding a new experiment, you need to add it here.
 * Please follow the pattern established in the examples and any
 * existing experiments.
 */
var editorExperimentsConfig = exports.editorExperimentsConfig = {
  // Added 2025-05-27
  platform_editor_reduce_noisy_steps_ncs: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_reduce_noisy_steps_ncs'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2024-08-08
  'example-boolean': (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'confluence_editor_experiment_test_new_package_boolean'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2024-08-08
  'example-multivariate': (0, _experimentBuilders.createMultivariateExperiment)({
    productKeys: {
      confluence: 'confluence_editor_experiment_test_new_package_multivariate'
    },
    param: 'variant',
    values: ['one', 'two', 'three'],
    defaultValue: 'one'
  }),
  // Added 2024-08-08
  'test-new-experiments-package': (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'confluence_editor_experiment_test_new_package',
      jira: 'jira_editor_experiment_test_new_package'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2024-09-05
  support_table_in_comment: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_support_table_in_comment_exp'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2024-09-07
  platform_editor_exp_lazy_node_views: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_exp_lazy_node_views'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2024-09-18
  platform_renderer_table_sticky_scrollbar: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_renderer_table_sticky_scrollbar'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-05-28
  platform_editor_controls_performance_fixes: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_controls_performance_fixes'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-05-26
  platform_editor_prevent_toolbar_layout_shifts: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_prevent_toolbar_layout_shifts'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2024-10-01
  comment_on_bodied_extensions: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_comment_on_bodied_extensions'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2024-10-14
  // https://console.statsig.com/LqivKg6ADZZaGczRfBKfX/experiments/platform_editor_advanced_layouts/setup
  advanced_layouts: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_advanced_layouts'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-3-15
  // https://console.statsig.com/LqivKg6ADZZaGczRfBKfX/experiments/platform_editor_single_column_layout/setup
  single_column_layouts: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_single_column_layout'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2024-10-08
  support_table_in_comment_jira: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      jira: 'platform_editor_support_table_in_comment_jira'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  'platform_editor_ai-prompts-placeholder': (0, _experimentBuilders.createMultivariateExperiment)({
    productKeys: {
      confluence: 'platform_editor_ai-prompts-placeholder'
    },
    param: 'cohort',
    values: ['control', 'test'],
    defaultValue: 'control'
  }),
  // added 2024-11-06
  // https://console.statsig.com/LqivKg6ADZZaGczRfBKfX/experiments/platform_editor_nested_tables/setup
  'nested-tables-in-tables': (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_nested_tables'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2024-11-26
  platform_editor_ai_unsplash_page_header: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_ai_unsplash_page_header'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2024-12-05
  platform_editor_blockquote_in_text_formatting_menu: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_blockquote_in_text_formatting_menu'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-01-09
  platform_editor_advanced_code_blocks: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_advanced_code_blocks'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-01-13
  platform_editor_element_drag_and_drop_multiselect: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_element_drag_and_drop_multiselect'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-01-19
  platform_editor_ai_edit_response_in_preview: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_ai_edit_response_in_preview'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-02-10
  platform_editor_controls: (0, _experimentBuilders.createMultivariateExperiment)({
    productKeys: {
      confluence: 'platform_editor_controls'
    },
    param: 'cohort',
    values: ['control', 'variant1'],
    defaultValue: 'control'
  }),
  // Added 28-02-2025
  platform_editor_insertion: (0, _experimentBuilders.createMultivariateExperiment)({
    productKeys: {
      confluence: 'platform_editor_insertion'
    },
    param: 'cohort',
    values: ['control', 'variant1'],
    defaultValue: 'control'
  }),
  // Added 2025-04-01
  // https://console.statsig.com/LqivKg6ADZZaGczRfBKfX/experiments/platform_editor_vanilla_dom/setup
  platform_editor_vanilla_dom: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_vanilla_dom'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-03-13
  editor_text_highlight_orange_to_yellow: (0, _experimentBuilders.createMultivariateExperiment)({
    productKeys: {
      confluence: 'editor_text_highlight_orange_to_yellow'
    },
    param: 'cohort',
    values: ['control', 'test'],
    defaultValue: 'control'
  }),
  // Added 2025-03-28
  platform_editor_ai_proactive_ai_nudge_parameters: (0, _experimentBuilders.createMultivariateExperiment)({
    productKeys: {
      confluence: 'platform_editor_ai_proactive_ai_nudge_parameters'
    },
    param: 'cohort',
    values: ['control', 'variant1'],
    defaultValue: 'control'
  }),
  // Added 2025-04-14
  // https://switcheroo.atlassian.com/ui/gates/b159b45a-86d9-4f4b-b482-f9aca5b615d6/key/platform_editor_offline_editing_web
  platform_editor_offline_editing_web: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_offline_editing_web'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-04-14
  editor_ai_inline_suggestion_date_v2: (0, _experimentBuilders.createMultivariateExperiment)({
    productKeys: {
      confluence: 'editor_ai_inline_suggestion_date_v2'
    },
    param: 'cohort',
    values: ['control', 'test'],
    defaultValue: 'control'
  }),
  // Added 2025-04-17
  platform_editor_tables_drag_and_drop: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_tables_drag_and_drop'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-04-17
  platform_editor_tables_table_selector: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_tables_table_selector'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-04-23
  platform_editor_usesharedpluginstateselector: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_usesharedpluginstateselector'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-04-23
  platform_renderer_fix_analytics_memo_callback: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_renderer_fix_analytics_memo_callback'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-05-05
  editor_ai_contextual_selection_toolbar_button: (0, _experimentBuilders.createMultivariateExperiment)({
    productKeys: {
      confluence: 'editor_ai_contextual_selection_toolbar_button'
    },
    param: 'cohort',
    values: ['control', 'test'],
    defaultValue: 'control'
  }),
  // Added 2025-05-08
  editor_ai_converge_free_gen_on_rovo: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'editor_ai_converge_free_gen_on_rovo'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-05-07
  editor_ai_cmd_palette_remove_retry: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'editor_ai_cmd_palette_remove_retry'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-05-26
  platform_editor_stop_width_reflows: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_stop_width_reflows'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-05-08
  platform_editor_core_static_emotion: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_core_static_emotion'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-05-13
  editor_ai_comment_freegen_rovo: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'editor_ai_comment_freegen_rovo'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-05-15
  confluence_p2m_style_recalc_and_expand_joint_exp: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'confluence_p2m_style_recalc_and_expand_joint_exp'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-05-21
  platform_editor_nodevisibility: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_nodevisibility'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-05-15
  platform_editor_breakout_resizing: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_breakout_resizing'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-05-30
  platform_editor_exp_disable_lnv: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_exp_disable_lnv'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-05-07
  platform_editor_ai_quickstart_command: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_ai_quickstart_command'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-05-22
  editor_ai_in_editor_streaming: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'editor_ai_in_editor_streaming'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  platform_editor_hide_floating_toolbar_in_ssr: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_hide_floating_toolbar_in_ssr'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  platform_editor_smart_card_open_overlay_perf: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_smart_card_open_overlay_perf'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-05-28
  platform_editor_toolbar_rerender_optimization_exp: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_toolbar_rerender_optimization_exp'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-05-30
  platform_editor_controls_toolbar_pinning_exp: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_controls_toolbar_pinning_exp'
    },
    param: 'isEnabled',
    defaultValue: false
  })
};
