// @atlaskit/smart-card/dist/cjs/view/FlexibleCard/components/blocks/preview-block/resolved/indexOld.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));
var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));
var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));
var _react = require("react");
var _react2 = require("@emotion/react");
var _constants = require("../../../../../../constants");
var _elements = require("../../../elements");
var _block = _interopRequireDefault(require("../../block"));
var _excluded = ["ignoreContainerPadding", "onError", "overrideCss", "placement", "testId", "overrideUrl"];
var _templateObject;
/**
 * @jsxRuntime classic
 * @jsx jsx
 */
// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766
/**
 * Due to its placement on the left/right and ignoreContainerPadding prop
 * rely on its parent container styling, css variables are declared in
 * <Container /> to preset the base values for the preview block styling.
 *
 * `--container-padding` is the padding of the Container. This value is based
 *   on size and hidePadding.
 * `--container-gap-left` and `--container-gap-right` are the gap or padding of
 *   the Container depending on whether the container has other preview blocks
 *   with left/right positioning.
 * `--preview-block-width` is the size of the preview image in relation to
 *   the Container width when the placement is left/right.
 * @param placement
 * @param ignoreContainerPadding
 */
var getPreviewBlockStyles = function getPreviewBlockStyles(placement, ignoreContainerPadding) {
  if (placement === _constants.MediaPlacement.Left || placement === _constants.MediaPlacement.Right) {
    var containerPadding = ignoreContainerPadding ? '0rem' : 'var(--container-padding)';
    // eslint-disable-next-line @atlaskit/design-system/no-css-tagged-template-expression -- needs manual remediation
    return (0, _react2.css)(_templateObject || (_templateObject = (0, _taggedTemplateLiteral2.default)(["\n\t\t\tposition: absolute;\n\t\t\ttop: ", ";\n\t\t\tbottom: ", ";\n\t\t\twidth: calc(var(--preview-block-width) - ", ");\n\n\t\t\t", "\n\t\t\t", "\n\n      [data-smart-element-media='image'] {\n\t\t\t\taspect-ratio: unset;\n\t\t\t\tpadding-top: unset;\n\t\t\t\twidth: 100%;\n\t\t\t\theight: 100%;\n\t\t\t\tobject-fit: cover;\n\t\t\t}\n\t\t"])), containerPadding, containerPadding, containerPadding, placement === _constants.MediaPlacement.Left ? "left: ".concat(containerPadding, ";") : '', placement === _constants.MediaPlacement.Right ? "right: ".concat(containerPadding, ";") : '');
  }
  if (ignoreContainerPadding) {
    return (0, _react2.css)({
      marginLeft: 'calc(var(--container-gap-left) * -1)',
      marginRight: 'calc(var(--container-gap-right) * -1)',
      // eslint-disable-next-line @atlaskit/ui-styling-standard/no-unsafe-selectors -- Ignored via go/DSP-18766
      '&:first-of-type': {
        marginTop: 'calc(var(--container-padding) * -1)'
      },
      // eslint-disable-next-line @atlaskit/ui-styling-standard/no-unsafe-selectors -- Ignored via go/DSP-18766
      '&:last-of-type': {
        marginBottom: 'calc(var(--container-padding) * -1)'
      }
    });
  }
};

/**
 * Represents a resolved PreviewBlock, which typically contains media or other large format content.
 * @public
 * @param {PreviewBlock} PreviewBlock
 * @see Block
 */
var PreviewBlockResolvedViewOld = function PreviewBlockResolvedViewOld(_ref) {
  var _ref$ignoreContainerP = _ref.ignoreContainerPadding,
    ignoreContainerPadding = _ref$ignoreContainerP === void 0 ? false : _ref$ignoreContainerP,
    onError = _ref.onError,
    overrideCss = _ref.overrideCss,
    placement = _ref.placement,
    testId = _ref.testId,
    overrideUrl = _ref.overrideUrl,
    blockProps = (0, _objectWithoutProperties2.default)(_ref, _excluded);
  var _useState = (0, _react.useState)(overrideCss),
    _useState2 = (0, _slicedToArray2.default)(_useState, 2),
    styles = _useState2[0],
    setStyles = _useState2[1];
  var updateStyles = (0, _react.useCallback)(function () {
    // eslint-disable-next-line @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
    setStyles((0, _react2.css)(getPreviewBlockStyles(placement, ignoreContainerPadding), overrideCss));
  }, [ignoreContainerPadding, overrideCss, placement]);
  (0, _react.useEffect)(function () {
    updateStyles();
  }, [ignoreContainerPadding, overrideCss, placement, updateStyles]);
  var handleOnLoad = (0, _react.useCallback)(function () {
    updateStyles();
  }, [updateStyles]);
  var handleOnError = (0, _react.useCallback)(function () {
    if (onError) {
      onError();
    }
  }, [onError]);
  return (0, _react2.jsx)(_block.default, (0, _extends2.default)({}, blockProps, {
    overrideCss: styles,
    testId: "".concat(testId, "-resolved-view")
  }), (0, _react2.jsx)(_elements.Preview, {
    onError: handleOnError,
    onLoad: handleOnLoad,
    overrideUrl: overrideUrl
  }));
};
var _default = exports.default = PreviewBlockResolvedViewOld;
