// @atlaskit/link-provider/dist/cjs/state/context/index.js
"use strict";

var _typeof = require("@babel/runtime/helpers/typeof");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SmartCardContext = exports.EditorSmartCardProviderValueGuard = exports.EditorSmartCardProvider = void 0;
Object.defineProperty(exports, "SmartCardProvider", {
  enumerable: true,
  get: function get() {
    return _provider.SmartCardProvider;
  }
});
exports.useSmartCardContext = exports.default = void 0;
exports.useSmartLinkContext = useSmartLinkContext;
var _react = _interopRequireWildcard(require("react"));
var _provider = require("../../provider");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
var SmartCardContext = exports.SmartCardContext = /*#__PURE__*/(0, _react.createContext)(undefined);
function useSmartLinkContext() {
  var context = (0, _react.useContext)(SmartCardContext);
  if (!context) {
    throw Error('useSmartCard() must be wrapped in <SmartCardProvider>');
  }
  return context;
}
var useSmartCardContext = exports.useSmartCardContext = function useSmartCardContext() {
  return useContextMemoized(SmartCardContext);
};
var EditorSmartCardProviderValueGuard = exports.EditorSmartCardProviderValueGuard = function EditorSmartCardProviderValueGuard(_ref) {
  var children = _ref.children;
  var cardContext = useSmartCardContext();
  if (!(cardContext !== null && cardContext !== void 0 && cardContext.value)) {
    return null;
  }
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, children);
};
function useContextMemoized(reactContext) {
  var value = (0, _react.useContext)(reactContext);
  var context = (0, _react.useMemo)(function () {
    return {
      Provider: reactContext.Provider,
      Consumer: reactContext.Consumer,
      value: value
    };
  }, [value, reactContext]);
  return context;
}
var EditorSmartCardProvider = exports.EditorSmartCardProvider = function EditorSmartCardProvider(_ref2) {
  var children = _ref2.children;
  var cardContext = useSmartCardContext();
  var Provider = cardContext.Provider;
  return /*#__PURE__*/_react.default.createElement(Provider, {
    value: cardContext.value
  }, children);
};
var _default = exports.default = SmartCardContext;
