// @atlaskit/link-datasource/dist/cjs/ui/common/modal/popup-select/menu-list/messages.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.asyncPopupSelectMessages = void 0;
var _reactIntlNext = require("react-intl-next");
var asyncPopupSelectMessages = exports.asyncPopupSelectMessages = (0, _reactIntlNext.defineMessages)({
  loadingMessage: {
    id: 'linkDataSource.basic-filter.loading-message',
    defaultMessage: 'Loading...',
    description: 'The text for when options are being loaded in dropdown'
  },
  noOptionsMessage: {
    id: 'linkDataSource.basic-filter.no-options-message',
    defaultMessage: 'No matches found',
    description: 'The text for when no matches are found in dropdown'
  },
  errorMessage: {
    id: 'linkDataSource.basic-filter.error-message',
    defaultMessage: 'Something went wrong',
    description: 'The text for when an error occurs when loading options'
  },
  showMoreMessage: {
    id: 'linkDataSource.basic-filter.showMoreButton',
    defaultMessage: 'Show more',
    description: 'The text to show more options in dropdown'
  }
});
