// @atlaskit/link-datasource/dist/cjs/ui/common/modal/popup-select/menu-list/selectMessageOld.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = _interopRequireDefault(require("react"));
var _reactIntlNext = require("react-intl-next");
var _heading = _interopRequireDefault(require("@atlaskit/heading"));
var _primitives = require("@atlaskit/primitives");
var boxStyles = (0, _primitives.xcss)({
  height: "var(--ds-space-800, 64px)",
  marginBottom: 'space.200'
});
var stackStyles = (0, _primitives.xcss)({
  paddingTop: 'space.100',
  paddingBottom: 'space.100',
  paddingLeft: 'space.150',
  paddingRight: 'space.150'
});
var CustomSelectMessage = function CustomSelectMessage(_ref) {
  var icon = _ref.icon,
    message = _ref.message,
    testId = _ref.testId;
  return /*#__PURE__*/_react.default.createElement(_primitives.Stack, {
    xcss: stackStyles,
    testId: testId,
    alignInline: "center"
  }, /*#__PURE__*/_react.default.createElement(_primitives.Flex, {
    xcss: boxStyles,
    alignItems: "center",
    justifyContent: "center"
  }, icon), /*#__PURE__*/_react.default.createElement(_heading.default, {
    size: "xsmall"
  }, /*#__PURE__*/_react.default.createElement(_reactIntlNext.FormattedMessage, message)));
};
var _default = exports.default = CustomSelectMessage;
