// @atlaskit/smart-card/dist/cjs/view/InlineCard/ForbiddenView/InlineCardForbiddenViewOld.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.InlineCardForbiddenViewOld = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));
var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));
var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _react = _interopRequireDefault(require("react"));
var _reactIntlNext = require("react-intl-next");
var _button = _interopRequireDefault(require("@atlaskit/button"));
var _lockLocked = _interopRequireDefault(require("@atlaskit/icon/core/lock-locked"));
var _lockFilled = _interopRequireDefault(require("@atlaskit/icon/glyph/lock-filled"));
var _lozenge = _interopRequireDefault(require("@atlaskit/lozenge"));
var _platformFeatureFlags = require("@atlaskit/platform-feature-flags");
var _primitives = require("@atlaskit/primitives");
var _colors = require("@atlaskit/theme/colors");
var _messages = require("../../../messages");
var _HoverCard = require("../../HoverCard");
var _Frame = require("../Frame");
var _Icon = require("../Icon");
var _IconEmotion = require("../Icon-emotion");
var _IconAndTitleLayout = require("../IconAndTitleLayout");
var _styled = require("../styled");
var _styledEmotion = require("../styled-emotion");
var _withFrameStyleControl = _interopRequireDefault(require("../utils/withFrameStyleControl"));
function _callSuper(t, o, e) { return o = (0, _getPrototypeOf2.default)(o), (0, _possibleConstructorReturn2.default)(t, _isNativeReflectConstruct() ? Reflect.construct(o, e || [], (0, _getPrototypeOf2.default)(t).constructor) : o.apply(t, e)); }
function _isNativeReflectConstruct() { try { var t = !Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); } catch (t) {} return (_isNativeReflectConstruct = function _isNativeReflectConstruct() { return !!t; })(); }
var iconWrapperStyles = (0, _primitives.xcss)({
  marginRight: 'space.negative.025'
});
var fallbackForbiddenIcon = function fallbackForbiddenIcon() {
  if ((0, _platformFeatureFlags.fg)('bandicoots-compiled-migration-smartcard')) {
    return (0, _platformFeatureFlags.fg)('platform-smart-card-icon-migration') ? /*#__PURE__*/_react.default.createElement(_primitives.Box, {
      as: "span",
      xcss: iconWrapperStyles
    }, /*#__PURE__*/_react.default.createElement(_lockLocked.default, {
      label: "error",
      color: "var(--ds-icon-danger, #C9372C)",
      LEGACY_fallbackIcon: _lockFilled.default,
      LEGACY_size: "small",
      testId: "forbidden-view-fallback-icon"
    })) : /*#__PURE__*/_react.default.createElement(_Icon.AKIconWrapper, null, /*#__PURE__*/_react.default.createElement(_lockFilled.default, {
      label: "error",
      size: "small",
      primaryColor: "var(--ds-icon-danger, ".concat(_colors.R400, ")"),
      testId: "forbidden-view-fallback-icon"
    }));
  } else {
    return (0, _platformFeatureFlags.fg)('platform-smart-card-icon-migration') ? /*#__PURE__*/_react.default.createElement(_primitives.Box, {
      as: "span",
      xcss: iconWrapperStyles
    }, /*#__PURE__*/_react.default.createElement(_lockLocked.default, {
      label: "error",
      color: "var(--ds-icon-danger, #C9372C)",
      LEGACY_fallbackIcon: _lockFilled.default,
      LEGACY_size: "small",
      testId: "forbidden-view-fallback-icon"
    })) : /*#__PURE__*/_react.default.createElement(_IconEmotion.AKIconWrapper, null, /*#__PURE__*/_react.default.createElement(_lockFilled.default, {
      label: "error",
      size: "small",
      primaryColor: "var(--ds-icon-danger, ".concat(_colors.R400, ")"),
      testId: "forbidden-view-fallback-icon"
    }));
  }
};
var InlineCardForbiddenViewOld = exports.InlineCardForbiddenViewOld = /*#__PURE__*/function (_React$Component) {
  function InlineCardForbiddenViewOld() {
    var _this$props;
    var _this;
    (0, _classCallCheck2.default)(this, InlineCardForbiddenViewOld);
    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }
    _this = _callSuper(this, InlineCardForbiddenViewOld, [].concat(args));
    (0, _defineProperty2.default)(_this, "frameRef", /*#__PURE__*/_react.default.createRef());
    (0, _defineProperty2.default)(_this, "state", {
      hasRequestAccessContextMessage: !!((_this$props = _this.props) !== null && _this$props !== void 0 && (_this$props = _this$props.requestAccessContext) !== null && _this$props !== void 0 && _this$props.callToActionMessageKey)
    });
    (0, _defineProperty2.default)(_this, "handleRetry", function (event) {
      var onAuthorise = _this.props.onAuthorise;
      event.preventDefault();
      event.stopPropagation();
      if (onAuthorise) {
        onAuthorise();
      } else {
        var _this$props2;
        (_this$props2 = _this.props) === null || _this$props2 === void 0 || (_this$props2 = _this$props2.requestAccessContext) === null || _this$props2 === void 0 || (_this$props2 = _this$props2.action) === null || _this$props2 === void 0 || _this$props2.promise();
      }
    });
    (0, _defineProperty2.default)(_this, "renderForbiddenAccessMessage", function () {
      var _this$props3;
      if ((_this$props3 = _this.props) !== null && _this$props3 !== void 0 && (_this$props3 = _this$props3.requestAccessContext) !== null && _this$props3 !== void 0 && _this$props3.callToActionMessageKey) {
        var callToActionMessageKey = _this.props.requestAccessContext.callToActionMessageKey;
        return /*#__PURE__*/_react.default.createElement(_reactIntlNext.FormattedMessage, (0, _extends2.default)({}, _messages.messages[callToActionMessageKey], {
          values: {
            product: _this.props.context
          }
        }));
      }
      return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_reactIntlNext.FormattedMessage, _messages.messages.invalid_permissions, function (formattedMessage) {
        return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, formattedMessage);
      }));
    });
    (0, _defineProperty2.default)(_this, "renderActionButton", function () {
      var _this$props$requestAc;
      var onAuthorise = _this.props.onAuthorise;
      var Button = (0, _withFrameStyleControl.default)(_button.default, _this.frameRef);
      var accessType = (_this$props$requestAc = _this.props.requestAccessContext) === null || _this$props$requestAc === void 0 ? void 0 : _this$props$requestAc.accessType;
      if (_this.state.hasRequestAccessContextMessage) {
        if ((0, _platformFeatureFlags.fg)('bandicoots-compiled-migration-smartcard')) {
          return /*#__PURE__*/_react.default.createElement(Button, {
            spacing: "none",
            onClick: _this.handleRetry,
            component: _styled.IconStyledButtonOldVisualRefresh,
            testId: "button-connect-other-account",
            role: "button",
            isDisabled: accessType === 'PENDING_REQUEST_EXISTS'
          }, _this.renderForbiddenAccessMessage());
        } else {
          return /*#__PURE__*/_react.default.createElement(Button, {
            spacing: "none",
            onClick: _this.handleRetry,
            component: _styledEmotion.IconStyledButton,
            testId: "button-connect-other-account",
            role: "button",
            isDisabled: accessType === 'PENDING_REQUEST_EXISTS'
          }, _this.renderForbiddenAccessMessage());
        }
      }
      if (onAuthorise) {
        return /*#__PURE__*/_react.default.createElement(Button, {
          spacing: "none",
          onClick: _this.handleRetry,
          appearance: "subtle-link",
          testId: "button-connect-other-account",
          role: "button"
        }, /*#__PURE__*/_react.default.createElement(_IconAndTitleLayout.LozengeWrapper, null, /*#__PURE__*/_react.default.createElement(_lozenge.default, (0, _extends2.default)({
          appearance: 'moved'
        }, (0, _platformFeatureFlags.fg)('platform-component-visual-refresh') ? {
          isBold: true
        } : undefined), _this.renderForbiddenAccessMessage())));
      }
      return null;
    });
    return _this;
  }
  (0, _inherits2.default)(InlineCardForbiddenViewOld, _React$Component);
  return (0, _createClass2.default)(InlineCardForbiddenViewOld, [{
    key: "render",
    value: function render() {
      var _this$props4 = this.props,
        url = _this$props4.url,
        icon = _this$props4.icon,
        onClick = _this$props4.onClick,
        isSelected = _this$props4.isSelected,
        _this$props4$testId = _this$props4.testId,
        testId = _this$props4$testId === void 0 ? 'inline-card-forbidden-view' : _this$props4$testId,
        truncateInline = _this$props4.truncateInline;
      var content = /*#__PURE__*/_react.default.createElement(_Frame.Frame, {
        testId: testId,
        isSelected: isSelected,
        ref: this.frameRef,
        truncateInline: truncateInline
      }, /*#__PURE__*/_react.default.createElement(_IconAndTitleLayout.IconAndTitleLayout, {
        icon: icon ? icon : fallbackForbiddenIcon(),
        link: url,
        title: url,
        onClick: onClick,
        titleColor: "var(--ds-text-subtle, ".concat(_colors.N500, ")")
      }), this.renderActionButton());
      if (this.props.showHoverPreview) {
        return /*#__PURE__*/_react.default.createElement(_HoverCard.HoverCard, {
          url: url
        }, content);
      }
      return content;
    }
  }]);
}(_react.default.Component);
