// @atlaskit/smart-card/dist/cjs/view/RelatedLinksModal/components/related-links-list/RelatedLinksListOld.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = require("@emotion/react");
var _reactIntlNext = require("react-intl-next");
var _primitives = require("@atlaskit/primitives");
var _messages = require("../../../../messages");
var _relatedLinkItem = _interopRequireDefault(require("../related-link-item"));
/**
 * @jsxRuntime classic
 * @jsx jsx
 */

// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766

var RelatedLinksList = function RelatedLinksList(_ref) {
  var urls = _ref.urls,
    title = _ref.title,
    testId = _ref.testId;
  return (0, _react.jsx)(_primitives.Stack, {
    testId: testId
  }, (0, _react.jsx)(_primitives.Box, {
    paddingBlockStart: "space.050"
    // eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values, @atlaskit/design-system/consistent-css-prop-usage -- Ignored via go/DSP-18766
    ,
    xcss: (0, _primitives.xcss)({
      font: "var(--ds-font-heading-xxsmall, normal 600 12px/16px ui-sans-serif, -apple-system, BlinkMacSystemFont, \"Segoe UI\", Ubuntu, \"Helvetica Neue\", sans-serif)",
      // eslint-disable-next-line @atlaskit/design-system/use-tokens-typography
      textTransform: 'uppercase'
    })
  }, (0, _react.jsx)(_reactIntlNext.FormattedMessage, title)), urls.length > 0 && (0, _react.jsx)(_primitives.Box, null, urls.map(function (url, idx) {
    return (0, _react.jsx)(_primitives.Stack, {
      key: "".concat(idx, "-").concat(url)
    }, (0, _react.jsx)(_relatedLinkItem.default, {
      url: url,
      testId: "".concat(testId, "-item-").concat(idx)
    }));
  })), urls.length === 0 &&
  // eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values, @atlaskit/design-system/consistent-css-prop-usage -- Ignored via go/DSP-18766
  (0, _react.jsx)(_primitives.Box, {
    paddingBlock: "space.100",
    xcss: (0, _primitives.xcss)({
      font: "var(--ds-font-body-small, normal 400 11px/16px ui-sans-serif, -apple-system, BlinkMacSystemFont, \"Segoe UI\", Ubuntu, \"Helvetica Neue\", sans-serif)"
    })
  }, (0, _react.jsx)(_reactIntlNext.FormattedMessage, _messages.messages.related_links_not_found)));
};
var _default = exports.default = RelatedLinksList;
