// @atlaskit/renderer/dist/cjs/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
var _typeof = require("@babel/runtime/helpers/typeof");
Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "ADFEncoder", {
  enumerable: true,
  get: function get() {
    return _utils.ADFEncoder;
  }
});
Object.defineProperty(exports, "AnnotationsWrapper", {
  enumerable: true,
  get: function get() {
    return _annotations.AnnotationsWrapper;
  }
});
Object.defineProperty(exports, "ReactRenderer", {
  enumerable: true,
  get: function get() {
    return _Renderer.default;
  }
});
Object.defineProperty(exports, "ReactSerializer", {
  enumerable: true,
  get: function get() {
    return _react.default;
  }
});
Object.defineProperty(exports, "RendererWithAnalytics", {
  enumerable: true,
  get: function get() {
    return _Renderer.RendererWithAnalytics;
  }
});
Object.defineProperty(exports, "TextSerializer", {
  enumerable: true,
  get: function get() {
    return _text.default;
  }
});
Object.defineProperty(exports, "ValidationContextProvider", {
  enumerable: true,
  get: function get() {
    return _ValidationContext.ValidationContextProvider;
  }
});
Object.defineProperty(exports, "defaultNodeComponents", {
  enumerable: true,
  get: function get() {
    return _nodes.nodeToReact;
  }
});
Object.defineProperty(exports, "renderDocument", {
  enumerable: true,
  get: function get() {
    return _renderDocument.renderDocument;
  }
});
var _react = _interopRequireDefault(require("./react"));
var _text = _interopRequireDefault(require("./text"));
var _Renderer = _interopRequireWildcard(require("./ui/Renderer"));
var _nodes = require("./react/nodes");
var _annotations = require("./ui/annotations");
var _ValidationContext = require("./ui/Renderer/ValidationContext");
var _utils = require("./utils");
var _renderDocument = require("./render-document");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
