// @atlaskit/smart-card/dist/esm/view/RelatedLinksModal/views/resolving/index.js
/* index.tsx generated by @compiled/babel-plugin v0.36.1 */
import "./index.compiled.css";
import * as React from 'react';
import { ax, ix } from "@compiled/react/runtime";
import { fg } from '@atlaskit/platform-feature-flags';
import { Box } from '@atlaskit/primitives/compiled';
import Spinner from '@atlaskit/spinner';
import RelatedLinksResolvingViewOld from './RelatedLinksResolvingViewOld';
var styles = {
  style: "_4cvr1h6o _1e0c1txw _2lx21bp4 _4t3i1osq _1bah1h6o"
};
var RelatedLinksResolvingViewNew = function RelatedLinksResolvingViewNew() {
  return /*#__PURE__*/React.createElement(Box, {
    xcss: styles.style,
    testId: "related-links-resolving-view"
  }, /*#__PURE__*/React.createElement(Spinner, {
    size: "large",
    testId: "related-links-resolving-view-spinner"
  }));
};
var RelatedLinksResolvingView = function RelatedLinksResolvingView() {
  if (fg('bandicoots-compiled-migration-smartcard')) {
    return /*#__PURE__*/React.createElement(RelatedLinksResolvingViewNew, null);
  }
  return /*#__PURE__*/React.createElement(RelatedLinksResolvingViewOld, null);
};
export default RelatedLinksResolvingView;
