// @atlaskit/renderer/dist/cjs/ui/annotations/hover/range-validator.js
"use strict";

var _typeof = require("@babel/runtime/helpers/typeof");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.RangeValidator = void 0;
var _react = _interopRequireWildcard(require("react"));
var _mounter = require("./mounter");
var _RendererActionsContext = require("../../RendererActionsContext");
var _AnnotationRangeContext = require("../contexts/AnnotationRangeContext");
var _AnnotationHoverContext = require("../contexts/AnnotationHoverContext");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
var RangeValidator = exports.RangeValidator = function RangeValidator(props) {
  var component = props.component,
    rendererRef = props.rendererRef,
    applyAnnotationDraftAt = props.applyAnnotationDraftAt,
    clearAnnotationDraft = props.clearAnnotationDraft,
    createAnalyticsEvent = props.createAnalyticsEvent;
  var actions = (0, _react.useContext)(_RendererActionsContext.RendererContext);
  var _useAnnotationRangeDi = (0, _AnnotationRangeContext.useAnnotationRangeDispatch)(),
    clearHoverRange = _useAnnotationRangeDi.clearHoverRange;
  var _useAnnotationRangeSt = (0, _AnnotationRangeContext.useAnnotationRangeState)(),
    range = _useAnnotationRangeSt.range,
    type = _useAnnotationRangeSt.type;
  var _useAnnotationHoverCo = (0, _AnnotationHoverContext.useAnnotationHoverContext)(),
    isWithinRange = _useAnnotationHoverCo.isWithinRange;
  if (!range || type !== 'hover') {
    return null;
  }
  var documentPosition = actions.getPositionFromRange(range);
  return /*#__PURE__*/_react.default.createElement(_mounter.Mounter, {
    isWithinRange: isWithinRange,
    range: range,
    wrapperDOM: rendererRef,
    component: component,
    onClose: clearHoverRange,
    documentPosition: documentPosition,
    isAnnotationAllowed: true,
    applyAnnotation: actions.applyAnnotation.bind(actions),
    applyAnnotationDraftAt: applyAnnotationDraftAt,
    generateIndexMatch: actions.generateAnnotationIndexMatch.bind(actions),
    clearAnnotationDraft: clearAnnotationDraft,
    createAnalyticsEvent: createAnalyticsEvent
  });
};
