// @atlaskit/link-datasource/dist/es2019/ui/confluence-search-modal/basic-filters/basic-filters-old.js
import React from 'react';
import { Flex, xcss } from '@atlaskit/primitives';
import { DateRangePicker } from './filters/date-range-picker';
import EditedOrCreatedByFilter from './filters/edited-or-created-by';
import { CLOLBasicFilters } from './types';
const basicFilterContainerStyles = xcss({
  paddingLeft: 'space.100'
});
const BasicFilterContainer = ({
  cloudId,
  onChange,
  selections = {},
  isHydrating
}) => {
  const {
    lastModified,
    editedOrCreatedBy
  } = selections;
  const [lastModifiedValue] = lastModified || [];
  const lastModifiedSelection = (lastModifiedValue === null || lastModifiedValue === void 0 ? void 0 : lastModifiedValue.optionType) === 'dateRange' ? lastModifiedValue : undefined;
  const onDateRangePickerChange = updatedOption => {
    onChange(CLOLBasicFilters.lastModified, {
      optionType: 'dateRange',
      label: updatedOption.value,
      ...updatedOption
    });
  };
  return /*#__PURE__*/React.createElement(Flex, {
    xcss: basicFilterContainerStyles,
    gap: "space.100",
    testId: "clol-basic-filter-container"
  }, /*#__PURE__*/React.createElement(EditedOrCreatedByFilter, {
    cloudId: cloudId,
    onSelectionChange: onChange,
    selection: editedOrCreatedBy || [],
    isHydrating: isHydrating
  }), /*#__PURE__*/React.createElement(DateRangePicker, {
    selection: lastModifiedSelection,
    onSelectionChange: onDateRangePickerChange,
    filterName: `clol-basic-filter-${CLOLBasicFilters.lastModified}`
  }));
};
export default BasicFilterContainer;
