// @atlaskit/link-picker/dist/cjs/ui/link-picker/search-results/search-results-container/old/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SearchResultsContainerOld = void 0;
var _react = require("react");
var _react2 = require("@emotion/react");
var _minHeightContainer = require("../../../../../common/ui/min-height-container");
/**
 * @jsxRuntime classic
 * @jsx jsx
 */

// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766

var flexColumn = (0, _react2.css)({
  display: 'flex',
  flexDirection: 'column',
  justifyContent: 'flex-start',
  width: '100%'
});
var SearchResultsContainerOld = exports.SearchResultsContainerOld = function SearchResultsContainerOld(_ref) {
  var hasTabs = _ref.hasTabs,
    adaptiveHeight = _ref.adaptiveHeight,
    isLoadingResults = _ref.isLoadingResults,
    children = _ref.children;
  var ref = (0, _react.useRef)(null);
  var currentHeight = (0, _react.useRef)(null);
  var fixedMinHeight = hasTabs ? '347px' : '302px';
  var adaptiveMinHeight = isLoadingResults && !!currentHeight.current ? "".concat(currentHeight.current, "px") : 'auto';
  var minheight = adaptiveHeight ? adaptiveMinHeight : fixedMinHeight;
  (0, _react.useLayoutEffect)(function () {
    if (ref.current && adaptiveHeight && !isLoadingResults) {
      currentHeight.current = ref.current.getBoundingClientRect().height;
    }
  });
  return (0, _react2.jsx)(_minHeightContainer.MinHeightContainer, {
    ref: ref,
    minHeight: minheight,
    css: flexColumn
  }, children);
};
