// @atlaskit/link-datasource/dist/cjs/ui/assets-modal/modal/render-assets-content/render-assets-content-old.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.RenderAssetsContentOld = exports.MODAL_HEIGHT = void 0;
var _react = require("react");
var _react2 = require("@emotion/react");
var _colors = require("@atlaskit/theme/colors");
var _accessRequired = require("../../../common/error-state/access-required");
var _modalLoadingError = require("../../../common/error-state/modal-loading-error");
var _noResults = require("../../../common/error-state/no-results");
var _issueLikeTableOld = require("../../../issue-like-table/issue-like-table-old");
var _initialStateView = require("./initial-state-view");
/**
 * @jsxRuntime classic
 * @jsx jsx
 */

// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766

var MODAL_HEIGHT = exports.MODAL_HEIGHT = 420;

// This is to prevent y scrollbar when showing table loading state
var disableOverflowStyles = (0, _react2.css)({
  overflow: 'hidden'
});
var contentContainerStyles = (0, _react2.css)({
  height: MODAL_HEIGHT,
  display: 'grid',
  overflow: 'auto'
});
var tableBordersStyles = (0, _react2.css)({
  border: "1px solid ".concat("var(--ds-border, ".concat(_colors.N40, ")")),
  borderTopLeftRadius: "var(--ds-border-radius-200, 8px)",
  borderTopRightRadius: "var(--ds-border-radius-200, 8px)",
  borderBottom: "2px solid ".concat("var(--ds-background-accent-gray-subtler, ".concat(_colors.N40, ")")),
  // eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values, @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
  backgroundImage: _issueLikeTableOld.scrollableContainerShadowsCssComponents.backgroundImage,
  // eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values, @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
  backgroundRepeat: _issueLikeTableOld.scrollableContainerShadowsCssComponents.backgroundRepeat,
  // eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values, @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
  backgroundSize: _issueLikeTableOld.scrollableContainerShadowsCssComponents.backgroundSize,
  // eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values, @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
  backgroundAttachment: _issueLikeTableOld.scrollableContainerShadowsCssComponents.backgroundAttachment
});
var RejectedView = function RejectedView() {
  return (0, _react2.jsx)("div", {
    css: [contentContainerStyles]
  }, (0, _react2.jsx)(_modalLoadingError.ModalLoadingError, null));
};
var UnauthorizedView = function UnauthorizedView() {
  return (0, _react2.jsx)("div", {
    css: [contentContainerStyles]
  }, (0, _react2.jsx)(_accessRequired.AccessRequired, null));
};
var EmptyView = function EmptyView() {
  return (0, _react2.jsx)("div", {
    css: [contentContainerStyles]
  }, (0, _react2.jsx)(_initialStateView.InitialStateView, null));
};
var NoResultsView = function NoResultsView() {
  return (0, _react2.jsx)("div", {
    css: [contentContainerStyles]
  }, (0, _react2.jsx)(_noResults.NoResults, null));
};
var LoadingView = function LoadingView() {
  return (0, _react2.jsx)("div", {
    css: [contentContainerStyles, contentContainerStyles, tableBordersStyles, disableOverflowStyles]
  }, (0, _react2.jsx)(_issueLikeTableOld.EmptyStateOld, {
    testId: "assets-aql-datasource-modal--loading-state"
  }));
};
var RenderAssetsContentOld = exports.RenderAssetsContentOld = function RenderAssetsContentOld(props) {
  var status = props.status,
    responseItems = props.responseItems,
    responseItemIds = props.responseItemIds,
    visibleColumnKeys = props.visibleColumnKeys,
    onNextPage = props.onNextPage,
    hasNextPage = props.hasNextPage,
    loadDatasourceDetails = props.loadDatasourceDetails,
    columns = props.columns,
    defaultVisibleColumnKeys = props.defaultVisibleColumnKeys,
    onVisibleColumnKeysChange = props.onVisibleColumnKeysChange,
    isFetchingInitialData = props.isFetchingInitialData;
  var resolvedWithNoResults = status === 'resolved' && !responseItems.length;
  var issueLikeDataTableView = (0, _react.useMemo)(function () {
    return (0, _react2.jsx)("div", {
      css: [contentContainerStyles, tableBordersStyles]
    }, (0, _react2.jsx)(_issueLikeTableOld.IssueLikeDataTableViewOld, {
      testId: "asset-datasource-table",
      status: status,
      columns: columns,
      items: responseItems,
      itemIds: responseItemIds,
      hasNextPage: hasNextPage,
      visibleColumnKeys: visibleColumnKeys || defaultVisibleColumnKeys,
      onNextPage: onNextPage,
      onLoadDatasourceDetails: loadDatasourceDetails,
      onVisibleColumnKeysChange: onVisibleColumnKeysChange
    }));
  }, [columns, defaultVisibleColumnKeys, hasNextPage, loadDatasourceDetails, onNextPage, onVisibleColumnKeysChange, responseItems, responseItemIds, status, visibleColumnKeys]);
  var renderAssetsContentView = (0, _react.useCallback)(function () {
    if (isFetchingInitialData) {
      // Placing this check first as it's a priority before all others
      return (0, _react2.jsx)(LoadingView, null);
    } else if (status === 'rejected') {
      return (0, _react2.jsx)(RejectedView, null);
    } else if (status === 'unauthorized') {
      return (0, _react2.jsx)(UnauthorizedView, null);
    } else if (status === 'empty') {
      return (0, _react2.jsx)(EmptyView, null);
    } else if (resolvedWithNoResults) {
      return (0, _react2.jsx)(NoResultsView, null);
    } else if (status === 'loading' && !columns.length) {
      return (0, _react2.jsx)(LoadingView, null);
    }
    return issueLikeDataTableView;
  }, [columns.length, isFetchingInitialData, issueLikeDataTableView, resolvedWithNoResults, status]);
  return renderAssetsContentView();
};
