// @atlaskit/link-picker/dist/cjs/ui/link-picker/text-input/old/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.testIds = exports.TextInputOld = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));
var _react = require("react");
var _react2 = require("@emotion/react");
var _form = require("@atlaskit/form");
var _crossCircleSelectClear = _interopRequireDefault(require("@atlaskit/icon/core/migration/cross-circle--select-clear"));
var _primitives = require("@atlaskit/primitives");
var _textfield = _interopRequireDefault(require("@atlaskit/textfield"));
var _tooltip = _interopRequireDefault(require("@atlaskit/tooltip"));
var _conditionalSpotlightTargetWrapper = require("../conditional-spotlight-target-wrapper");
var _utils = require("../utils");
var _excluded = ["name", "label", "autoFocus", "onRedo", "onUndo", "onKeyDown", "onClear", "clearLabel", "error", "spotlightTargetName"];
/* eslint-disable @atlassian/tangerine/import/no-parent-imports */
/**
 * @jsxRuntime classic
 * @jsx jsx
 */
// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766
/**
 * Overidding text input margin top which design system provides as a default spacer
 * but it gets in the way of our layout
 */
// eslint-disable-next-line @atlaskit/ui-styling-standard/no-exported-styles -- Ignored via go/DSP-18766
var fieldStyles = (0, _react2.css)({
  // eslint-disable-next-line @atlaskit/ui-styling-standard/no-nested-selectors -- Ignored via go/DSP-18766
  '> div': {
    marginTop: 0
  },
  // eslint-disable-next-line @atlaskit/ui-styling-standard/no-nested-selectors, @atlaskit/ui-styling-standard/no-unsafe-selectors -- Ignored via go/DSP-18766
  '& + &': {
    marginTop: "var(--ds-space-200, 16px)"
  }
});
var testIds = exports.testIds = {
  urlError: 'link-error',
  clearUrlButton: 'clear-text'
};
var clearTextButtonStyles = (0, _primitives.xcss)({
  padding: 'space.0',
  marginRight: 'space.050',
  backgroundColor: 'color.background.neutral.subtle',
  border: 'none',
  verticalAlign: 'middle'
});
var TextInputOld = exports.TextInputOld = function TextInputOld(_ref) {
  var name = _ref.name,
    label = _ref.label,
    autoFocus = _ref.autoFocus,
    onRedo = _ref.onRedo,
    onUndo = _ref.onUndo,
    onKeyDown = _ref.onKeyDown,
    onClear = _ref.onClear,
    clearLabel = _ref.clearLabel,
    error = _ref.error,
    spotlightTargetName = _ref.spotlightTargetName,
    restProps = (0, _objectWithoutProperties2.default)(_ref, _excluded);
  var inputRef = (0, _react.useRef)(null);
  var handleRef = (0, _react.useCallback)(function (input) {
    if (input) {
      inputRef.current = input;
      if (autoFocus) {
        // Need this to prevent jumping when we render TextInput inside Portal @see ED-2992
        input.focus({
          preventScroll: true
        });
      }
    }
  }, [autoFocus]);
  var handleKeydown = (0, _react.useCallback)(function (e) {
    if (typeof onUndo === 'function' && (0, _utils.isUndoEvent)(e)) {
      e.preventDefault();
      onUndo();
    } else if (typeof onRedo === 'function' && (0, _utils.isRedoEvent)(e)) {
      e.preventDefault();
      onRedo();
    }
    if (onKeyDown) {
      onKeyDown(e);
    }
  }, [onUndo, onRedo, onKeyDown]);
  var handleClear = (0, _react.useCallback)(function (e) {
    var _inputRef$current;
    e.preventDefault();
    e.stopPropagation();
    onClear === null || onClear === void 0 || onClear(name);
    (_inputRef$current = inputRef.current) === null || _inputRef$current === void 0 || _inputRef$current.focus();
  }, [name, onClear]);
  var clearText = restProps.value !== '' && (0, _react2.jsx)(_tooltip.default, {
    content: clearLabel
  }, (0, _react2.jsx)(_primitives.Pressable, {
    xcss: clearTextButtonStyles,
    onClick: handleClear,
    testId: testIds.clearUrlButton
  }, (0, _react2.jsx)(_crossCircleSelectClear.default, {
    LEGACY_size: "medium",
    label: clearLabel || '',
    color: "var(--ds-icon-subtle, #626F86)",
    spacing: "spacious"
  })));
  return (0, _react2.jsx)("div", {
    css: fieldStyles
  }, (0, _react2.jsx)(_form.Field, {
    label: label,
    name: name
  }, function (_ref2) {
    var fieldProps = _ref2.fieldProps;
    return (0, _react2.jsx)(_conditionalSpotlightTargetWrapper.ConditionalSpotlightTargetWrapper, {
      spotlightTargetName: spotlightTargetName
    }, (0, _react2.jsx)(_react.Fragment, null, (0, _react2.jsx)(_textfield.default, (0, _extends2.default)({}, fieldProps, restProps, {
      onKeyDown: handleKeydown,
      ref: handleRef,
      elemAfterInput: clearText,
      isInvalid: !!error
    })), error && (0, _react2.jsx)(_form.ErrorMessage, {
      testId: testIds.urlError
    }, error)));
  }));
};
