// @atlaskit/smart-card/dist/cjs/view/RelatedLinksModal/components/EmptyStateOld.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.EmptyStateOld = void 0;
var _react = _interopRequireDefault(require("react"));
var _heading = _interopRequireDefault(require("@atlaskit/heading"));
var _primitives = require("@atlaskit/primitives");
var emptyStateStyles = (0, _primitives.xcss)({
  alignItems: 'center',
  display: 'flex',
  flexDirection: 'column',
  gap: 'space.100',
  height: '100%',
  justifyContent: 'center',
  marginLeft: 'space.300',
  marginRight: 'space.300',
  textAlign: 'center'
});
var descriptionStyles = (0, _primitives.xcss)({
  marginTop: 'space.0',
  marginBottom: 'space.300',
  color: 'color.text'
});
var EmptyStateOld = exports.EmptyStateOld = function EmptyStateOld(_ref) {
  var testId = _ref.testId,
    header = _ref.header,
    description = _ref.description,
    renderImage = _ref.renderImage;
  return /*#__PURE__*/_react.default.createElement(_primitives.Box, {
    xcss: emptyStateStyles,
    testId: testId
  }, renderImage === null || renderImage === void 0 ? void 0 : renderImage(), /*#__PURE__*/_react.default.createElement(_heading.default, {
    size: "small"
  }, header), description && /*#__PURE__*/_react.default.createElement(_primitives.Box, {
    as: "p",
    xcss: descriptionStyles
  }, description));
};
