// @atlaskit/task-decision/dist/cjs/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.DecisionList = exports.DecisionItem = void 0;
Object.defineProperty(exports, "PubSubSpecialEventType", {
  enumerable: true,
  get: function get() {
    return _types.PubSubSpecialEventType;
  }
});
Object.defineProperty(exports, "ResourcedTaskItem", {
  enumerable: true,
  get: function get() {
    return _ResourcedTaskItem.default;
  }
});
Object.defineProperty(exports, "TaskDecisionResource", {
  enumerable: true,
  get: function get() {
    return _TaskDecisionResource.default;
  }
});
exports.TaskList = exports.TaskItem = void 0;
var _DecisionItem = _interopRequireDefault(require("./components/DecisionItem"));
var _DecisionItem2 = _interopRequireDefault(require("./components/compiled/DecisionItem"));
var _DecisionList = _interopRequireDefault(require("./components/DecisionList"));
var _DecisionList2 = _interopRequireDefault(require("./components/compiled/DecisionList"));
var _ResourcedTaskItem = _interopRequireDefault(require("./components/ResourcedTaskItem"));
var _TaskDecisionResource = _interopRequireDefault(require("./api/TaskDecisionResource"));
var _TaskItem = _interopRequireDefault(require("./components/TaskItem"));
var _TaskItem2 = _interopRequireDefault(require("./components/compiled/TaskItem"));
var _TaskList = _interopRequireDefault(require("./components/TaskList"));
var _TaskList2 = _interopRequireDefault(require("./components/compiled/TaskList"));
var _platformFeatureFlagsReact = require("@atlaskit/platform-feature-flags-react");
var _types = require("./types");
var DecisionItem = exports.DecisionItem = (0, _platformFeatureFlagsReact.componentWithFG)('platform_editor_css_migrate_stage_1', _DecisionItem2.default, _DecisionItem.default);
var DecisionList = exports.DecisionList = (0, _platformFeatureFlagsReact.componentWithFG)('platform_editor_css_migrate_stage_1', _DecisionList2.default, _DecisionList.default);
var TaskItem = exports.TaskItem = (0, _platformFeatureFlagsReact.componentWithFG)('platform_editor_css_migrate_stage_1', _TaskItem2.default, _TaskItem.default);
var TaskList = exports.TaskList = (0, _platformFeatureFlagsReact.componentWithFG)('platform_editor_css_migrate_stage_1', _TaskList2.default, _TaskList.default);
