// @atlaskit/link-datasource/dist/es2019/ui/common/modal/popup-select/menu-list/selectMessage.js
/* selectMessage.tsx generated by @compiled/babel-plugin v0.36.1 */
import "./selectMessage.compiled.css";
import { ax, ix } from "@compiled/react/runtime";
import React from 'react';
import { FormattedMessage } from 'react-intl-next';
import Heading from '@atlaskit/heading';
import { fg } from '@atlaskit/platform-feature-flags';
import { Flex, Stack } from '@atlaskit/primitives/compiled';
import CustomSelectMessageOld from './selectMessageOld';
const styles = {
  boxStyles: "_4t3ivbgk _otyrpxbi",
  stackStyles: "_ca0qu2gc _n3tdu2gc _19bvutpp _u5f3utpp"
};
const CustomSelectMessage = ({
  icon,
  message,
  testId
}) => {
  return /*#__PURE__*/React.createElement(Stack, {
    xcss: styles.stackStyles,
    testId: testId,
    alignInline: "center"
  }, /*#__PURE__*/React.createElement(Flex, {
    xcss: styles.boxStyles,
    alignItems: "center",
    justifyContent: "center"
  }, icon), /*#__PURE__*/React.createElement(Heading, {
    size: "xsmall"
  }, /*#__PURE__*/React.createElement(FormattedMessage, message)));
};
export const CustomSelectMessageExported = props => {
  if (fg('bandicoots-compiled-migration-link-datasource')) {
    return /*#__PURE__*/React.createElement(CustomSelectMessage, props);
  } else {
    return /*#__PURE__*/React.createElement(CustomSelectMessageOld, props);
  }
};
export default CustomSelectMessageExported;
