// @atlaskit/smart-card/dist/cjs/view/FlexibleCard/components/blocks/element-group/ElementGroupOld.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getElementGroupStyles = exports.default = void 0;
var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));
var _react = require("@emotion/react");
var _constants = require("../../../../../constants");
var _utils = require("../../utils");
var _utils2 = require("../utils");
var _templateObject;
/**
 * @jsxRuntime classic
 * @jsx jsx
 */
// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766
var getAlignmentStyles = function getAlignmentStyles(align) {
  switch (align) {
    case _constants.SmartLinkAlignment.Right:
      return (0, _react.css)({
        WebkitBoxAlign: 'end',
        MsFlexAlign: 'end',
        justifyContent: 'flex-end',
        textAlign: 'right'
      });
    case _constants.SmartLinkAlignment.Left:
    default:
      return (0, _react.css)({
        WebkitBoxAlign: 'start',
        MsFlexAlign: 'start',
        justifyContent: 'flex-start',
        textAlign: 'left'
      });
  }
};
var getGapStyles = function getGapStyles(size, align) {
  var gap = (0, _utils2.getGapSize)(size);
  if (align === _constants.SmartLinkAlignment.Right) {
    return (0, _react.css)({
      // eslint-disable-next-line @atlaskit/ui-styling-standard/no-nested-selectors -- Ignored via go/DSP-18766
      '> span': {
        // eslint-disable-next-line @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
        marginLeft: "".concat(gap, "rem")
      },
      // eslint-disable-next-line @atlaskit/ui-styling-standard/no-nested-selectors, @atlaskit/ui-styling-standard/no-unsafe-selectors -- Ignored via go/DSP-18766
      '> span:first-child': {
        marginLeft: 'initial'
      }
    });
  }
  return (0, _react.css)({
    // eslint-disable-next-line @atlaskit/ui-styling-standard/no-nested-selectors -- Ignored via go/DSP-18766
    '> span': {
      // eslint-disable-next-line @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
      marginRight: "".concat(gap, "rem"),
      // eslint-disable-next-line @atlaskit/ui-styling-standard/no-unsafe-selectors -- Ignored via go/DSP-18766
      '&:last-child': {
        marginRight: 'initial'
      }
    }
  });
};
var getHorizontalDirectionStyles = function getHorizontalDirectionStyles(size, align) {
  var lineHeight = (0, _utils.getMaxLineHeight)(size);
  // eslint-disable-next-line @atlaskit/design-system/no-css-tagged-template-expression -- needs manual remediation
  return (0, _react.css)(_templateObject || (_templateObject = (0, _taggedTemplateLiteral2.default)(["\n\t\tdisplay: block;\n\t\tvertical-align: middle;\n\t\t", "\n\n\t\t> span, > div {\n\t\t\tvertical-align: middle;\n\n\t\t\t&[data-smart-element-date-time],\n\t\t\t&[data-smart-element-text] {\n\t\t\t\t// Show all/wrapped/truncated\n\t\t\t\tdisplay: inline;\n\t\t\t}\n\t\t}\n\n\t\t", "\n\t"])), (0, _utils.getTruncateStyles)(1, lineHeight + 'rem'), getGapStyles(size, align));
};
var getElementGroupStyles = exports.getElementGroupStyles = function getElementGroupStyles(direction, size, align, width, position) {
  return (0, _react.css)(
  // eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values, @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
  (0, _utils2.getBaseStyles)(direction, size),
  // eslint-disable-next-line @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
  getAlignmentStyles(align), {
    minWidth: '10%'
  },
  // eslint-disable-next-line @atlaskit/ui-styling-standard/no-unsafe-values, @atlaskit/ui-styling-standard/no-imported-style-values -- Ignored via go/DSP-18766
  width === _constants.SmartLinkWidth.Flexible ? "flex: 1 3;" : '',
  // eslint-disable-next-line @atlaskit/ui-styling-standard/no-unsafe-values, @atlaskit/ui-styling-standard/no-imported-style-values -- Ignored via go/DSP-18766
  direction === _constants.SmartLinkDirection.Horizontal ? getHorizontalDirectionStyles(size, align) : '',
  // eslint-disable-next-line @atlaskit/ui-styling-standard/no-unsafe-values, @atlaskit/ui-styling-standard/no-imported-style-values -- Ignored via go/DSP-18766
  position === _constants.SmartLinkPosition.Top ? 'align-self: flex-start;' : '');
};

/**
 * Creates a group of Action components. Accepts an array of Actions, in addition to some styling
 * preferences.
 * @internal
 * @param {ActionGroupProps} ActionGroupProps
 * @see Action
 */
var ElementGroup = function ElementGroup(_ref) {
  var _ref$align = _ref.align,
    align = _ref$align === void 0 ? _constants.SmartLinkAlignment.Left : _ref$align,
    children = _ref.children,
    overrideCss = _ref.overrideCss,
    _ref$direction = _ref.direction,
    direction = _ref$direction === void 0 ? _constants.SmartLinkDirection.Horizontal : _ref$direction,
    _ref$size = _ref.size,
    size = _ref$size === void 0 ? _constants.SmartLinkSize.Medium : _ref$size,
    _ref$testId = _ref.testId,
    testId = _ref$testId === void 0 ? 'smart-element-group' : _ref$testId,
    _ref$width = _ref.width,
    width = _ref$width === void 0 ? _constants.SmartLinkWidth.FitToContent : _ref$width,
    _ref$position = _ref.position,
    position = _ref$position === void 0 ? _constants.SmartLinkPosition.Center : _ref$position;
  return (0, _react.jsx)("div", {
    // eslint-disable-next-line @atlaskit/design-system/consistent-css-prop-usage -- Ignored via go/DSP-18766
    css: [getElementGroupStyles(direction, size, align, width, position), overrideCss],
    "data-smart-element-group": true,
    "data-testid": testId
  }, (0, _utils2.renderChildren)(children, size));
};
var _default = exports.default = ElementGroup;
