// @atlaskit/smart-card/dist/cjs/view/FlexibleCard/components/actions/automation-action/automation-manual-triggers/manual-triggers-modal/sub-components/error-state/AutomationModalErrorStateOld.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.AutomationModalErrorStateOld = void 0;
var _react = _interopRequireDefault(require("react"));
var _reactIntlNext = require("react-intl-next");
var _primitives = require("@atlaskit/primitives");
var _errorIcon = _interopRequireDefault(require("./error-icon"));
var i18n = (0, _reactIntlNext.defineMessages)({
  errorDescription: {
    id: 'automation-menu.modal.error.description',
    defaultMessage: 'This is taking awhile to load. Try refreshing the page.',
    description: "Description for the error section shown when rules can't be fetched"
  },
  errorIconLabel: {
    id: 'automation-menu.modal.error.image.alt',
    defaultMessage: 'Error icon',
    description: 'Alternative text for the image that renders when there is an error.'
  }
});
var errorStateStyles = (0, _primitives.xcss)({
  textAlign: 'center',
  marginTop: 'space.500'
});
var imageContainerStyles = (0, _primitives.xcss)({
  display: 'block',
  maxWidth: '116px',
  maxHeight: '156px',
  marginBottom: 'space.300',
  marginTop: 'space.050'
});
var descriptionStyles = (0, _primitives.xcss)({
  maxWidth: '360px'
});
var AutomationModalErrorStateOld = exports.AutomationModalErrorStateOld = function AutomationModalErrorStateOld() {
  var _useIntl = (0, _reactIntlNext.useIntl)(),
    formatMessage = _useIntl.formatMessage;
  return /*#__PURE__*/_react.default.createElement(_primitives.Stack, {
    xcss: errorStateStyles,
    alignInline: "center",
    alignBlock: "center"
  }, /*#__PURE__*/_react.default.createElement(_primitives.Box, {
    xcss: imageContainerStyles
  }, /*#__PURE__*/_react.default.createElement(_errorIcon.default, null)), /*#__PURE__*/_react.default.createElement(_primitives.Box, {
    xcss: descriptionStyles
  }, formatMessage(i18n.errorDescription)));
};
