// @atlaskit/task-decision/dist/cjs/components/DecisionItem.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = require("@emotion/react");
var _decisionEditorDecision = _interopRequireDefault(require("@atlaskit/icon/core/migration/decision--editor-decision"));
var _Item = _interopRequireDefault(require("./Item"));
/**
 * @jsxRuntime classic
 * @jsx jsx
 */

// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766

var iconStyles = function iconStyles(showPlaceholder) {
  return (0, _react.css)({
    flex: '0 0 16px',
    height: '16px',
    width: '16px',
    margin: "var(--ds-space-050, 4px)".concat(" ", "var(--ds-space-150, 12px)", " 0 0"),
    // eslint-disable-next-line @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
    color: showPlaceholder ? "var(--ds-icon-subtle, #626F86)" : "var(--ds-icon-success, #22A06B)",
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center'
  });
};
var DecisionItem = function DecisionItem(_ref) {
  var appearance = _ref.appearance,
    children = _ref.children,
    contentRef = _ref.contentRef,
    placeholder = _ref.placeholder,
    showPlaceholder = _ref.showPlaceholder,
    dataAttributes = _ref.dataAttributes;
  var icon =
  // eslint-disable-next-line @atlaskit/design-system/consistent-css-prop-usage -- Ignored via go/DSP-18766
  (0, _react.jsx)("span", {
    contentEditable: false,
    css: iconStyles(showPlaceholder)
  }, (0, _react.jsx)(_decisionEditorDecision.default, {
    label: "Decision",
    spacing: "spacious",
    color: "currentColor",
    LEGACY_size: "large"
  }));
  return (0, _react.jsx)(_Item.default, {
    appearance: appearance,
    contentRef: contentRef,
    icon: icon,
    placeholder: placeholder,
    showPlaceholder: showPlaceholder,
    itemType: "DECISION",
    dataAttributes: dataAttributes
  }, children);
};
var _default = exports.default = DecisionItem;
