// @atlaskit/task-decision/dist/cjs/components/theme.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _colors = require("@atlaskit/theme/colors");
/**
 * Styles taken from packages/design-system/checkbox/src/internal/theme.tsx
 * To be used until mobile editor does not require legacy themed() API anymore
 */
var checkboxTheme = {
  light: {
    borderColor: {
      rest: "var(--ds-border-input, ".concat(_colors.N100, ")"),
      hovered: "var(--ds-border-input, ".concat(_colors.N100, ")"),
      disabled: "var(--ds-background-disabled, ".concat(_colors.N20, ")"),
      checked: "var(--ds-background-selected-bold, ".concat(_colors.B400, ")"),
      active: "var(--ds-border, ".concat(_colors.B50, ")"),
      focused: "var(--ds-border-focused, ".concat(_colors.B100, ")"),
      hoveredAndChecked: "var(--ds-background-selected-bold-hovered, ".concat(_colors.B300, ")")
    },
    boxColor: {
      rest: "var(--ds-background-input, ".concat(_colors.N10, ")"),
      hovered: "var(--ds-background-input-hovered, ".concat(_colors.N30, ")"),
      disabled: "var(--ds-background-disabled, ".concat(_colors.N20, ")"),
      active: "var(--ds-background-input-pressed, ".concat(_colors.B50, ")"),
      hoveredAndChecked: "var(--ds-background-selected-bold-hovered, ".concat(_colors.B300, ")"),
      checked: "var(--ds-background-selected-bold, ".concat(_colors.B400, ")")
    },
    tickColor: {
      disabledAndChecked: "var(--ds-icon-disabled, ".concat(_colors.N70, ")"),
      activeAndChecked: "var(--ds-icon-inverse, ".concat(_colors.B400, ")"),
      checked: "var(--ds-icon-inverse, ".concat(_colors.N10, ")")
    }
  },
  /**
   * Fallback colours for dark mode taken from
   * packages/design-system/tokens/src/artifacts/themes/atlassian-dark.tsx
   * To be used to keep mobile / web checkbox dark mode consistent
   * until mobile editor does not require legacy themed() API anymore
   */
  dark: {
    borderColor: {
      rest: "var(--ds-border-input, ".concat(_colors.DN200, ")"),
      hovered: "var(--ds-border-input, ".concat(_colors.DN200, ")"),
      disabled: "var(--ds-background-disabled, #BCD6F00A)",
      checked: "var(--ds-background-selected-bold, #579DFF)",
      active: "var(--ds-border, #A6C5E229)",
      focused: "var(--ds-border-focused, #85B8FF)",
      hoveredAndChecked: "var(--ds-background-selected-bold-hovered, #85B8FF)"
    },
    boxColor: {
      rest: "var(--ds-background-input, #22272B)",
      hovered: "var(--ds-background-input-hovered, #282E33)",
      disabled: "var(--ds-background-disabled, #BCD6F00A)",
      active: "var(--ds-background-input-pressed, #22272B)",
      hoveredAndChecked: "var(--ds-background-selected-bold-hovered, #85B8FF)",
      checked: "var(--ds-background-selected-bold, #579DFF)"
    },
    tickColor: {
      disabledAndChecked: "var(--ds-icon-disabled, #BFDBF847)",
      activeAndChecked: "var(--ds-icon-inverse, #1D2125)",
      checked: "var(--ds-icon-inverse, #1D2125)"
    }
  }
};
var _default = exports.default = checkboxTheme;
