// @atlaskit/link-provider/dist/es2019/provider.js
import React, { useContext, useMemo } from 'react';
import { createStore } from 'redux';
import { getUrl } from '@atlaskit/linking-common';
import { fg } from '@atlaskit/platform-feature-flags';
import { extractPreview, extractSmartLinkEmbed } from '@atlaskit/link-extractors';
import { cardReducer } from './reducers';
import { SmartCardContext } from './state/context';
import CardClient from './client';
export function SmartCardProvider({
  storeOptions,
  client: customClient,
  authFlow: customAuthFlow,
  children,
  renderers,
  featureFlags,
  isAdminHubAIEnabled,
  product,
  shouldControlDataExport
}) {
  const parentContext = useContext(SmartCardContext);
  const defaultInitialState = useMemo(() => {
    return {};
  }, []);
  const {
    initialState
  } = storeOptions || {
    initialState: defaultInitialState
  };
  const store = useMemo(() => {
    return createStore(cardReducer, initialState);
  }, [initialState]);
  const providerValue = useMemo(() => {
    const client = customClient || new CardClient();

    // If product is passed into provider, set it on the client if setProduct is supported
    if (product && client.setProduct) {
      client.setProduct(product);
    }
    const authFlow = customAuthFlow || 'oauth2';
    const getPreview = (url, platform) => {
      const cardState = getUrl(store, url);
      return cardState.details ? fg('smart_links_noun_support') ? extractSmartLinkEmbed(cardState.details) : extractPreview(cardState.details.data, platform) : undefined;
    };
    return {
      renderers,
      store,
      prefetchStore: {},
      connections: {
        client
      },
      config: {
        authFlow
      },
      extractors: {
        getPreview
      },
      featureFlags,
      isAdminHubAIEnabled,
      product,
      shouldControlDataExport
    };
  }, [customClient, customAuthFlow, isAdminHubAIEnabled, product, shouldControlDataExport, renderers, featureFlags, store]);
  return /*#__PURE__*/React.createElement(SmartCardContext.Provider, {
    value: parentContext || providerValue
  }, children);
}
export default SmartCardProvider;
