// @atlaskit/link-picker/dist/cjs/ui/link-picker/search-results/link-search-list/old/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.testIds = exports.messages = exports.LinkSearchListOld = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _react = require("react");
var _react2 = require("@emotion/react");
var _reactIntlNext = require("react-intl-next");
var _primitives = require("@atlaskit/primitives");
var _spinner = _interopRequireDefault(require("@atlaskit/spinner"));
var _visuallyHidden = _interopRequireDefault(require("@atlaskit/visually-hidden"));
var _minHeightContainer = require("../../../../../common/ui/min-height-container");
var _handleNavKeyDown = require("../../../../../common/utils/handleNavKeyDown");
var _linkSearchNoResults = require("../link-search-no-results");
var _listItem = require("../list-item");
var _useTrackResultsShown = require("../use-track-results-shown");
var _styled = require("./styled");
var _excluded = ["onChange", "onSelect", "onKeyDown", "items", "activeIndex", "selectedIndex", "isLoading", "ariaControls", "ariaLabelledBy", "ariaReadOnly", "role", "id", "hasSearchTerm", "activePlugin", "adaptiveHeight"];
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { (0, _defineProperty2.default)(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; } /* eslint-disable @atlassian/tangerine/import/no-parent-imports */ /**
 * @jsxRuntime classic
 * @jsx jsx
 */ // eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766
var messages = exports.messages = (0, _reactIntlNext.defineMessages)({
  titleRecentlyViewed: {
    id: 'fabric.linkPicker.listTitle.recentlyViewed',
    defaultMessage: 'Recently Viewed',
    description: 'Describes type of items shown in the list for screen-reader users'
  },
  titleResults: {
    id: 'fabric.linkPicker.listTitle.results',
    defaultMessage: 'Results',
    description: 'Describes type of items shown in the list for screen-reader users'
  },
  searchLinkResults: {
    id: 'fabric.linkPicker.hyperlink.searchLinkResults',
    defaultMessage: '{count, plural, =0 {no results} one {# result} other {# results}} found',
    description: 'Announce search results for screen-reader users.'
  }
});
var testIds = exports.testIds = _objectSpread(_objectSpread(_objectSpread({}, _linkSearchNoResults.testIds), _listItem.testIds), {}, {
  resultListTitle: 'link-picker-list-title',
  searchResultList: 'link-search-list',
  searchResultLoadingIndicator: 'link-picker.results-loading-indicator',
  tabsLoadingIndicator: 'link-picker.tabs-loading-indicator'
});
var emptyStateNoResultsWrapper = (0, _primitives.xcss)({
  minHeight: 'space.200'
});
var LinkSearchListOld = exports.LinkSearchListOld = /*#__PURE__*/(0, _react.forwardRef)(function (_ref, ref) {
  var _activePlugin$uiOptio;
  var onChange = _ref.onChange,
    onSelect = _ref.onSelect,
    onKeyDown = _ref.onKeyDown,
    items = _ref.items,
    activeIndex = _ref.activeIndex,
    selectedIndex = _ref.selectedIndex,
    isLoading = _ref.isLoading,
    ariaControls = _ref.ariaControls,
    ariaLabelledBy = _ref.ariaLabelledBy,
    ariaReadOnly = _ref.ariaReadOnly,
    role = _ref.role,
    id = _ref.id,
    hasSearchTerm = _ref.hasSearchTerm,
    activePlugin = _ref.activePlugin,
    adaptiveHeight = _ref.adaptiveHeight,
    restProps = (0, _objectWithoutProperties2.default)(_ref, _excluded);
  var itemsContent;
  var loadingContent;
  var linkListTitle = hasSearchTerm ? messages.titleResults : messages.titleRecentlyViewed;
  (0, _useTrackResultsShown.useTrackResultsShown)(isLoading, items, hasSearchTerm);
  var itemRefs = (0, _react.useRef)({});
  var itemRefCallback = (0, _react.useCallback)(function (el, id) {
    if (el === null) {
      delete itemRefs.current[id];
    } else {
      itemRefs.current[id] = el;
    }
  }, []);
  var getTabIndex = (0, _react.useCallback)(function (index) {
    if (selectedIndex > -1) {
      return selectedIndex === index ? 0 : -1;
    }
    if (index === 0) {
      return 0;
    }
    return -1;
  }, [selectedIndex]);
  var handleKeyDown = (0, _react.useCallback)(function (event) {
    var updatedIndex = activeIndex;
    if (onKeyDown) {
      onKeyDown(event);
    }
    if (!(items !== null && items !== void 0 && items.length)) {
      return;
    }
    updatedIndex = (0, _handleNavKeyDown.handleNavKeyDown)(event, items.length, activeIndex);
    var item = items === null || items === void 0 ? void 0 : items[updatedIndex];
    if (['ArrowDown', 'ArrowUp', 'Home', 'End'].includes(event.key) && item) {
      if (itemRefs.current) {
        var _itemRefs$current$ite;
        (_itemRefs$current$ite = itemRefs.current[item.objectId]) === null || _itemRefs$current$ite === void 0 || _itemRefs$current$ite.focus();
      }
    }
  }, [activeIndex, items, onKeyDown]);
  if ((items === null || items === void 0 ? void 0 : items.length) === 0) {
    if (!hasSearchTerm) {
      var _activePlugin$emptySt;
      var emptyState = activePlugin === null || activePlugin === void 0 || (_activePlugin$emptySt = activePlugin.emptyStateNoResults) === null || _activePlugin$emptySt === void 0 ? void 0 : _activePlugin$emptySt.call(activePlugin);
      if (emptyState) {
        return (0, _react2.jsx)(_primitives.Box, {
          xcss: emptyStateNoResultsWrapper
        }, emptyState);
      }
    }
    return (0, _react2.jsx)(_linkSearchNoResults.NoResults, null);
  }
  var listItemNameMaxLines = activePlugin === null || activePlugin === void 0 || (_activePlugin$uiOptio = activePlugin.uiOptions) === null || _activePlugin$uiOptio === void 0 ? void 0 : _activePlugin$uiOptio.listItemNameMaxLines;
  if (items && items.length > 0) {
    itemsContent = (0, _react2.jsx)(_react.Fragment, null, (0, _react2.jsx)("div", {
      // eslint-disable-next-line @atlaskit/design-system/consistent-css-prop-usage, @atlaskit/ui-styling-standard/no-imported-style-values -- Ignored via go/DSP-18766
      css: _styled.listTitleStyles,
      id: testIds.resultListTitle,
      "data-testid": testIds.resultListTitle
    }, (0, _react2.jsx)(_reactIntlNext.FormattedMessage, linkListTitle)), (0, _react2.jsx)(_visuallyHidden.default, {
      id: "fabric.smartcard.linkpicker.suggested.results"
    }, hasSearchTerm && (0, _react2.jsx)(_reactIntlNext.FormattedMessage, (0, _extends2.default)({}, messages.searchLinkResults, {
      values: {
        count: items.length
      },
      "aria-live": "polite",
      "aria-atomic": "true"
    }))), (0, _react2.jsx)("ul", {
      id: id,
      role: role
      // eslint-disable-next-line @atlaskit/design-system/consistent-css-prop-usage, @atlaskit/ui-styling-standard/no-imported-style-values
      ,
      css: _styled.listStyles,
      "aria-controls": "fabric.smartcard.linkpicker.suggested.results",
      "aria-labelledby": testIds.resultListTitle,
      "aria-readonly": ariaReadOnly,
      "data-testid": testIds.searchResultList
    }, items.map(function (item, index) {
      return (0, _react2.jsx)(_listItem.LinkSearchListItem, {
        id: "".concat(testIds.searchResultItem, "-").concat(index),
        role: role && 'option',
        item: item,
        selected: selectedIndex === index,
        active: activeIndex === index,
        onFocus: function onFocus() {
          return onChange(item.objectId);
        },
        onKeyDown: handleKeyDown,
        onSelect: onSelect,
        key: item.objectId,
        tabIndex: getTabIndex(index),
        ref: function ref(el) {
          return itemRefCallback(el, item.objectId);
        },
        nameMaxLines: listItemNameMaxLines
      });
    })));
  }
  if (isLoading) {
    loadingContent = (0, _react2.jsx)(_minHeightContainer.MinHeightContainer, {
      minHeight: adaptiveHeight ? '80px' : '50px'
      // eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values
      ,
      css: _styled.spinnerContainerStyles
    }, (0, _react2.jsx)(_spinner.default, {
      testId: testIds.searchResultLoadingIndicator,
      interactionName: "link-picker-search-list-loading",
      size: "medium"
    }));
  }
  return (
    // eslint-disable-next-line @atlaskit/design-system/consistent-css-prop-usage, @atlaskit/ui-styling-standard/no-imported-style-values -- Ignored via go/DSP-18766
    (0, _react2.jsx)("div", (0, _extends2.default)({
      ref: ref,
      css: _styled.listContainerStyles
    }, restProps), itemsContent, loadingContent)
  );
});
