// @atlaskit/smart-card/dist/esm/view/EmbedModal/components/embed-content/index.js
/* index.tsx generated by @compiled/babel-plugin v0.36.1 */
import _extends from "@babel/runtime/helpers/extends";
import "./index.compiled.css";
import * as React from 'react';
import { ax, ix } from "@compiled/react/runtime";
import { di } from 'react-magnetic-di';
import { fg } from '@atlaskit/platform-feature-flags';
import { getIframeSandboxAttribute } from '../../../../utils';
import { IFrame } from '../../../EmbedCard/components/IFrame';
import EmbedContentOld from './indexOld';
var iframeCss = null;
var EmbedContent = function EmbedContent(_ref) {
  var isTrusted = _ref.isTrusted,
    name = _ref.name,
    src = _ref.src,
    testId = _ref.testId;
  var sandbox = getIframeSandboxAttribute(isTrusted);
  var props = {
    frameBorder: 0,
    name: name,
    sandbox: sandbox,
    src: src,
    'data-testid': "".concat(testId, "-embed")
  };
  return /*#__PURE__*/React.createElement(IFrame, _extends({}, props, {
    className: ax(["_1bsb1osq _4t3i815d"])
  }));
};
var Exported = function Exported(props) {
  if (fg('bandicoots-compiled-migration-smartcard')) {
    return /*#__PURE__*/React.createElement(EmbedContent, props);
  } else {
    return /*#__PURE__*/React.createElement(EmbedContentOld, props);
  }
};
export default Exported;
