// @atlaskit/link-picker/dist/esm/ui/link-picker/search-results/search-results-container/index.js
/* index.tsx generated by @compiled/babel-plugin v0.36.1 */
import "./index.compiled.css";
import * as React from 'react';
import { ax, ix } from "@compiled/react/runtime";
import { useLayoutEffect, useRef } from 'react';
import { fg } from '@atlaskit/platform-feature-flags';
import { MinHeightContainer } from '../../../../common/ui/min-height-container';
import { SearchResultsContainerOld } from './old';
var flexColumn = null;
export var SearchResultsContainerNew = function SearchResultsContainerNew(_ref) {
  var hasTabs = _ref.hasTabs,
    adaptiveHeight = _ref.adaptiveHeight,
    isLoadingResults = _ref.isLoadingResults,
    children = _ref.children;
  var ref = useRef(null);
  var currentHeight = useRef(null);
  var fixedMinHeight = hasTabs ? '347px' : '302px';
  var adaptiveMinHeight = isLoadingResults && !!currentHeight.current ? "".concat(currentHeight.current, "px") : 'auto';
  var minheight = adaptiveHeight ? adaptiveMinHeight : fixedMinHeight;
  useLayoutEffect(function () {
    if (ref.current && adaptiveHeight && !isLoadingResults) {
      currentHeight.current = ref.current.getBoundingClientRect().height;
    }
  });
  return /*#__PURE__*/React.createElement(MinHeightContainer, {
    ref: ref,
    minHeight: minheight,
    className: ax(["_1e0c1txw _2lx21bp4 _1bah1y6m _1bsb1osq"])
  }, children);
};
export var SearchResultsContainer = function SearchResultsContainer(props) {
  if (fg('platform_bandicoots-link-picker-css')) {
    return /*#__PURE__*/React.createElement(SearchResultsContainerNew, props);
  }
  return /*#__PURE__*/React.createElement(SearchResultsContainerOld, props);
};
