// @atlaskit/link-datasource/dist/es2019/ui/common/modal/popup-select/footer-old.js
import _extends from "@babel/runtime/helpers/extends";
import React from 'react';
import { FormattedMessage } from 'react-intl-next';
import { Flex, Inline, xcss } from '@atlaskit/primitives';
import { N40 } from '@atlaskit/theme/colors';
import { asyncPopupSelectMessages } from './messages';
const footerContainerStyles = xcss({
  paddingTop: 'space.050',
  paddingBottom: 'space.050',
  borderTop: `${`var(--ds-border, ${N40})`} solid 1px`
});
const footerPaginationInfoStyles = xcss({
  color: 'color.text.subtlest',
  marginBlock: 'space.100',
  marginInline: 'space.150'
});
const PopupFooter = ({
  currentDisplayCount,
  totalCount,
  filterName
}) => {
  return /*#__PURE__*/React.createElement(Flex, {
    testId: `${filterName}--footer`,
    direction: "row",
    alignItems: "center",
    justifyContent: "end",
    xcss: footerContainerStyles
  }, /*#__PURE__*/React.createElement(Inline, {
    xcss: footerPaginationInfoStyles
  }, /*#__PURE__*/React.createElement(FormattedMessage, _extends({}, asyncPopupSelectMessages.paginationDetails, {
    values: {
      currentDisplayCount,
      totalCount
    }
  }))));
};
export default PopupFooter;
