// @atlaskit/link-datasource/dist/es2019/ui/common/modal/popup-select/formatOptionLabelOld.js
import React from 'react';
import Avatar from '@atlaskit/avatar';
import PeopleGroupIcon from '@atlaskit/icon/core/migration/people-group';
import Lozenge from '@atlaskit/lozenge';
import { Box, Flex, xcss } from '@atlaskit/primitives';
const commonLabelStyles = xcss({
  overflow: 'hidden',
  textOverflow: 'ellipsis'
});
const avatarOptionLabelStyles = xcss({
  marginLeft: 'space.050'
});
const groupWrapperStyles = xcss({
  width: "var(--ds-space-250, 20px)",
  minWidth: "var(--ds-space-250, 20px)",
  height: "var(--ds-space-250, 20px)"
});
const IconOptionLabel = ({
  data
}) => {
  const {
    label,
    icon: avatar,
    value
  } = data;
  const avatarOptionLabelData = {
    label,
    avatar,
    value,
    optionType: 'avatarLabel',
    isSquare: true
  };
  return /*#__PURE__*/React.createElement(AvatarOptionLabel, {
    data: avatarOptionLabelData,
    testId: "basic-filter-popup-select-option--icon-label"
  });
};
const LozengeOptionLabel = ({
  data
}) => {
  return /*#__PURE__*/React.createElement(Lozenge, {
    appearance: data.appearance,
    testId: "basic-filter-popup-select-option--lozenge"
  }, /*#__PURE__*/React.createElement(Box, {
    xcss: [commonLabelStyles]
  }, data.label));
};
const AvatarOptionLabel = ({
  data,
  testId
}) => {
  return /*#__PURE__*/React.createElement(Flex, {
    alignItems: "center",
    testId: testId || 'basic-filter-popup-select-option--avatar'
  }, data.isGroup ? /*#__PURE__*/React.createElement(Flex, {
    alignItems: "center",
    justifyContent: "center",
    xcss: groupWrapperStyles
  }, /*#__PURE__*/React.createElement(PeopleGroupIcon, {
    color: "currentColor",
    LEGACY_size: "small",
    label: ""
  })) : /*#__PURE__*/React.createElement(Avatar, {
    appearance: data.isSquare ? 'square' : 'circle',
    src: data.avatar,
    size: "xsmall"
  }), /*#__PURE__*/React.createElement(Box, {
    xcss: [commonLabelStyles, avatarOptionLabelStyles]
  }, data.label));
};
const formatOptionLabel = data => {
  if (data.optionType === 'lozengeLabel') {
    return /*#__PURE__*/React.createElement(LozengeOptionLabel, {
      data: data
    });
  }
  if (data.optionType === 'avatarLabel') {
    return /*#__PURE__*/React.createElement(AvatarOptionLabel, {
      data: data
    });
  }
  if (data.optionType === 'iconLabel') {
    return /*#__PURE__*/React.createElement(IconOptionLabel, {
      data: data
    });
  }
  return /*#__PURE__*/React.createElement(React.Fragment, null);
};
export default formatOptionLabel;
