// @atlaskit/smart-card/dist/cjs/view/FlexibleCard/components/blocks/action-block/action-footer/ActionFooterOld.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ActionFooterOld = void 0;
var _react = _interopRequireDefault(require("react"));
var _error = _interopRequireDefault(require("@atlaskit/icon/utility/migration/error"));
var _primitives = require("@atlaskit/primitives");
var _motionWrapper = _interopRequireDefault(require("../../../common/motion-wrapper"));
var containerStyles = (0, _primitives.xcss)({
  all: 'unset',
  borderRadius: 'border.radius',
  margin: 'space.100',
  marginBottom: 'space.0',
  // eslint-disable-next-line @atlaskit/design-system/use-tokens-typography
  lineHeight: '20px',
  padding: 'space.075',
  paddingInline: 'space.100',
  width: '100%',
  ':empty': {
    display: 'none'
  }
});
var errorStyles = (0, _primitives.xcss)({
  backgroundColor: 'color.background.danger'
});
var errorContentStyles = (0, _primitives.xcss)({
  paddingInlineStart: 'space.025'
});
var titleStyles = (0, _primitives.xcss)({
  color: 'color.text.subtle',
  font: "var(--ds-font-body-UNSAFE_small, normal 400 12px/16px ui-sans-serif, -apple-system, BlinkMacSystemFont, \"Segoe UI\", Ubuntu, \"Helvetica Neue\", sans-serif)"
});
var getIcon = function getIcon(appearance) {
  switch (appearance) {
    case 'error':
      return /*#__PURE__*/_react.default.createElement(_error.default, {
        color: "var(--ds-icon-danger, #C9372C)",
        spacing: "compact",
        label: "Error",
        LEGACY_size: "small",
        testId: "action-error-icon",
        LEGACY_margin: "0 ".concat("var(--ds-space-negative-025, -2px)")
      });
    default:
      return null;
  }
};
var ActionFooterOld = exports.ActionFooterOld = function ActionFooterOld(_ref) {
  var message = _ref.message,
    testId = _ref.testId;
  if (!message) {
    return null;
  }
  return /*#__PURE__*/_react.default.createElement(_primitives.Box, {
    testId: "".concat(testId, "-footer"),
    xcss: [containerStyles, message.appearance === 'error' && errorStyles]
  }, /*#__PURE__*/_react.default.createElement(_motionWrapper.default, {
    isFadeIn: true,
    show: true,
    showTransition: true
  }, /*#__PURE__*/_react.default.createElement(_primitives.Inline, {
    alignBlock: "start",
    grow: "fill",
    space: "space.075",
    xcss: errorContentStyles
  }, message.icon || getIcon(message.appearance), /*#__PURE__*/_react.default.createElement(_primitives.Box, {
    xcss: titleStyles
  }, message.title))));
};
