// @atlaskit/smart-card/dist/cjs/view/RelatedLinksModal/views/resolved/ResolvedOld.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = require("@emotion/react");
var _analyticsNext = require("@atlaskit/analytics-next");
var _primitives = require("@atlaskit/primitives");
var _messages = require("../../../../messages");
var _relatedLinksList = _interopRequireDefault(require("../../components/related-links-list"));
/**
 * @jsxRuntime classic
 * @jsx jsx
 */

// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766

var RelatedLinksResolvedView = function RelatedLinksResolvedView(_ref) {
  var _ref$incomingLinks = _ref.incomingLinks,
    incomingLinks = _ref$incomingLinks === void 0 ? [] : _ref$incomingLinks,
    _ref$outgoingLinks = _ref.outgoingLinks,
    outgoingLinks = _ref$outgoingLinks === void 0 ? [] : _ref$outgoingLinks;
  return (0, _react.jsx)(_primitives.Stack, {
    space: "space.150"
  }, (0, _react.jsx)(_analyticsNext.AnalyticsContext, {
    data: {
      component: 'relatedLinksIncoming'
    }
  }, (0, _react.jsx)(_relatedLinksList.default, {
    urls: incomingLinks,
    title: _messages.messages.related_links_found_in,
    testId: "incoming-related-links-list"
  })), (0, _react.jsx)(_analyticsNext.AnalyticsContext, {
    data: {
      component: 'relatedLinksOutgoing'
    }
  }, (0, _react.jsx)(_relatedLinksList.default, {
    urls: outgoingLinks,
    title: _messages.messages.related_links_includes_links_to,
    testId: "outgoing-related-links-list"
  })));
};
var _default = exports.default = RelatedLinksResolvedView;
