// @atlaskit/link-datasource/dist/cjs/ui/common/modal/popup-select/dropdownIndicatorOld.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = _interopRequireDefault(require("react"));
var _crossCircle = _interopRequireDefault(require("@atlaskit/icon/core/migration/cross-circle"));
var _search = _interopRequireDefault(require("@atlaskit/icon/core/migration/search"));
var _primitives = require("@atlaskit/primitives");
var _select = require("@atlaskit/select");
var customDropdownIndicatorStyles = (0, _primitives.xcss)({
  display: 'flex',
  cursor: 'pointer',
  justifyContent: 'center',
  width: "var(--ds-space-400, 32px)"
});
var CustomDropdownIndicator = function CustomDropdownIndicator(props) {
  var selectProps = props.selectProps;
  return /*#__PURE__*/_react.default.createElement(_select.components.DropdownIndicator, props, /*#__PURE__*/_react.default.createElement(_primitives.Box, {
    xcss: customDropdownIndicatorStyles,
    onClick: function onClick() {
      if (selectProps.inputValue) {
        selectProps.onInputChange && selectProps.onInputChange('', {
          action: 'input-change',
          prevInputValue: selectProps.inputValue
        });
      }
    }
  }, selectProps.inputValue ? /*#__PURE__*/_react.default.createElement(_crossCircle.default, {
    LEGACY_size: "small",
    label: "",
    color: "currentColor"
  }) : /*#__PURE__*/_react.default.createElement(_search.default, {
    LEGACY_size: "small",
    label: "",
    color: "currentColor"
  })));
};
var _default = exports.default = CustomDropdownIndicator;
