// @atlaskit/link-datasource/dist/cjs/ui/common/modal/popup-select/footer-old.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _react = _interopRequireDefault(require("react"));
var _reactIntlNext = require("react-intl-next");
var _primitives = require("@atlaskit/primitives");
var _colors = require("@atlaskit/theme/colors");
var _messages = require("./messages");
var footerContainerStyles = (0, _primitives.xcss)({
  paddingTop: 'space.050',
  paddingBottom: 'space.050',
  borderTop: "".concat("var(--ds-border, ".concat(_colors.N40, ")"), " solid 1px")
});
var footerPaginationInfoStyles = (0, _primitives.xcss)({
  color: 'color.text.subtlest',
  marginBlock: 'space.100',
  marginInline: 'space.150'
});
var PopupFooter = function PopupFooter(_ref) {
  var currentDisplayCount = _ref.currentDisplayCount,
    totalCount = _ref.totalCount,
    filterName = _ref.filterName;
  return /*#__PURE__*/_react.default.createElement(_primitives.Flex, {
    testId: "".concat(filterName, "--footer"),
    direction: "row",
    alignItems: "center",
    justifyContent: "end",
    xcss: footerContainerStyles
  }, /*#__PURE__*/_react.default.createElement(_primitives.Inline, {
    xcss: footerPaginationInfoStyles
  }, /*#__PURE__*/_react.default.createElement(_reactIntlNext.FormattedMessage, (0, _extends2.default)({}, _messages.asyncPopupSelectMessages.paginationDetails, {
    values: {
      currentDisplayCount: currentDisplayCount,
      totalCount: totalCount
    }
  }))));
};
var _default = exports.default = PopupFooter;
