// @atlaskit/link-provider/dist/cjs/editor/transformer.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Transformer = void 0;
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var Transformer = exports.Transformer = /*#__PURE__*/function () {
  function Transformer() {
    (0, _classCallCheck2.default)(this, Transformer);
  }
  return (0, _createClass2.default)(Transformer, [{
    key: "buildInlineAdf",
    value: function buildInlineAdf(url) {
      return {
        type: 'inlineCard',
        attrs: {
          url: url
        }
      };
    }
  }, {
    key: "buildBlockAdf",
    value: function buildBlockAdf(url) {
      return {
        type: 'blockCard',
        attrs: {
          url: url
        }
      };
    }
  }, {
    key: "buildEmbedAdf",
    value: function buildEmbedAdf(url) {
      return {
        type: 'embedCard',
        attrs: {
          url: url,
          layout: 'wide'
        }
      };
    }
  }, {
    key: "toAdf",
    value: function toAdf(url, appearance) {
      switch (appearance) {
        case 'inline':
          return this.buildInlineAdf(url);
        case 'block':
          return this.buildBlockAdf(url);
        case 'embed':
          return this.buildEmbedAdf(url);
      }
    }
  }]);
}();
