// @atlaskit/link-datasource/dist/cjs/ui/confluence-search-modal/basic-filters/filters/date-range-picker/PopupComponent.js
/* PopupComponent.tsx generated by @compiled/babel-plugin v0.36.1 */
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
var _typeof = require("@babel/runtime/helpers/typeof");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.PopupComponentOld = exports.PopupComponentNew = exports.PopupComponent = void 0;
require("./PopupComponent.compiled.css");
var _runtime = require("@compiled/react/runtime");
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));
var _react = _interopRequireWildcard(require("react"));
var _platformFeatureFlags = require("@atlaskit/platform-feature-flags");
var _colors = require("@atlaskit/theme/colors");
var _styledOld = require("./styled-old");
var _excluded = ["as", "style"];
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
// eslint-disable-next-line @atlaskit/ui-styling-standard/no-styled
var PopupComponentContainer = /*#__PURE__*/(0, _react.forwardRef)(function (_ref, __cmplr) {
  var _ref$as = _ref.as,
    C = _ref$as === void 0 ? "div" : _ref$as,
    __cmpls = _ref.style,
    __cmplp = (0, _objectWithoutProperties2.default)(_ref, _excluded);
  return /*#__PURE__*/_react.default.createElement(C, (0, _extends2.default)({}, __cmplp, {
    style: __cmpls,
    ref: __cmplr,
    className: (0, _runtime.ax)(["_16jlkb7n _1o9zkb7n _i0dl1wug _1reoewfl _18m9ewfl _2rkoiti9 _vchhusvi _1e0c1ule _bfhk1j9a _16qs1sus _nt751r31 _49pcglyw _1hvw1o36", __cmplp.className])
  }));
});
if (process.env.NODE_ENV !== 'production') {
  PopupComponentContainer.displayName = 'PopupComponentContainer';
}
var PopupComponentNew = exports.PopupComponentNew = /*#__PURE__*/(0, _react.forwardRef)(function (props, ref) {
  return /*#__PURE__*/_react.default.createElement(PopupComponentContainer, (0, _extends2.default)({}, props, {
    "data-testId": 'confluence-search-datasource-popup-container',
    ref: ref
  }));
});
var PopupComponentOld = exports.PopupComponentOld = /*#__PURE__*/(0, _react.forwardRef)(function (props, ref) {
  return /*#__PURE__*/_react.default.createElement(_styledOld.PopupComponentContainerOld, (0, _extends2.default)({}, props, {
    "data-testId": 'confluence-search-datasource-popup-container',
    ref: ref
  }));
});
var PopupComponent = exports.PopupComponent = /*#__PURE__*/(0, _react.forwardRef)(function (props, ref) {
  if ((0, _platformFeatureFlags.fg)('bandicoots-compiled-migration-link-datasource')) {
    return /*#__PURE__*/_react.default.createElement(PopupComponentNew, (0, _extends2.default)({}, props, {
      ref: ref
    }));
  } else {
    return /*#__PURE__*/_react.default.createElement(PopupComponentOld, (0, _extends2.default)({}, props, {
      ref: ref
    }));
  }
});
