// @atlaskit/link-datasource/dist/cjs/ui/confluence-search-modal/basic-filters/filters/date-range-picker/styled-old.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SelectDateRangeButtonOld = exports.PopupComponentContainerOld = exports.DateRangeErrorMessageOld = exports.DatePickersWrapperOld = exports.CustomDropdownOld = exports.CustomDropdownItemOld = exports.CustomDateWrapperOld = void 0;
var _styled = _interopRequireDefault(require("@emotion/styled"));
var _colors = require("@atlaskit/theme/colors");
var _constants = require("@atlaskit/theme/constants");
// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766

// eslint-disable-next-line @atlaskit/ui-styling-standard/no-styled, @atlaskit/ui-styling-standard/no-exported-styles -- Ignored via go/DSP-18766
var PopupComponentContainerOld = exports.PopupComponentContainerOld = _styled.default.div({
  boxSizing: 'border-box',
  display: 'block',
  flex: '1 1 auto',
  overflow: 'visible',
  borderRadius: "var(--ds-border-radius-100, 4px)",
  background: "var(--ds-background-input, ".concat(_colors.N0, ")"),
  // eslint-disable-next-line @atlaskit/ui-styling-standard/no-unsafe-selectors -- Ignored via go/DSP-18766
  ':focus': {
    outline: 'none'
  },
  boxShadow: "var(--ds-shadow-overlay, 0px 0px 1px 0px rgba(9, 30, 66, 0.31), 0px 3px 5px 0px rgba(9, 30, 66, 0.20))"
});

// eslint-disable-next-line @atlaskit/ui-styling-standard/no-styled, @atlaskit/ui-styling-standard/no-exported-styles -- Ignored via go/DSP-18766
var CustomDropdownOld = exports.CustomDropdownOld = _styled.default.div({
  width: '340px',
  background: "var(--ds-background-input, ".concat(_colors.N0, ")"),
  borderRadius: "var(--ds-border-radius-100, 4px)",
  boxShadow: "var(--ds-shadow-overlay, 0px 0px 1px 0px rgba(9, 30, 66, 0.31), 0px 3px 5px 0px rgba(9, 30, 66, 0.20))",
  zIndex: _constants.layers.modal()
});

// eslint-disable-next-line @atlaskit/ui-styling-standard/no-styled, @atlaskit/ui-styling-standard/no-exported-styles, @atlaskit/ui-styling-standard/no-dynamic-styles -- Ignored via go/DSP-18766
var CustomDropdownItemOld = exports.CustomDropdownItemOld = _styled.default.div(function (props) {
  return {
    height: '36px',
    width: '100%',
    display: 'flex',
    justifyContent: 'flex-start',
    alignItems: 'center',
    // eslint-disable-next-line @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
    paddingLeft: props.isSelected ? '15px' : '16px',
    boxSizing: 'border-box',
    // eslint-disable-next-line @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
    color: props.isSelected ? "var(--ds-text-selected, ".concat(_colors.B400, ")") : 'inherit',
    // eslint-disable-next-line @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
    background: props.isSelected ? "var(--ds-background-accent-blue-subtlest, ".concat(_colors.B50, ")") : 'transparent',
    // eslint-disable-next-line @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
    borderLeft: props.isSelected ? "2px solid ".concat("var(--ds-text-selected, ".concat(_colors.B400, ")")) : 'none',
    '&:hover': {
      // eslint-disable-next-line @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
      background: props.isSelected ? "var(--ds-background-accent-blue-subtlest, ".concat(_colors.B50, ")") : "var(--ds-background-input-hovered, ".concat(_colors.N20, ")"),
      cursor: 'default'
    }
  };
});

// eslint-disable-next-line @atlaskit/ui-styling-standard/no-styled, @atlaskit/ui-styling-standard/no-exported-styles -- Ignored via go/DSP-18766
var CustomDateWrapperOld = exports.CustomDateWrapperOld = _styled.default.div({
  padding: "var(--ds-space-150, 12px)",
  boxSizing: 'border-box'
});

// eslint-disable-next-line @atlaskit/ui-styling-standard/no-styled, @atlaskit/ui-styling-standard/no-exported-styles -- Ignored via go/DSP-18766
var DatePickersWrapperOld = exports.DatePickersWrapperOld = _styled.default.div({
  display: 'flex',
  justifyContent: 'space-between',
  alignItems: 'center',
  width: '100%'
});

// eslint-disable-next-line @atlaskit/ui-styling-standard/no-styled, @atlaskit/ui-styling-standard/no-exported-styles -- Ignored via go/DSP-18766
var SelectDateRangeButtonOld = exports.SelectDateRangeButtonOld = _styled.default.button({
  background: "var(--ds-background-accent-gray-subtler, ".concat(_colors.N20, ")"),
  border: 'none',
  font: 'inherit',
  width: '70px',
  height: '40px',
  marginTop: "var(--ds-space-150, 12px)",
  borderRadius: "var(--ds-border-radius-100, 4px)",
  '&:hover': {
    background: "var(--ds-background-accent-gray-subtler, ".concat(_colors.N30, ")"),
    cursor: 'pointer'
  }
});

// eslint-disable-next-line @atlaskit/ui-styling-standard/no-styled, @atlaskit/ui-styling-standard/no-exported-styles -- Ignored via go/DSP-18766
var DateRangeErrorMessageOld = exports.DateRangeErrorMessageOld = _styled.default.div({
  display: 'flex',
  marginTop: "var(--ds-space-050, 2px)",
  gap: "var(--ds-space-025, 2px)",
  paddingInlineStart: "var(--ds-space-025, 2px)",
  font: "var(--ds-font-body-UNSAFE_small, normal 400 12px/16px ui-sans-serif, -apple-system, BlinkMacSystemFont, \"Segoe UI\", Ubuntu, \"Helvetica Neue\", sans-serif)",
  color: "var(--ds-text-danger, ".concat(_colors.R400, ")"),
  alignItems: 'center'
});
