// @atlaskit/link-datasource/dist/cjs/ui/confluence-search-modal/basic-filters/filters/date-range-picker/trigger.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.PopupTrigger = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _react = _interopRequireDefault(require("react"));
var _standardButton = _interopRequireDefault(require("@atlaskit/button/standard-button"));
var _chevronDown = _interopRequireDefault(require("@atlaskit/icon/utility/migration/chevron-down"));
var PopupTrigger = exports.PopupTrigger = function PopupTrigger(_ref) {
  var triggerProps = _ref.triggerProps,
    isSelected = _ref.isSelected,
    labelPrefix = _ref.labelPrefix,
    selectedLabel = _ref.selectedLabel,
    onClick = _ref.onClick;
  return /*#__PURE__*/_react.default.createElement(_standardButton.default, (0, _extends2.default)({}, triggerProps, {
    testId: 'confluence-search-modal--date-range-button',
    onClick: onClick,
    isSelected: isSelected,
    iconAfter:
    /*#__PURE__*/
    // eslint-disable-next-line @atlaskit/ui-styling-standard/enforce-style-prop -- Ignored via go/DSP-18766
    _react.default.createElement("span", {
      style: {
        display: 'flex',
        alignItems: 'center'
      }
    }, /*#__PURE__*/_react.default.createElement(_chevronDown.default, {
      LEGACY_size: "medium",
      label: "",
      color: "currentColor"
    }))
  }), labelPrefix, selectedLabel ? ": ".concat(selectedLabel) : '');
};
