// @atlaskit/smart-card/dist/cjs/view/FlexibleCard/components/actions/automation-action/automation-manual-triggers/manual-triggers-modal/sub-components/header/AutomationModalHeaderOld.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.AutomationModalHeaderOld = void 0;
var _react = _interopRequireDefault(require("react"));
var _reactIntlNext = require("react-intl-next");
var _heading = _interopRequireDefault(require("@atlaskit/heading"));
var _manualTriggerIcon = _interopRequireDefault(require("@atlaskit/legacy-custom-icons/manual-trigger-icon"));
var _modalDialog = require("@atlaskit/modal-dialog");
var _primitives = require("@atlaskit/primitives");
var _colors = require("@atlaskit/theme/colors");
var _menuContext = require("../../menu-context");
var i18n = (0, _reactIntlNext.defineMessages)({
  modalHeaderIconLabel: {
    id: 'automation-menu.modal.header.icon.label',
    defaultMessage: 'Automation modal header icon',
    description: 'A label for the icon in the header of the automation modal.'
  }
});
var iconStyle = (0, _primitives.xcss)({
  marginRight: 'space.150',
  backgroundColor: 'color.background.accent.green.subtlest',
  borderRadius: 'border.radius'
});
var modalDescriptionStyle = (0, _primitives.xcss)({
  marginTop: 'space.150'
});
var AutomationModalHeaderOld = exports.AutomationModalHeaderOld = function AutomationModalHeaderOld(_ref) {
  var modalTitle = _ref.modalTitle,
    modalDescription = _ref.modalDescription;
  var _useIntl = (0, _reactIntlNext.useIntl)(),
    formatMessage = _useIntl.formatMessage;
  var _useAutomationMenu = (0, _menuContext.useAutomationMenu)(),
    initialised = _useAutomationMenu.initialised,
    rules = _useAutomationMenu.rules;
  var _useModal = (0, _modalDialog.useModal)(),
    titleId = _useModal.titleId;
  var showDescription = initialised && rules.length > 0 && !!modalDescription;
  return /*#__PURE__*/_react.default.createElement(_modalDialog.ModalHeader, null, /*#__PURE__*/_react.default.createElement(_primitives.Stack, null, /*#__PURE__*/_react.default.createElement(_primitives.Inline, {
    alignBlock: "center"
  }, /*#__PURE__*/_react.default.createElement(_primitives.Box, {
    xcss: iconStyle,
    paddingInline: "space.050",
    paddingBlock: "space.025"
  }, /*#__PURE__*/_react.default.createElement(_manualTriggerIcon.default, {
    size: "small",
    label: formatMessage(i18n.modalHeaderIconLabel),
    primaryColor: "var(--ds-icon-accent-green, ".concat(_colors.G50, ")")
  })), /*#__PURE__*/_react.default.createElement(_heading.default, {
    size: "medium",
    id: titleId
  }, modalTitle)), showDescription && /*#__PURE__*/_react.default.createElement(_primitives.Box, {
    xcss: modalDescriptionStyle
  }, modalDescription)));
};
