// @atlaskit/smart-card/dist/cjs/view/FlexibleCard/components/blocks/title-block/errored/TitleBlockErroredViewOld.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));
var _react = require("@emotion/react");
var _reactIntlNext = require("react-intl-next");
var _primitives = require("@atlaskit/primitives");
var _constants = require("../../../../../../constants");
var _elements = require("../../../elements");
var _utils = require("../../../utils");
var _block = _interopRequireDefault(require("../../block"));
var _elementGroup = _interopRequireDefault(require("../../element-group"));
var _excluded = ["actionGroup", "retry", "position", "testId", "title", "icon", "hideIcon"];
/**
 * @jsxRuntime classic
 * @jsx jsx
 */
// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766
var actionStyles = (0, _react.css)({
  cursor: 'pointer',
  // eslint-disable-next-line @atlaskit/ui-styling-standard/no-unsafe-selectors -- Ignored via go/DSP-18766
  ':hover': {
    color: "var(--ds-text-subtle, #8993A4)",
    textDecoration: 'underline'
  }
});
var actionHoverStyles = (0, _react.css)({
  position: 'relative',
  zIndex: 1,
  '&:focus': {
    outlineOffset: "var(--ds-space-negative-025, -2px)"
  }
});

/**
 * This function does not work in the component below as it needs to be xcss.
 * The component instead uses inherited styles
 */
var getMessageStyles = function getMessageStyles(size, hasAction) {
  var sizeStyles = (0, _utils.getLinkSizeStyles)(size);
  return (0, _react.css)({
    flex: '1 1 auto',
    justifyContent: 'flex-end'
  },
  // eslint-disable-next-line @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
  hasAction ? actionStyles : '',
  // eslint-disable-next-line @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
  sizeStyles,
  // eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values, @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
  (0, _utils.getTruncateStyles)(1, (0, _utils.getLinkLineHeight)(size)), {
    color: "var(--ds-text-disabled, #6B778C)",
    // eslint-disable-next-line @atlaskit/ui-styling-standard/no-unsafe-selectors -- Ignored via go/DSP-18766
    ':focus': {
      outline: "var(--ds-border-focused, #388BFF)".concat(" solid 2px")
    }
  },
  // eslint-disable-next-line @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
  hasAction ? actionHoverStyles : '');
};

/**
 * Represents an Errored TitleBlock view.
 * This will render when a Smart Link did not successfully resolve.
 * This may be a result of a Smart Link not having the correct credentials,
 * or the backend response was errored or malformed.
 * @see TitleBlock
 */
var TitleBlockErroredViewOld = function TitleBlockErroredViewOld(_ref) {
  var actionGroup = _ref.actionGroup,
    retry = _ref.retry,
    position = _ref.position,
    testId = _ref.testId,
    title = _ref.title,
    icon = _ref.icon,
    hideIcon = _ref.hideIcon,
    blockProps = (0, _objectWithoutProperties2.default)(_ref, _excluded);
  var _ref2 = retry || {},
    descriptor = _ref2.descriptor,
    onClick = _ref2.onClick,
    values = _ref2.values;
  var _blockProps$size = blockProps.size,
    size = _blockProps$size === void 0 ? _constants.SmartLinkSize.Medium : _blockProps$size;
  var hasAction = onClick !== undefined;
  return (0, _react.jsx)(_block.default, (0, _extends2.default)({}, blockProps, {
    testId: "".concat(testId, "-errored-view")
  }), !hideIcon && (0, _react.jsx)(_elements.LinkIcon, {
    overrideIcon: icon,
    position: position
  }), title, descriptor && (0, _react.jsx)(_elementGroup.default, {
    direction: _constants.SmartLinkDirection.Horizontal,
    align: _constants.SmartLinkAlignment.Right
  }, (0, _react.jsx)(_primitives.Box
  // eslint-disable-next-line @atlaskit/design-system/consistent-css-prop-usage -- Ignored via go/DSP-18766
  , {
    xcss: getMessageStyles(size, hasAction),
    onClick: onClick,
    testId: "".concat(testId, "-errored-view-message"),
    tabIndex: hasAction ? 0 : -1
  }, (0, _react.jsx)(_reactIntlNext.FormattedMessage, (0, _extends2.default)({}, descriptor, {
    values: values
  })))), actionGroup);
};
var _default = exports.default = TitleBlockErroredViewOld;
