// @atlaskit/link-datasource/dist/cjs/ui/issue-like-table/drag-column-preview-old.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.DragColumnPreviewOld = void 0;
var _react = _interopRequireDefault(require("react"));
var _react2 = require("@emotion/react");
var _styled = _interopRequireDefault(require("@emotion/styled"));
var _colors = require("@atlaskit/theme/colors");
var _typography = require("@atlaskit/theme/typography");
var _styled2 = require("./styled");
/**
 * @jsxRuntime classic
 * @jsx jsx
 */

// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766

// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766

var maxWidth = '200px';
var firstLastChildOverride = "\n  &:first-of-type, &:last-of-type {\n    padding-left: ".concat("var(--ds-space-100, 8px)", ";\n    padding-right: ", "var(--ds-space-100, 8px)", ";\n  }\n");
// eslint-disable-next-line @atlaskit/ui-styling-standard/no-styled -- To migrate as part of go/ui-styling-standard
var TablePreviewHeading = (0, _styled.default)(_styled2.TableHeading)({
  overflow: 'hidden',
  textOverflow: 'ellipsis',
  whiteSpace: 'nowrap',
  background: "var(--ds-background-disabled, #091E4224)",
  maxWidth: maxWidth
},
// eslint-disable-next-line @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
firstLastChildOverride);

// eslint-disable-next-line @atlaskit/ui-styling-standard/no-styled -- To migrate as part of go/ui-styling-standard
var TablePreviewCell = _styled.default.td({
  overflow: 'hidden',
  textOverflow: 'ellipsis',
  whiteSpace: 'nowrap',
  maxWidth: maxWidth,
  padding: "var(--ds-space-050, 4px)".concat(" ", "var(--ds-space-100, 8px)"),
  boxSizing: 'border-box',
  borderBottom: "0.5px solid ".concat("var(--ds-border, ".concat(_colors.N40, ")")),
  height: '30px'
},
// eslint-disable-next-line @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
firstLastChildOverride);
var tableStyles = (0, _react2.css)({
  background: "var(--ds-surface, #FFF)",
  borderCollapse: 'separate',
  borderSpacing: 0,
  // eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values -- Ignored via go/DSP-18766
  font: "var(--ds-font-body, ".concat(_typography.fontFallback.body.medium, ")"),
  maxWidth: maxWidth
});
var DragColumnPreviewOld = exports.DragColumnPreviewOld = function DragColumnPreviewOld(_ref) {
  var title = _ref.title,
    rows = _ref.rows;
  return (0, _react2.jsx)("table", {
    css: tableStyles
  }, (0, _react2.jsx)("thead", null, (0, _react2.jsx)("tr", null, (0, _react2.jsx)(TablePreviewHeading, null, title))), (0, _react2.jsx)("tbody", null, rows.map(function (data, i) {
    return (0, _react2.jsx)("tr", {
      key: i
    }, (0, _react2.jsx)(TablePreviewCell, null, data));
  }), (0, _react2.jsx)("tr", null, (0, _react2.jsx)(TablePreviewCell, null, "..."))));
};
