// @atlaskit/smart-card/dist/cjs/view/FlexibleCard/components/elements/lozenge/LozengeOld.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = require("@emotion/react");
var _lozenge = _interopRequireDefault(require("@atlaskit/lozenge"));
var _flexibleUiContext = require("../../../../../state/flexible-ui-context");
var _lozengeAction = _interopRequireDefault(require("./lozenge-action"));
/**
 * @jsxRuntime classic
 * @jsx jsx
 */
// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766

var styles = (0, _react.css)({
  display: 'inline-flex',
  // eslint-disable-next-line @atlaskit/design-system/use-tokens-typography
  lineHeight: 'inherit'
});

/**
 * A base element that displays a Lozenge.
 * @internal
 * @param {LozengeProps} LozengeProps - The props necessary for the Lozenge element.
 * @see State
 */
var LozengeOld = function LozengeOld(_ref) {
  var action = _ref.action,
    _ref$appearance = _ref.appearance,
    appearance = _ref$appearance === void 0 ? 'default' : _ref$appearance,
    name = _ref.name,
    overrideCss = _ref.overrideCss,
    style = _ref.style,
    text = _ref.text,
    _ref$testId = _ref.testId,
    testId = _ref$testId === void 0 ? 'smart-element-lozenge' : _ref$testId;
  var ui = (0, _flexibleUiContext.useFlexibleUiOptionContext)();
  if (!text) {
    return null;
  }
  var lozenge = action ? (0, _react.jsx)(_lozengeAction.default, {
    action: action,
    appearance: appearance,
    testId: testId,
    text: text,
    zIndex: ui === null || ui === void 0 ? void 0 : ui.zIndex
  }) : (0, _react.jsx)(_lozenge.default, {
    appearance: appearance
    // eslint-disable-next-line @atlaskit/ui-styling-standard/enforce-style-prop -- Ignored via go/DSP-18766
    ,
    style: style,
    testId: "".concat(testId, "-lozenge")
  }, text);
  return (0, _react.jsx)("span", {
    // eslint-disable-next-line @atlaskit/design-system/consistent-css-prop-usage -- Ignored via go/DSP-18766
    css: [styles, overrideCss],
    "data-fit-to-content": true,
    "data-smart-element": name,
    "data-smart-element-lozenge": true,
    "data-testid": testId
  }, lozenge);
};
var _default = exports.default = LozengeOld;
