// @atlaskit/link-datasource/dist/esm/ui/common/modal/count-view-smart-link/count-view-smart-link-old.js
import React from 'react';
import { FormattedMessage } from 'react-intl-next';
import { Box, xcss } from '@atlaskit/primitives';
import LinkRenderType from '../../../issue-like-table/render-type/link';
var placeholderSmartLinkStyles = xcss({
  backgroundColor: 'elevation.surface.raised',
  borderRadius: "var(--ds-border-radius-200, 3px)",
  color: 'color.text.brand',
  paddingTop: 'space.0',
  paddingBottom: 'space.0',
  paddingLeft: 'space.025',
  paddingRight: 'space.025',
  boxShadow: 'elevation.shadow.raised'
});
var smartLinkContainerStyles = xcss({
  paddingLeft: 'space.025'
});
export var SmartCardPlaceholderOld = function SmartCardPlaceholderOld(_ref) {
  var placeholderText = _ref.placeholderText;
  return /*#__PURE__*/React.createElement(Box, {
    xcss: smartLinkContainerStyles
  }, /*#__PURE__*/React.createElement(Box, {
    as: "span",
    testId: "datasource-modal--smart-card-placeholder",
    xcss: placeholderSmartLinkStyles
  }, /*#__PURE__*/React.createElement(FormattedMessage, placeholderText)));
};
export var SmartLinkOld = function SmartLinkOld(_ref2) {
  var url = _ref2.url;
  return /*#__PURE__*/React.createElement(Box, {
    xcss: smartLinkContainerStyles
  }, /*#__PURE__*/React.createElement(LinkRenderType, {
    url: url
  }));
};
