// @atlaskit/link-picker/dist/cjs/ui/link-picker/search-results/scrolling-tabs/old/styles.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.scrollingContainerStyles = exports.nextButtonStyles = exports.containerStyles = exports.buttonContainerStyles = exports.backButtonStyles = void 0;
var _react = require("@emotion/react");
var _colors = require("@atlaskit/theme/colors");
// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766

// eslint-disable-next-line @atlaskit/ui-styling-standard/no-exported-styles -- Ignored via go/DSP-18766
var scrollingContainerStyles = exports.scrollingContainerStyles = (0, _react.css)({
  overflowX: 'auto',
  scrollBehavior: 'smooth',
  scrollPadding: '0 24px',
  scrollbarWidth: 'none',
  // eslint-disable-next-line @atlaskit/ui-styling-standard/no-unsafe-selectors -- Ignored via go/DSP-18766
  '&::-webkit-scrollbar': {
    display: 'none'
  },
  '&::before': {
    content: '""',
    borderRadius: 2,
    bottom: 0,
    margin: 0,
    position: 'absolute',
    width: 'inherit',
    left: "var(--ds-space-100, 8px)",
    right: "var(--ds-space-100, 8px)",
    height: 2,
    backgroundColor: "var(--ds-border, ".concat(_colors.N30, ")")
  }
});

// eslint-disable-next-line @atlaskit/ui-styling-standard/no-exported-styles -- Ignored via go/DSP-18766
var containerStyles = exports.containerStyles = (0, _react.css)({
  position: 'relative',
  // eslint-disable-next-line @atlaskit/ui-styling-standard/no-nested-selectors -- Ignored via go/DSP-18766
  '[role="tablist"]': {
    '&::before': {
      display: 'none'
    }
  },
  // Overrides Atlaskit tabs styles to stop overflowing with ellipsis
  // eslint-disable-next-line @atlaskit/ui-styling-standard/no-nested-selectors -- Ignored via go/DSP-18766
  '[role="tab"]': {
    // eslint-disable-next-line @atlaskit/ui-styling-standard/no-important-styles -- Ignored via go/DSP-18766
    overflow: 'unset !important',
    // eslint-disable-next-line @atlaskit/ui-styling-standard/no-important-styles -- Ignored via go/DSP-18766
    textOverflow: 'unset !important'
  }
});

// eslint-disable-next-line @atlaskit/ui-styling-standard/no-exported-styles -- Ignored via go/DSP-18766
var buttonContainerStyles = exports.buttonContainerStyles = (0, _react.css)({
  position: 'absolute',
  top: "var(--ds-space-050, 4px)",
  zIndex: 999,
  backgroundColor: "var(--ds-surface, ".concat(_colors.N0, ")")
});

// eslint-disable-next-line @atlaskit/ui-styling-standard/no-array-arguments, @atlaskit/ui-styling-standard/no-exported-styles -- Ignored via go/DSP-18766
var backButtonStyles = exports.backButtonStyles = (0, _react.css)([
// eslint-disable-next-line @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
buttonContainerStyles, {
  left: 0
}]);

// eslint-disable-next-line @atlaskit/ui-styling-standard/no-array-arguments, @atlaskit/ui-styling-standard/no-exported-styles -- Ignored via go/DSP-18766
var nextButtonStyles = exports.nextButtonStyles = (0, _react.css)([
// eslint-disable-next-line @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
buttonContainerStyles, {
  right: 0
}]);
