// @atlaskit/link-datasource/dist/cjs/ui/common/modal/search-count/search-count-old.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.AssetsItemCountOld = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _react = _interopRequireDefault(require("react"));
var _reactIntlNext = require("react-intl-next");
var _heading = _interopRequireDefault(require("@atlaskit/heading"));
var _platformFeatureFlags = require("@atlaskit/platform-feature-flags");
var _primitives = require("@atlaskit/primitives");
var _linkUrl = _interopRequireDefault(require("@atlaskit/smart-card/link-url"));
var _colors = require("@atlaskit/theme/colors");
var _messages = require("../../../table-footer/messages");
var _messages2 = require("./messages");
var searchCountStyles = (0, _primitives.xcss)({
  flex: 1
});
var ItemCountWrapper = function ItemCountWrapper(_ref) {
  var url = _ref.url,
    children = _ref.children,
    testId = _ref.testId;
  return /*#__PURE__*/_react.default.createElement(_primitives.Flex, {
    testId: testId,
    xcss: searchCountStyles,
    alignItems: "center"
  }, /*#__PURE__*/_react.default.createElement(_linkUrl.default, {
    href: url,
    target: "_blank",
    testId: "item-count-url"
    // eslint-disable-next-line @atlaskit/ui-styling-standard/enforce-style-prop -- Ignored via go/DSP-18766
    ,
    style: {
      color: "var(--ds-text-accent-gray, ".concat(_colors.N800, ")"),
      textDecoration: !url ? 'none' : ''
    }
  }, /*#__PURE__*/_react.default.createElement(_heading.default, {
    size: "xxsmall"
  }, children)));
};
var AssetsItemCountOld = exports.AssetsItemCountOld = function AssetsItemCountOld(_ref2) {
  var searchCount = _ref2.searchCount,
    url = _ref2.url,
    testId = _ref2.testId;
  var capCount = searchCount >= 1000;
  var displayCount = capCount ? 1000 : searchCount;
  var countModifier = capCount ? '+' : '';
  return /*#__PURE__*/_react.default.createElement(ItemCountWrapper, {
    testId: testId,
    url: url
  }, /*#__PURE__*/_react.default.createElement(_reactIntlNext.FormattedNumber, {
    value: displayCount
  }), countModifier, ' ', /*#__PURE__*/_react.default.createElement(_reactIntlNext.FormattedMessage, (0, _extends2.default)({}, _messages.footerMessages.itemText, {
    values: {
      itemCount: searchCount
    }
  })));
};
var TableSearchCount = function TableSearchCount(_ref3) {
  var url = _ref3.url,
    searchCount = _ref3.searchCount,
    _ref3$testId = _ref3.testId,
    testId = _ref3$testId === void 0 ? 'datasource-table-total-results-count' : _ref3$testId,
    _ref3$prefixTextType = _ref3.prefixTextType,
    prefixTextType = _ref3$prefixTextType === void 0 ? 'issue' : _ref3$prefixTextType;
  var isIssue = prefixTextType === 'issue';
  var messageKey = isIssue && (0, _platformFeatureFlags.fg)('confluence-issue-terminology-refresh') ? 'issueCountTextIssueTermRefresh' : "".concat(prefixTextType, "CountText");
  return /*#__PURE__*/_react.default.createElement(ItemCountWrapper, {
    testId: testId,
    url: url
  }, /*#__PURE__*/_react.default.createElement(_reactIntlNext.FormattedMessage, (0, _extends2.default)({}, _messages2.searchCountMessages[messageKey], {
    values: {
      searchCount: searchCount
    }
  })));
};
var _default = exports.default = TableSearchCount;
