// @atlaskit/smart-card/dist/cjs/view/common/utils.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.mq = exports.handleClickCommon = exports.gs = exports.br = void 0;
var _facepaint = _interopRequireDefault(require("facepaint"));
// this function is used for spacing, fontsize and line height,
// once design tokens have been established for the all of these categories
// then this function can be replaced with the tokens
/**
 * @deprecated, these can be replaced with constants
 */
var gs = exports.gs = function gs(times) {
  return "".concat(8 * times, "px");
};
/**
 * @deprecated, these can be replaced with constants
 */
var br = exports.br = function br() {
  var times = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : 1;
  return "".concat(3 * times, "px");
};
/**
 * @deprecated, these can be replaced with constants
 */
var mq = exports.mq = (0, _facepaint.default)(['@media(min-width: 576px)']);
var handleClickCommon = exports.handleClickCommon = function handleClickCommon(event, onClick) {
  if (onClick) {
    event.preventDefault();
    event.stopPropagation();
    onClick(event);
  }
};
