// @atlaskit/smart-card/dist/cjs/view/LinkUrl/LinkWarningModal/WarningModalOld.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
var _typeof = require("@babel/runtime/helpers/typeof");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _react = require("@emotion/react");
var _reactIntlNext = require("react-intl-next");
var _standardButton = _interopRequireDefault(require("@atlaskit/button/standard-button"));
var _modalDialog = _interopRequireWildcard(require("@atlaskit/modal-dialog"));
var _messages = require("../../../messages");
var _styled = require("./styled");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
/**
 * @jsxRuntime classic
 * @jsx jsx
 */
// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766

var WarningModalOld = function WarningModalOld(props) {
  var isOpen = props.isOpen,
    unsafeLinkText = props.unsafeLinkText,
    url = props.url,
    onClose = props.onClose,
    onContinue = props.onContinue,
    intl = props.intl;
  var content = (0, _react.jsx)(_modalDialog.ModalTransition, null, isOpen && (0, _react.jsx)(_modalDialog.default, {
    onClose: onClose,
    testId: "link-with-safety-warning"
  }, (0, _react.jsx)(_modalDialog.ModalHeader, null, (0, _react.jsx)(_modalDialog.ModalTitle, {
    appearance: "warning"
  }, (0, _react.jsx)(_reactIntlNext.FormattedMessage, _messages.messages.check_this_link))), (0, _react.jsx)(_modalDialog.ModalBody, null, (0, _react.jsx)("div", {
    css: _styled.breakWordsCss
  }, url && unsafeLinkText && (0, _react.jsx)(_reactIntlNext.FormattedMessage, (0, _extends2.default)({}, _messages.messages.link_safety_warning_message, {
    values: {
      unsafeLinkText: unsafeLinkText,
      a: function a() {
        return (0, _react.jsx)("a", {
          href: url,
          target: "_blank",
          rel: "noopener noreferrer"
        }, url);
      }
    }
  })))), (0, _react.jsx)(_modalDialog.ModalFooter, null, (0, _react.jsx)(_standardButton.default, {
    appearance: "subtle",
    onClick: onClose
  }, (0, _react.jsx)(_reactIntlNext.FormattedMessage, _messages.messages.go_back)), (0, _react.jsx)(_standardButton.default, {
    appearance: "warning",
    onClick: onContinue,
    autoFocus: true
  }, (0, _react.jsx)(_reactIntlNext.FormattedMessage, _messages.messages.continue)))));
  return intl ? content : (0, _react.jsx)(_reactIntlNext.IntlProvider, {
    locale: "en"
  }, content);
};
var _default = exports.default = WarningModalOld;
