// @atlaskit/smart-card/dist/cjs/view/common/ai-prism/ai-glowing-border/AIGlowingBorderOld.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.borderContentStyles = void 0;
var _react = require("@emotion/react");
var _constants = require("../constants");
var _animatedSvgContainer = _interopRequireDefault(require("./animated-svg-container"));
/**
 * @jsxRuntime classic
 * @jsx jsx
 */
// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766

var borderContainerStyles = (0, _react.css)({
  display: 'flex',
  position: 'relative',
  padding: "var(--ds-space-025, 2px)",
  width: 'fit-content'
});

// eslint-disable-next-line @atlaskit/ui-styling-standard/no-exported-styles -- Ignored via go/DSP-18766
var borderContentStyles = exports.borderContentStyles = (0, _react.css)({
  // eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values, @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
  borderRadius: _constants.INNER_BORDER_RADIUS,
  flexGrow: 1,
  zIndex: 1 // needs to be more than the svg container at least
});

/**
 * The bulk of this file is originally from
 * https://bitbucket.org/atlassian/barrel/src/master/ui/platform/ui-kit/ai
 * with modifications.
 */
var AIGlowingBorderOld = function AIGlowingBorderOld(_ref) {
  var children = _ref.children,
    palette = _ref.palette,
    _ref$isMoving = _ref.isMoving,
    isMoving = _ref$isMoving === void 0 ? true : _ref$isMoving,
    isGlowing = _ref.isGlowing,
    testId = _ref.testId,
    additionalCss = _ref.additionalCss;
  return (0, _react.jsx)("div", {
    css: [borderContainerStyles, additionalCss === null || additionalCss === void 0 ? void 0 : additionalCss.container],
    "data-testid": testId
  }, (0, _react.jsx)(_animatedSvgContainer.default, {
    palette: palette,
    isMoving: isMoving,
    additionalCss: additionalCss === null || additionalCss === void 0 ? void 0 : additionalCss.animatedSvgContainer
  }), isGlowing && (0, _react.jsx)(_animatedSvgContainer.default, {
    palette: palette,
    isMoving: isMoving,
    isGlowing: true,
    additionalCss: additionalCss === null || additionalCss === void 0 ? void 0 : additionalCss.animatedSvgContainer
  }), (0, _react.jsx)("div", {
    css: borderContentStyles
  }, children, " "));
};
var _default = exports.default = AIGlowingBorderOld;
