// @atlaskit/smart-card/dist/cjs/view/EmbedModal/components/embed-content/indexOld.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = require("@emotion/react");
var _reactMagneticDi = require("react-magnetic-di");
var _utils = require("../../../../utils");
var _IFrame = require("../../../EmbedCard/components/IFrame");
/**
 * @jsxRuntime classic
 * @jsx jsx
 */
// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766

var iframeCss = (0, _react.css)({
  width: '100%',
  height: 'calc(100vh - 208px)'
});
var EmbedContent = function EmbedContent(_ref) {
  var isTrusted = _ref.isTrusted,
    name = _ref.name,
    src = _ref.src,
    testId = _ref.testId;
  var sandbox = (0, _utils.getIframeSandboxAttribute)(isTrusted);
  var props = {
    css: iframeCss,
    frameBorder: 0,
    name: name,
    sandbox: sandbox,
    src: src,
    'data-testid': "".concat(testId, "-embed")
  };
  return (0, _react.jsx)(_IFrame.IFrame, props);
};
var _default = exports.default = EmbedContent;
