// @atlaskit/smart-card/dist/cjs/view/HoverCard/components/views/unauthorised/styled.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.titleBlockStyles = exports.mainTextStyles = exports.connectButtonStyles = void 0;
var _react = require("@emotion/react");
// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766

var blockGap = '0.5rem';
var iconGap = '0.5rem';

/**
 * @deprecated remove on FF clean up bandicoots-compiled-migration-smartcard
 */
// eslint-disable-next-line @atlaskit/ui-styling-standard/no-exported-styles -- Ignored via go/DSP-18766
var connectButtonStyles = exports.connectButtonStyles = (0, _react.css)({
  justifyContent: 'flex-end',
  marginTop: blockGap
});

/**
 * @deprecated remove on FF clean up bandicoots-compiled-migration-smartcard
 */
// eslint-disable-next-line @atlaskit/ui-styling-standard/no-exported-styles -- Ignored via go/DSP-18766
var titleBlockStyles = exports.titleBlockStyles = (0, _react.css)({
  gap: "".concat(blockGap, " ").concat(iconGap),
  // eslint-disable-next-line @atlaskit/ui-styling-standard/no-nested-selectors -- Ignored via go/DSP-18766
  a: {
    fontWeight: "var(--ds-font-weight-medium, 500)"
  }
});

/**
 * @deprecated remove on FF clean up bandicoots-compiled-migration-smartcard
 */
// eslint-disable-next-line @atlaskit/ui-styling-standard/no-exported-styles -- Ignored via go/DSP-18766
var mainTextStyles = exports.mainTextStyles = (0, _react.css)({
  marginTop: blockGap,
  font: "var(--ds-font-body-UNSAFE_small, normal 400 12px/16px ui-sans-serif, -apple-system, BlinkMacSystemFont, \"Segoe UI\", Ubuntu, \"Helvetica Neue\", sans-serif)"
});
