// @atlaskit/progress-tracker/dist/cjs/internal/link-new.js
/* link-new.tsx generated by @compiled/babel-plugin v0.36.1 */
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
require("./link-new.compiled.css");
var _runtime = require("@compiled/react/runtime");
var _react = _interopRequireDefault(require("react"));
var _compiled = require("@atlaskit/primitives/compiled");
var styles = {
  anchor: "_4bfu1r31 _1hmsglyw _ajmmnqa1 _11c81kw7 _syaz1fxt _9oik1r31 _1bnx8stv _jf4cnqa1 _30l314q2 _9h8h16c2"
};

/**
 * __Progress tracker link__
 */
var Link = function Link(_ref) {
  var href = _ref.href,
    onClick = _ref.onClick,
    label = _ref.label,
    testId = _ref.testId;
  return /*#__PURE__*/_react.default.createElement(_compiled.Anchor, {
    xcss: styles.anchor
    // TODO: We should not be rendering empty hrefs on anchors. This should be plain text or a button/pressable if `onClick` is provided.
    ,
    href: href || '',
    onClick: onClick,
    testId: testId
  }, label);
};
var _default = exports.default = Link;
