// @atlaskit/react-ufo/dist/cjs/assets/utils.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.round = exports.calculateTransferType = exports.cacheableTypes = exports.NETWORK_KEY = exports.MEMORY_KEY = exports.DISK_KEY = void 0;
/* Borrowed from https://bitbucket.org/atlassian/atlassian-frontend/src/master/packages/performance/browser-metrics/src/plugins/timings/resource.ts */
var cacheableTypes = exports.cacheableTypes = ['script', 'link'];
var MEMORY_KEY = exports.MEMORY_KEY = 'mem';
var DISK_KEY = exports.DISK_KEY = 'disk';
var NETWORK_KEY = exports.NETWORK_KEY = 'net';
var calculateTransferType = exports.calculateTransferType = function calculateTransferType(type, duration, size) {
  if (!cacheableTypes.includes(type)) {
    return null;
  }
  if ((size === undefined || size === 0) && duration === 0) {
    return MEMORY_KEY;
  }
  if (size === 0 && duration > 0) {
    return DISK_KEY;
  }
  if (size === undefined) {
    return null;
  }
  return NETWORK_KEY;
};
var round = exports.round = function round(n) {
  if (isNaN(n)) {
    return 0;
  }
  return Math.round(n * 10000) / 10000;
};
