// @atlaskit/renderer/dist/es2019/react/nodes/blockCard.js
import _extends from "@babel/runtime/helpers/extends";
/**
 * @jsxRuntime classic
 * @jsx jsx
 */
// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766
import { css, jsx } from '@emotion/react';
import { Card } from '@atlaskit/smart-card';
import { UnsupportedBlock, UnsupportedInline, WidthConsumer } from '@atlaskit/editor-common/ui';
import { CardErrorBoundary } from './fallback';
import { getCardClickHandler } from '../utils/getCardClickHandler';
import InlineCard from './inlineCard';
import { AnalyticsContext } from '@atlaskit/analytics-next';
import { DatasourceTableView } from '@atlaskit/link-datasource';
import { N40 } from '@atlaskit/theme/colors';
import { calcBreakoutWidth, canRenderDatasource } from '@atlaskit/editor-common/utils';
const datasourceContainerStyle = css({
  borderRadius: `${"var(--ds-border-radius-200, 8px)"}`,
  border: `1px solid ${`var(--ds-border, ${N40})`}`,
  overflow: 'hidden',
  // eslint-disable-next-line @atlaskit/design-system/use-tokens-space
  marginLeft: '50%',
  marginBottom: `${"var(--ds-space-150, 0.75rem)"}`,
  transform: 'translateX(-50%)'
});
export default function BlockCard(props) {
  const {
    url,
    data,
    eventHandlers,
    portal,
    smartLinks,
    isNodeNested
  } = props;
  const {
    actionOptions
  } = smartLinks || {};
  const onClick = getCardClickHandler(eventHandlers, url);
  const platform = 'web';
  const cardProps = {
    url,
    data,
    onClick,
    container: portal,
    isDatasource: !!props.datasource,
    actionOptions
  };
  const analyticsData = {
    attributes: {
      location: 'renderer'
    },
    // Below is added for the future implementation of Linking Platform namespaced analytic context
    location: 'renderer'
  };
  const onError = ({
    err
  }) => {
    if (err) {
      throw err;
    }
  };
  if (props.datasource) {
    const views = props.datasource.views;
    const tableView = views.find(view => view.type === 'table');
    const shouldRenderDatasource = tableView && canRenderDatasource(props.datasource.id);
    if (shouldRenderDatasource) {
      var _tableView$properties;
      const columns = (_tableView$properties = tableView.properties) === null || _tableView$properties === void 0 ? void 0 : _tableView$properties.columns;
      const visibleColumnKeys = columns === null || columns === void 0 ? void 0 : columns.map(({
        key
      }) => key);
      const columnCustomSizesEntries = columns === null || columns === void 0 ? void 0 : columns.filter(c => !!c.width).map(({
        key,
        width
      }) => [key, width]);
      const columnCustomSizes = columnCustomSizesEntries !== null && columnCustomSizesEntries !== void 0 && columnCustomSizesEntries.length ? Object.fromEntries(columnCustomSizesEntries) : undefined;
      const wrappedColumnKeys = columns === null || columns === void 0 ? void 0 : columns.filter(c => c.isWrapped).map(c => c.key);
      const {
        datasource,
        layout
      } = props;
      return jsx(AnalyticsContext, {
        data: analyticsData
      }, jsx(CardErrorBoundary, _extends({
        unsupportedComponent: UnsupportedInline,
        datasourceId: props.datasource.id
        // Ignored via go/ees005
        // eslint-disable-next-line react/jsx-props-no-spreading
      }, cardProps), jsx(WidthConsumer, null, ({
        width
      }) => jsx("div", {
        css: datasourceContainerStyle,
        "data-testid": "renderer-datasource-table",
        style: {
          // eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values -- Ignored via go/DSP-18766
          width: isNodeNested ? '100%' : calcBreakoutWidth(layout, width)
        }
      }, jsx(DatasourceTableView, {
        datasourceId: datasource.id,
        parameters: datasource.parameters,
        visibleColumnKeys: visibleColumnKeys,
        columnCustomSizes: columnCustomSizes,
        wrappedColumnKeys: wrappedColumnKeys && wrappedColumnKeys.length > 0 ? wrappedColumnKeys : undefined,
        url: url
      })))));
    }
    return jsx(InlineCard, {
      data: data,
      url: url
    });
  }
  return jsx(AnalyticsContext, {
    data: analyticsData
  }, jsx("div", {
    // eslint-disable-next-line @atlaskit/ui-styling-standard/no-classname-prop -- Ignored via go/DSP-18766
    className: "blockCardView-content-wrap",
    "data-block-card": true,
    "data-card-data": data ? JSON.stringify(data) : undefined,
    "data-card-url": url
  }, jsx(CardErrorBoundary, _extends({
    unsupportedComponent: UnsupportedBlock
    // Ignored via go/ees005
    // eslint-disable-next-line react/jsx-props-no-spreading
  }, cardProps), jsx(Card, _extends({
    appearance: "block",
    platform: platform
    // Ignored via go/ees005
    // eslint-disable-next-line react/jsx-props-no-spreading
  }, cardProps, {
    onError: onError
  })))));
}
