// @atlaskit/renderer/dist/es2019/react/nodes/loosely-lazy.js
/* eslint-disable @atlaskit/editor/no-re-export */
// Mapping file

import { lazyForPaint } from 'react-loosely-lazy';
import Table from './table';
import TableRow from './tableRow';

// Ignored via go/ees005
// eslint-disable-next-line import/no-named-as-default

const CodeBlock = lazyForPaint(() => import( /* webpackChunkName: "@atlaskit-internal_renderer-node_CodeBlock" */
'./codeBlock/codeBlock').then(mod => mod.default));
const TaskList = lazyForPaint(() => import( /* webpackChunkName: "@atlaskit-internal_renderer-node_TaskList" */
'./taskList').then(mod => mod.default));
const TaskItem = lazyForPaint(() => import( /* webpackChunkName: "@atlaskit-internal_renderer-node_TaskItem" */
'./taskItem').then(mod => mod.default));
const DecisionList = lazyForPaint(() => import( /* webpackChunkName: "@atlaskit-internal_renderer-node_DecisionList" */
'./decisionList').then(mod => mod.default));
const DecisionItem = lazyForPaint(() => import( /* webpackChunkName: "@atlaskit-internal_renderer-node_DecisionItem" */
'./decisionItem').then(mod => mod.default));
const Date = lazyForPaint(() => import( /* webpackChunkName: "@atlaskit-internal_renderer-node_Date" */
'./date').then(mod => mod.default));
const Status = lazyForPaint(() => import( /* webpackChunkName: "@atlaskit-internal_renderer-node_Status" */
'./status').then(mod => mod.default));
const Emoji = lazyForPaint(() => import( /* webpackChunkName: "@atlaskit-internal_renderer-node_Emoji" */
'./emoji').then(mod => mod.default));
const Panel = lazyForPaint(() => import( /* webpackChunkName: "@atlaskit-internal_renderer-node_Panel" */
'./panel').then(mod => mod.default));
const EmbedCard = lazyForPaint(() => import( /* webpackChunkName: "@atlaskit-internal_renderer-node_EmbedCard" */
'./embedCard').then(mod => mod.default));
const InlineCard = lazyForPaint(() => import( /* webpackChunkName: "@atlaskit-internal_renderer-node_InlineCard" */
'./inlineCard').then(mod => mod.default));
const BlockCard = lazyForPaint(() => import( /* webpackChunkName: "@atlaskit-internal_renderer-node_BlockCard" */
'./blockCard').then(mod => mod.default));
const Media = lazyForPaint(() => import( /* webpackChunkName: "@atlaskit-internal_renderer-node_Media" */
'./media').then(mod => mod.default));
const MediaGroup = lazyForPaint(() => import( /* webpackChunkName: "@atlaskit-internal_renderer-node_MediaGroup" */
'./mediaGroup').then(mod => mod.default));
const MediaInline = lazyForPaint(() => import( /* webpackChunkName: "@atlaskit-internal_renderer-node_MediaInline" */
'./mediaInline').then(mod => mod.default));
const MediaSingle = lazyForPaint(() => import( /* webpackChunkName: "@atlaskit-internal_renderer-node_MediaSingle" */
'./mediaSingle').then(mod => mod.default));
const Mention = lazyForPaint(() => import( /* webpackChunkName: "@atlaskit-internal_renderer-node_Mention" */
'./mention').then(mod => mod.default));
const Expand = lazyForPaint(() => import( /* webpackChunkName: "@atlaskit-internal_renderer-node_Expand" */
'../../ui/Expand').then(mod => mod.default));
export const nodeToReact = {
  blockCard: BlockCard,
  date: Date,
  decisionItem: DecisionItem,
  decisionList: DecisionList,
  emoji: Emoji,
  codeBlock: CodeBlock,
  inlineCard: InlineCard,
  media: Media,
  mediaGroup: MediaGroup,
  mediaInline: MediaInline,
  mediaSingle: MediaSingle,
  mention: Mention,
  panel: Panel,
  status: Status,
  taskItem: TaskItem,
  taskList: TaskList,
  table: Table,
  tableRow: TableRow,
  expand: Expand,
  nestedExpand: Expand,
  embedCard: EmbedCard
};
