// @atlaskit/link-provider/dist/cjs/provider.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
var _typeof = require("@babel/runtime/helpers/typeof");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SmartCardProvider = SmartCardProvider;
exports.default = void 0;
var _react = _interopRequireWildcard(require("react"));
var _redux = require("redux");
var _linkingCommon = require("@atlaskit/linking-common");
var _platformFeatureFlags = require("@atlaskit/platform-feature-flags");
var _linkExtractors = require("@atlaskit/link-extractors");
var _reducers = require("./reducers");
var _context = require("./state/context");
var _client = _interopRequireDefault(require("./client"));
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function SmartCardProvider(_ref) {
  var storeOptions = _ref.storeOptions,
    customClient = _ref.client,
    customAuthFlow = _ref.authFlow,
    children = _ref.children,
    renderers = _ref.renderers,
    featureFlags = _ref.featureFlags,
    isAdminHubAIEnabled = _ref.isAdminHubAIEnabled,
    product = _ref.product,
    shouldControlDataExport = _ref.shouldControlDataExport;
  var parentContext = (0, _react.useContext)(_context.SmartCardContext);
  var defaultInitialState = (0, _react.useMemo)(function () {
    return {};
  }, []);
  var _ref2 = storeOptions || {
      initialState: defaultInitialState
    },
    initialState = _ref2.initialState;
  var store = (0, _react.useMemo)(function () {
    return (0, _redux.createStore)(_reducers.cardReducer, initialState);
  }, [initialState]);
  var providerValue = (0, _react.useMemo)(function () {
    var client = customClient || new _client.default();

    // If product is passed into provider, set it on the client if setProduct is supported
    if (product && client.setProduct) {
      client.setProduct(product);
    }
    var authFlow = customAuthFlow || 'oauth2';
    var getPreview = function getPreview(url, platform) {
      var cardState = (0, _linkingCommon.getUrl)(store, url);
      return cardState.details ? (0, _platformFeatureFlags.fg)('smart_links_noun_support') ? (0, _linkExtractors.extractSmartLinkEmbed)(cardState.details) : (0, _linkExtractors.extractPreview)(cardState.details.data, platform) : undefined;
    };
    return {
      renderers: renderers,
      store: store,
      prefetchStore: {},
      connections: {
        client: client
      },
      config: {
        authFlow: authFlow
      },
      extractors: {
        getPreview: getPreview
      },
      featureFlags: featureFlags,
      isAdminHubAIEnabled: isAdminHubAIEnabled,
      product: product,
      shouldControlDataExport: shouldControlDataExport
    };
  }, [customClient, customAuthFlow, isAdminHubAIEnabled, product, shouldControlDataExport, renderers, featureFlags, store]);
  return /*#__PURE__*/_react.default.createElement(_context.SmartCardContext.Provider, {
    value: parentContext || providerValue
  }, children);
}
var _default = exports.default = SmartCardProvider;
