// @atlaskit/link-datasource/dist/es2019/ui/common/error-state/modal-loading-error-old.js
/**
 * @jsxRuntime classic
 * @jsx jsx
 */
import { useEffect } from 'react';

// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766
import { css, jsx } from '@emotion/react';
import { FormattedMessage } from 'react-intl-next';
import { Box, Text } from '@atlaskit/primitives';
import { useDatasourceAnalyticsEvents } from '../../../analytics';
import { LoadingErrorSVGOld } from './loading-error-svg-old';
import { loadingErrorMessages } from './messages';
const errorContainerStyles = css({
  display: 'grid',
  gap: "var(--ds-space-300, 24px)",
  placeItems: 'center',
  placeSelf: 'center'
});
const errorMessageContainerStyles = css({
  display: 'grid',
  gap: "var(--ds-space-100, 8px)",
  placeItems: 'center'
});
export const ModalLoadingErrorOld = ({
  errorMessage = jsx(FormattedMessage, loadingErrorMessages.checkConnection)
}) => {
  const {
    fireEvent
  } = useDatasourceAnalyticsEvents();
  useEffect(() => {
    fireEvent('ui.error.shown', {
      reason: 'network'
    });
  }, [fireEvent]);
  return jsx(Box, {
    xcss: errorContainerStyles,
    testId: "datasource-modal--loading-error"
  }, jsx(LoadingErrorSVGOld, null), jsx(Box, {
    xcss: errorMessageContainerStyles
  }, jsx(Text, {
    size: "small"
  }, jsx(FormattedMessage, loadingErrorMessages.unableToLoadResults)), jsx(Text, {
    as: "p"
  }, errorMessage)));
};
