// @atlaskit/link-datasource/dist/es2019/ui/common/error-state/no-instances.js
/* no-instances.tsx generated by @compiled/babel-plugin v0.36.1 */
import "./no-instances.compiled.css";
import * as React from 'react';
import { ax, ix } from "@compiled/react/runtime";
import { useIntl } from 'react-intl-next';
import { fg } from '@atlaskit/platform-feature-flags';
import { Flex, Inline } from '@atlaskit/primitives/compiled';
import { NoInstancesViewOld } from './no-instances-old';
import { NoInstancesSvg } from './no-instances-svg';
const styles = {
  titleStyles: "_11c8nbxd _19pkpxbi",
  descriptionStyles: "_19pku2gc",
  containerStyles: "_19pkxncg"
};
export const NoInstancesViewNew = ({
  title,
  description,
  testId
}) => {
  const {
    formatMessage
  } = useIntl();
  return /*#__PURE__*/React.createElement(Flex, {
    testId: testId,
    direction: "column",
    alignItems: "center",
    xcss: styles.containerStyles
  }, /*#__PURE__*/React.createElement(NoInstancesSvg, null), /*#__PURE__*/React.createElement(Inline, {
    as: "span",
    xcss: styles.titleStyles
  }, formatMessage(title)), /*#__PURE__*/React.createElement(Inline, {
    as: "span",
    xcss: styles.descriptionStyles
  }, formatMessage(description)));
};
export const NoInstancesView = props => {
  if (fg('bandicoots-compiled-migration-link-datasource')) {
    return /*#__PURE__*/React.createElement(NoInstancesViewNew, props);
  } else {
    return /*#__PURE__*/React.createElement(NoInstancesViewOld, props);
  }
};
