// @atlaskit/smart-card/dist/cjs/view/HoverCard/components/ContentContainerOld.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));
var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));
var _react = require("react");
var _react2 = require("@emotion/react");
var _reactMagneticDi = require("react-magnetic-di");
var _useAiSummaryAction = _interopRequireDefault(require("../../../state/hooks/use-ai-summary-action"));
var _aiPrism = _interopRequireDefault(require("../../common/ai-prism"));
var _styled = require("../styled");
var _HoverCardContent = require("./HoverCardContent");
var _excluded = ["children", "isAIEnabled", "testId", "url"],
  _excluded2 = ["children", "isAIEnabled", "testId", "url"];
/**
 * @jsxRuntime classic
 * @jsx jsx
 */
// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766
var ConnectedAIPrismContainer = function ConnectedAIPrismContainer(_ref) {
  var children = _ref.children,
    _ref$isAIEnabled = _ref.isAIEnabled,
    isAIEnabled = _ref$isAIEnabled === void 0 ? false : _ref$isAIEnabled,
    testId = _ref.testId,
    url = _ref.url,
    props = (0, _objectWithoutProperties2.default)(_ref, _excluded);
  var _useAISummaryAction = (0, _useAiSummaryAction.default)(url),
    status = _useAISummaryAction.state.status;
  var _useState = (0, _react.useState)(status === 'loading'),
    _useState2 = (0, _slicedToArray2.default)(_useState, 2),
    showPrism = _useState2[0],
    setShowPrism = _useState2[1];
  (0, _react.useEffect)(function () {
    setShowPrism(status === 'loading');
  }, [status]);
  var container = (0, _react2.jsx)("div", (0, _extends2.default)({
    // eslint-disable-next-line @atlaskit/ui-styling-standard/no-classname-prop -- Ignored via go/DSP-18766
    className: _HoverCardContent.hoverCardClassName
    // eslint-disable-next-line @atlaskit/design-system/consistent-css-prop-usage, @atlaskit/ui-styling-standard/no-imported-style-values -- Ignored via go/DSP-18766
    ,
    css: [_styled.HoverCardContainer, !isAIEnabled ? _styled.popupContainerStyles : undefined],
    "data-testid": testId
  }, props), children);
  return isAIEnabled ? (0, _react2.jsx)(_aiPrism.default, {
    isVisible: showPrism,
    testId: "".concat(testId, "-prism")
  }, container) : container;
};
var ContentContainerOld = function ContentContainerOld(_ref2) {
  var children = _ref2.children,
    _ref2$isAIEnabled = _ref2.isAIEnabled,
    isAIEnabled = _ref2$isAIEnabled === void 0 ? false : _ref2$isAIEnabled,
    testId = _ref2.testId,
    url = _ref2.url,
    props = (0, _objectWithoutProperties2.default)(_ref2, _excluded2);
  return (0, _react2.jsx)(ConnectedAIPrismContainer, (0, _extends2.default)({
    isAIEnabled: isAIEnabled,
    url: url,
    testId: testId
  }, props), children);
};
var _default = exports.default = ContentContainerOld;
