// @atlaskit/smart-card/dist/cjs/view/EmbedCard/components/ErrorFrameOld.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.FrameOld = exports.ExpandedFrameOld = exports.CompactFrameOld = void 0;
var _react = _interopRequireDefault(require("react"));
var _react2 = require("@emotion/react");
var _colors = require("@atlaskit/theme/colors");
var _utils = require("../../common/utils");
/**
 * @jsxRuntime classic
 * @jsx jsx
 */

// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766

var FrameOld = exports.FrameOld = function FrameOld() {
  var props = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {
    isSelected: false,
    isHoverable: false,
    isFluidHeight: false
  };
  return props.compact ? (0, _react2.jsx)(CompactFrameOld, props) : (0, _react2.jsx)(ExpandedFrameOld, props);
};
var sharedBaseFrameStyles = (0, _react2.css)({
  width: '100%',
  display: 'flex'
});

// eslint-disable-next-line @atlaskit/design-system/consistent-css-prop-usage -- Ignored via go/DSP-18766
var sharedFrameStyles = {
  maxWidth: (0, _utils.gs)(95),
  backgroundColor: "var(--ds-surface-raised, white)"
};
var ExpandedFrameOld = exports.ExpandedFrameOld = function ExpandedFrameOld(_ref) {
  var children = _ref.children,
    isSelected = _ref.isSelected,
    isHoverable = _ref.isHoverable,
    testId = _ref.testId,
    className = _ref.className,
    isFluidHeight = _ref.isFluidHeight;
  return (0, _react2.jsx)("div", {
    css: [sharedBaseFrameStyles, sharedFrameStyles,
    // eslint-disable-next-line @atlaskit/design-system/consistent-css-prop-usage, @atlaskit/ui-styling-standard/no-imported-style-values -- Ignored via go/DSP-18766
    (0, _utils.mq)({
      '&:hover': isHoverable ? {
        // TODO: https://product-fabric.atlassian.net/browse/DSP-4064
        backgroundColor: "var(--ds-background-neutral-subtle-hovered, ".concat(_colors.N20A, ")"),
        cursor: 'pointer'
      } : undefined,
      // eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values -- Ignored via go/DSP-18766
      minHeight: isFluidHeight ? 0 : [(0, _utils.gs)(21), (0, _utils.gs)(15)],
      // eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values -- Ignored via go/DSP-18766
      borderRadius: isSelected ? (0, _utils.br)() : (0, _utils.br)(0.5),
      border: "2px solid ".concat(isSelected ? "var(--ds-border-selected, ".concat(_colors.B200, ")") : 'transparent'),
      justifyContent: 'space-between',
      overflow: 'hidden',
      boxShadow: "var(--ds-shadow-raised, ".concat("0 1px 1px ".concat(_colors.N50A, ", 0 0 1px 1px ").concat(_colors.N40A), ")")
    })],
    "data-testid": testId
    // eslint-disable-next-line @atlaskit/ui-styling-standard/no-classname-prop -- Ignored via go/DSP-18766
    ,
    className: className,
    "data-trello-do-not-use-override": testId
  }, children);
};
var CompactFrameOld = exports.CompactFrameOld = function CompactFrameOld(_ref2) {
  var children = _ref2.children,
    isHoverable = _ref2.isHoverable,
    isSelected = _ref2.isSelected,
    testId = _ref2.testId,
    className = _ref2.className,
    inheritDimensions = _ref2.inheritDimensions;
  return (0, _react2.jsx)("div", {
    css: [sharedBaseFrameStyles, sharedFrameStyles,
    // eslint-disable-next-line @atlaskit/design-system/consistent-css-prop-usage, @atlaskit/ui-styling-standard/no-imported-style-values -- Ignored via go/DSP-18766
    (0, _utils.mq)({
      '&:hover': isHoverable ? {
        backgroundColor: "var(--ds-background-neutral-hovered, ".concat(_colors.N30A, ")")
      } : undefined,
      // eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values -- Ignored via go/DSP-18766
      borderRadius: isSelected ? (0, _utils.br)() : (0, _utils.br)(0.5),
      border: isSelected ? "2px solid ".concat("var(--ds-border-selected, ".concat(_colors.B200, ")")) : '',
      justifyContent: 'center',
      alignItems: 'center',
      // eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values -- Ignored via go/DSP-18766
      height: inheritDimensions ? '100%' : (0, _utils.gs)(5),
      backgroundColor: "var(--ds-background-neutral, ".concat(_colors.N20A, ")"),
      width: ['calc(100% - 16px)', '100%'],
      // eslint-disable-next-line @atlaskit/design-system/ensure-design-token-usage/preview, @atlaskit/ui-styling-standard/no-imported-style-values -- Ignored via go/DSP-18766
      padding: ["0px ".concat((0, _utils.gs)(1)), '0']
    })],
    "data-testid": testId
    // eslint-disable-next-line @atlaskit/ui-styling-standard/no-classname-prop -- Ignored via go/DSP-18766
    ,
    className: className
  }, children);
};
