// @atlaskit/smart-card/dist/es2019/view/FlexibleCard/components/blocks/action-block/ActionBlockOld.js
/**
 * @jsxRuntime classic
 * @jsx jsx
 */
import { useCallback, useMemo, useState } from 'react';

// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766
import { css, jsx } from '@emotion/react';
import { di } from 'react-magnetic-di';
import { xcss } from '@atlaskit/primitives';
import { SmartLinkSize } from '../../../../../constants';
import { useFlexibleUiContext, useFlexibleUiOptionContext } from '../../../../../state/flexible-ui-context';
import * as Actions from '../../actions';
import { getPrimitivesPaddingSpaceBySize } from '../../utils';
import { ActionFooter } from './action-footer';
const ignoreContainerPaddingStyles = css({
  display: 'flex',
  flexDirection: 'column',
  boxSizing: 'border-box',
  flexGrow: 1,
  width: '100%',
  // We have to find a better way to ignore container padding
  // This has become more and more of a common use case.
  marginLeft: 'calc(var(--container-gap-left) * -1)',
  marginRight: 'calc(var(--container-gap-right) * -1)'
});
const DEFAULT_SORT_ORDER = ['PreviewAction', 'CopyLinkAction', 'AISummaryAction'];
const sort = (a, b) => {
  let idxA = DEFAULT_SORT_ORDER.indexOf(a);
  let idxB = DEFAULT_SORT_ORDER.indexOf(b);
  if (idxA === -1) {
    idxA = DEFAULT_SORT_ORDER.length;
  }
  if (idxB === -1) {
    idxB = DEFAULT_SORT_ORDER.length;
  }
  return idxA - idxB;
};
const ActionBlockOld = ({
  blockRef,
  onClick: onClickCallback,
  size,
  spaceInline,
  testId = 'smart-block-action'
}) => {
  const context = useFlexibleUiContext();
  const ui = useFlexibleUiOptionContext();
  const [message, setMessage] = useState();
  const [isLoading, setIsLoading] = useState(false);
  const onLoadingChange = useCallback(isLoading => {
    setIsLoading(isLoading);
  }, [setIsLoading]);
  const padding = !(ui !== null && ui !== void 0 && ui.hidePadding) ? getPrimitivesPaddingSpaceBySize((ui === null || ui === void 0 ? void 0 : ui.size) || SmartLinkSize.Medium) : undefined;
  const onClick = useCallback(name => {
    setMessage(undefined);
    onClickCallback === null || onClickCallback === void 0 ? void 0 : onClickCallback(name);
  }, [onClickCallback]);
  const onError = useCallback(error => {
    setMessage(error);
  }, []);
  const actions = useMemo(() => {
    if (!(context !== null && context !== void 0 && context.actions)) {
      return;
    }
    const arr = Object.keys(context.actions);
    arr.sort(sort);
    return arr.map(name => {
      const Action = name in Actions ? Actions[name] : undefined;
      return Action ? jsx(Action, {
        as: "stack-item",
        spaceInline: spaceInline,
        key: name,
        onClick: () => onClick(name),
        onError: onError,
        onLoadingChange: onLoadingChange,
        size: size
        // eslint-disable-next-line @atlaskit/ui-styling-standard/no-unsafe-values, @atlaskit/ui-styling-standard/no-imported-style-values, @atlaskit/design-system/consistent-css-prop-usage -- Ignored via go/DSP-18766
        ,
        xcss: xcss({
          paddingInline: padding
        }),
        hideTooltip: isLoading
      }) : null;
    });
  }, [context === null || context === void 0 ? void 0 : context.actions, onClick, onError, padding, size, spaceInline, isLoading, onLoadingChange]);
  return actions ? jsx("div", {
    css: ignoreContainerPaddingStyles,
    ref: blockRef,
    "data-testid": testId
  }, actions, jsx(ActionFooter, {
    message: message,
    paddingInline: padding,
    testId: testId
  })) : null;
};
export default ActionBlockOld;
