// @atlaskit/link-datasource/dist/cjs/ui/common/error-state/loading-error-old.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.LoadingErrorOld = void 0;
var _react = require("react");
var _react2 = require("@emotion/react");
var _reactIntlNext = require("react-intl-next");
var _standardButton = _interopRequireDefault(require("@atlaskit/button/standard-button"));
var _primitives = require("@atlaskit/primitives");
var _typography = require("@atlaskit/theme/typography");
var _analytics = require("../../../analytics");
var _loadingErrorSvgOld = require("./loading-error-svg-old");
var _messages = require("./messages");
/**
 * @jsxRuntime classic
 * @jsx jsx
 */

// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766

var errorContainerStyles = (0, _react2.css)({
  display: 'grid',
  gap: "var(--ds-space-200, 16px)",
  placeItems: 'center',
  placeSelf: 'center'
});
var errorMessageContainerStyles = (0, _react2.css)({
  display: 'grid',
  gap: "var(--ds-space-100, 8px)",
  placeItems: 'center'
});
var errorMessageStyles = (0, _react2.css)({
  // eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values -- Ignored via go/DSP-18766
  font: "var(--ds-font-heading-small, ".concat(_typography.fontFallback.heading.small, ")")
});
var LoadingErrorOld = exports.LoadingErrorOld = function LoadingErrorOld(_ref) {
  var onRefresh = _ref.onRefresh;
  var _useDatasourceAnalyti = (0, _analytics.useDatasourceAnalyticsEvents)(),
    fireEvent = _useDatasourceAnalyti.fireEvent;
  (0, _react.useEffect)(function () {
    fireEvent('ui.error.shown', {
      reason: 'network'
    });
  }, [fireEvent]);
  return (0, _react2.jsx)(_primitives.Box, {
    xcss: errorContainerStyles,
    testId: "datasource--loading-error"
  }, (0, _react2.jsx)(_loadingErrorSvgOld.LoadingErrorSVGOld, null), (0, _react2.jsx)(_primitives.Box, {
    xcss: errorMessageContainerStyles
  }, (0, _react2.jsx)(_primitives.Inline, {
    as: "span",
    xcss: errorMessageStyles
  }, (0, _react2.jsx)(_reactIntlNext.FormattedMessage, _messages.loadingErrorMessages.unableToLoadItems)), (0, _react2.jsx)(_primitives.Text, {
    as: "p"
  }, (0, _react2.jsx)(_reactIntlNext.FormattedMessage, _messages.loadingErrorMessages.checkConnection)), onRefresh && (0, _react2.jsx)(_standardButton.default, {
    appearance: "primary",
    onClick: onRefresh
  }, (0, _react2.jsx)(_reactIntlNext.FormattedMessage, _messages.loadingErrorMessages.refresh))));
};
