// @atlaskit/link-datasource/dist/cjs/ui/common/error-state/loading-error.js
/* loading-error.tsx generated by @compiled/babel-plugin v0.36.1 */
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
var _typeof = require("@babel/runtime/helpers/typeof");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.LoadingErrorNew = exports.LoadingError = void 0;
require("./loading-error.compiled.css");
var _react = _interopRequireWildcard(require("react"));
var React = _react;
var _runtime = require("@compiled/react/runtime");
var _reactIntlNext = require("react-intl-next");
var _standardButton = _interopRequireDefault(require("@atlaskit/button/standard-button"));
var _platformFeatureFlags = require("@atlaskit/platform-feature-flags");
var _compiled = require("@atlaskit/primitives/compiled");
var _typography = require("@atlaskit/theme/typography");
var _analytics = require("../../../analytics");
var _search = require("../../../common/ui/spot/error-state/search");
var _loadingErrorOld = require("./loading-error-old");
var _loadingErrorSvgOld = require("./loading-error-svg-old");
var _messages = require("./messages");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
var styles = {
  errorContainerStyles: "_zulppxbi _4cvr1h6o _6a6z1h6o _1wpz1h6o _10fp1h6o _1e0c11p5",
  errorMessageContainerStyles: "_zulpu2gc _4cvr1h6o _6a6z1h6o _1e0c11p5",
  errorMessageStyles: "_11c8nbxd"
};
var LoadingErrorNew = exports.LoadingErrorNew = function LoadingErrorNew(_ref) {
  var onRefresh = _ref.onRefresh;
  var _useDatasourceAnalyti = (0, _analytics.useDatasourceAnalyticsEvents)(),
    fireEvent = _useDatasourceAnalyti.fireEvent;
  (0, _react.useEffect)(function () {
    fireEvent('ui.error.shown', {
      reason: 'network'
    });
  }, [fireEvent]);
  return /*#__PURE__*/React.createElement(_compiled.Box, {
    xcss: styles.errorContainerStyles,
    testId: "datasource--loading-error"
  }, (0, _platformFeatureFlags.fg)('bandicoots-update-sllv-icons') ? /*#__PURE__*/React.createElement(_search.SpotErrorSearch, {
    size: 'xlarge',
    alt: ""
  }) : /*#__PURE__*/React.createElement(_loadingErrorSvgOld.LoadingErrorSVGOld, null), /*#__PURE__*/React.createElement(_compiled.Box, {
    xcss: styles.errorMessageContainerStyles
  }, /*#__PURE__*/React.createElement(_compiled.Inline, {
    as: "span",
    xcss: styles.errorMessageStyles
  }, /*#__PURE__*/React.createElement(_reactIntlNext.FormattedMessage, _messages.loadingErrorMessages.unableToLoadItems)), /*#__PURE__*/React.createElement(_compiled.Text, {
    as: "p"
  }, /*#__PURE__*/React.createElement(_reactIntlNext.FormattedMessage, _messages.loadingErrorMessages.checkConnection)), onRefresh && /*#__PURE__*/React.createElement(_standardButton.default, {
    appearance: "primary",
    onClick: onRefresh
  }, /*#__PURE__*/React.createElement(_reactIntlNext.FormattedMessage, _messages.loadingErrorMessages.refresh))));
};
var LoadingError = exports.LoadingError = function LoadingError(props) {
  if ((0, _platformFeatureFlags.fg)('bandicoots-compiled-migration-link-datasource')) {
    return /*#__PURE__*/React.createElement(LoadingErrorNew, props);
  } else {
    return /*#__PURE__*/React.createElement(_loadingErrorOld.LoadingErrorOld, props);
  }
};
