// @atlaskit/link-datasource/dist/cjs/ui/common/error-state/no-instances-old.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.NoInstancesViewOld = void 0;
var _react = require("@emotion/react");
var _reactIntlNext = require("react-intl-next");
var _primitives = require("@atlaskit/primitives");
var _noInstancesSvg = require("./no-instances-svg");
/**
 * @jsxRuntime classic
 * @jsx jsx
 */
// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766

var titleStyles = (0, _primitives.xcss)({
  font: 'font.heading.small',
  marginTop: 'space.200'
});
var descriptionStyles = (0, _primitives.xcss)({
  marginTop: 'space.100'
});
var containerStyles = (0, _primitives.xcss)({
  marginTop: 'space.800'
});
var NoInstancesViewOld = exports.NoInstancesViewOld = function NoInstancesViewOld(_ref) {
  var title = _ref.title,
    description = _ref.description,
    testId = _ref.testId;
  var _useIntl = (0, _reactIntlNext.useIntl)(),
    formatMessage = _useIntl.formatMessage;
  return (0, _react.jsx)(_primitives.Flex, {
    testId: testId,
    direction: "column",
    alignItems: "center",
    xcss: containerStyles
  }, (0, _react.jsx)(_noInstancesSvg.NoInstancesSvg, null), (0, _react.jsx)(_primitives.Inline, {
    as: "span",
    xcss: titleStyles
  }, formatMessage(title)), (0, _react.jsx)(_primitives.Inline, {
    as: "span",
    xcss: descriptionStyles
  }, formatMessage(description)));
};
