// @atlaskit/smart-card/dist/cjs/view/FlexibleCard/components/actions/automation-action/automation-manual-triggers/manual-triggers-modal/sub-components/rule-list/main.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.AutomationModalRuleList = void 0;
var _react = _interopRequireDefault(require("react"));
var _menu = require("@atlaskit/menu");
var _platformFeatureFlags = require("@atlaskit/platform-feature-flags");
var _compiled = require("@atlaskit/primitives/compiled");
var _menuContext = require("../../menu-context");
var _emptyState = require("../empty-state");
var _rule = require("../rule");
var _AutomationModalRuleListOld = require("./AutomationModalRuleListOld");
var AutomationModalRuleListNew = function AutomationModalRuleListNew(_ref) {
  var selectedRule = _ref.selectedRule,
    setSelectedRule = _ref.setSelectedRule;
  var _useAutomationMenu = (0, _menuContext.useAutomationMenu)(),
    rules = _useAutomationMenu.rules;
  if (rules.length === 0) {
    return /*#__PURE__*/_react.default.createElement(_emptyState.AutomationModalEmptyState, null);
  }
  return /*#__PURE__*/_react.default.createElement(_menu.MenuGroup, null, /*#__PURE__*/_react.default.createElement(_compiled.Stack, {
    space: "space.050"
  }, rules.map(function (rule) {
    return /*#__PURE__*/_react.default.createElement(_rule.AutomationModalRule, {
      key: "manual-rule-".concat(rule.id),
      rule: rule,
      selectedRule: selectedRule,
      setSelectedRule: setSelectedRule
    });
  })));
};
var AutomationModalRuleList = exports.AutomationModalRuleList = function AutomationModalRuleList(props) {
  if ((0, _platformFeatureFlags.fg)('bandicoots-compiled-migration-smartcard')) {
    return /*#__PURE__*/_react.default.createElement(AutomationModalRuleListNew, props);
  }
  return /*#__PURE__*/_react.default.createElement(_AutomationModalRuleListOld.AutomationModalRuleListOld, props);
};
