// @atlaskit/smart-card/dist/cjs/view/BlockCard/views/unresolved-view/UnresolvedViewOld.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _react = _interopRequireDefault(require("react"));
var _react2 = require("@emotion/react");
var _FlexibleCard = _interopRequireDefault(require("../../../FlexibleCard"));
var _blocks = require("../../../FlexibleCard/components/blocks");
var _utils = require("../utils");
// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled

var customBlockStyles = (0, _react2.css)({
  alignItems: 'flex-start'
});
var UnresolvedViewOld = function UnresolvedViewOld(_ref) {
  var actions = _ref.actions,
    cardState = _ref.cardState,
    children = _ref.children,
    onAuthorize = _ref.onAuthorize,
    onClick = _ref.onClick,
    onError = _ref.onError,
    _ref$showPreview = _ref.showPreview,
    showPreview = _ref$showPreview === void 0 ? false : _ref$showPreview,
    testId = _ref.testId,
    title = _ref.title,
    url = _ref.url;
  return /*#__PURE__*/_react.default.createElement(_FlexibleCard.default, {
    appearance: "block",
    cardState: cardState,
    onAuthorize: onAuthorize,
    onClick: onClick,
    onError: onError,
    origin: "smartLinkCard",
    testId: testId,
    ui: _utils.FlexibleCardUiOptions,
    url: url
  }, /*#__PURE__*/_react.default.createElement(_blocks.TitleBlock, (0, _extends2.default)({}, _utils.titleBlockOptionsOld, {
    hideIcon: !!title,
    text: title
  })), /*#__PURE__*/_react.default.createElement(_blocks.CustomBlock, {
    overrideCss: customBlockStyles
  }, children), showPreview && /*#__PURE__*/_react.default.createElement(_blocks.PreviewBlock, _utils.PreviewBlockOptions), /*#__PURE__*/_react.default.createElement(_blocks.InternalFooterBlock, (0, _extends2.default)({}, _utils.FooterBlockOptionsOld, {
    actions: actions,
    testId: "smart-block-card-footer"
  })));
};
var _default = exports.default = UnresolvedViewOld;
