// @atlaskit/smart-card/dist/es2019/view/EmbedCard/components/ErrorFrameOld.js
/**
 * @jsxRuntime classic
 * @jsx jsx
 */
import React from 'react';

// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766
import { css, jsx } from '@emotion/react';
import { B200, N20A, N30A, N40A, N50A } from '@atlaskit/theme/colors';
import { br, gs, mq } from '../../common/utils';
export const FrameOld = (props = {
  isSelected: false,
  isHoverable: false,
  isFluidHeight: false
}) => props.compact ? jsx(CompactFrameOld, props) : jsx(ExpandedFrameOld, props);
const sharedBaseFrameStyles = css({
  width: '100%',
  display: 'flex'
});

// eslint-disable-next-line @atlaskit/design-system/consistent-css-prop-usage -- Ignored via go/DSP-18766
const sharedFrameStyles = {
  maxWidth: gs(95),
  backgroundColor: "var(--ds-surface-raised, white)"
};
export const ExpandedFrameOld = ({
  children,
  isSelected,
  isHoverable,
  testId,
  className,
  isFluidHeight
}) => {
  return jsx("div", {
    css: [sharedBaseFrameStyles, sharedFrameStyles,
    // eslint-disable-next-line @atlaskit/design-system/consistent-css-prop-usage, @atlaskit/ui-styling-standard/no-imported-style-values -- Ignored via go/DSP-18766
    mq({
      '&:hover': isHoverable ? {
        // TODO: https://product-fabric.atlassian.net/browse/DSP-4064
        backgroundColor: `var(--ds-background-neutral-subtle-hovered, ${N20A})`,
        cursor: 'pointer'
      } : undefined,
      // eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values -- Ignored via go/DSP-18766
      minHeight: isFluidHeight ? 0 : [gs(21), gs(15)],
      // eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values -- Ignored via go/DSP-18766
      borderRadius: isSelected ? br() : br(0.5),
      border: `2px solid ${isSelected ? `var(--ds-border-selected, ${B200})` : 'transparent'}`,
      justifyContent: 'space-between',
      overflow: 'hidden',
      boxShadow: `var(--ds-shadow-raised, ${`0 1px 1px ${N50A}, 0 0 1px 1px ${N40A}`})`
    })],
    "data-testid": testId
    // eslint-disable-next-line @atlaskit/ui-styling-standard/no-classname-prop -- Ignored via go/DSP-18766
    ,
    className: className,
    "data-trello-do-not-use-override": testId
  }, children);
};
export const CompactFrameOld = ({
  children,
  isHoverable,
  isSelected,
  testId,
  className,
  inheritDimensions
}) => {
  return jsx("div", {
    css: [sharedBaseFrameStyles, sharedFrameStyles,
    // eslint-disable-next-line @atlaskit/design-system/consistent-css-prop-usage, @atlaskit/ui-styling-standard/no-imported-style-values -- Ignored via go/DSP-18766
    mq({
      '&:hover': isHoverable ? {
        backgroundColor: `var(--ds-background-neutral-hovered, ${N30A})`
      } : undefined,
      // eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values -- Ignored via go/DSP-18766
      borderRadius: isSelected ? br() : br(0.5),
      border: isSelected ? `2px solid ${`var(--ds-border-selected, ${B200})`}` : '',
      justifyContent: 'center',
      alignItems: 'center',
      // eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values -- Ignored via go/DSP-18766
      height: inheritDimensions ? '100%' : gs(5),
      backgroundColor: `var(--ds-background-neutral, ${N20A})`,
      width: ['calc(100% - 16px)', '100%'],
      // eslint-disable-next-line @atlaskit/design-system/ensure-design-token-usage/preview, @atlaskit/ui-styling-standard/no-imported-style-values -- Ignored via go/DSP-18766
      padding: [`0px ${gs(1)}`, '0']
    })],
    "data-testid": testId
    // eslint-disable-next-line @atlaskit/ui-styling-standard/no-classname-prop -- Ignored via go/DSP-18766
    ,
    className: className
  }, children);
};
