// @atlaskit/link-datasource/dist/cjs/ui/assets-modal/search-container/search-container-old.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.AssetsSearchContainerOld = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _react = require("@emotion/react");
var _form = _interopRequireDefault(require("@atlaskit/form"));
var _analytics = require("../../../analytics");
var _aqlSearchInput = require("./aql-search-input");
var _objectSchemaSelect = require("./object-schema-select");
var _styledOld = require("./styled-old");
/**
 * @jsxRuntime classic
 * @jsx jsx
 */
// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766

var DEFAULT_AQL_QUERY = '';
var SEARCH_FORM_ID = 'linkDataSource.assets.configModal.searchContainer-form';
var AssetsSearchContainerOld = exports.AssetsSearchContainerOld = function AssetsSearchContainerOld(props) {
  var onSearch = props.onSearch,
    workspaceId = props.workspaceId,
    initialSearchData = props.initialSearchData,
    modalTitle = props.modalTitle,
    isSearching = props.isSearching;
  var _useDatasourceAnalyti = (0, _analytics.useDatasourceAnalyticsEvents)(),
    fireEvent = _useDatasourceAnalyti.fireEvent;
  var onFormSubmit = function onFormSubmit(searchFormValues) {
    var aql = searchFormValues.aql,
      objectSchema = searchFormValues.objectSchema;
    if (aql && objectSchema) {
      fireEvent('ui.aqlEditor.searched', {});
      // Pass the validated aql and object schema back to modal
      onSearch(aql, objectSchema.value);
    }
  };
  return (0, _react.jsx)(_form.default, {
    onSubmit: onFormSubmit
  }, function (_ref) {
    var _initialSearchData$ob, _initialSearchData$ob2, _initialSearchData$aq;
    var formProps = _ref.formProps;
    return (0, _react.jsx)(_styledOld.FormContainer, (0, _extends2.default)({}, formProps, {
      id: SEARCH_FORM_ID
    }), (0, _react.jsx)(_styledOld.FormRowContainer, {
      isNarrowGap: true
    }, modalTitle, (0, _react.jsx)(_styledOld.SchemaSelectContainer, null, (0, _react.jsx)(_objectSchemaSelect.AssetsObjectSchemaSelect, {
      value: (_initialSearchData$ob = initialSearchData.objectSchema) !== null && _initialSearchData$ob !== void 0 ? _initialSearchData$ob : undefined,
      workspaceId: workspaceId,
      initialObjectSchemas: (_initialSearchData$ob2 = initialSearchData.objectSchemas) !== null && _initialSearchData$ob2 !== void 0 ? _initialSearchData$ob2 : undefined,
      classNamePrefix: "assets-datasource-modal--object-schema-select"
    }))), (0, _react.jsx)(_styledOld.FormRowContainer, null, (0, _react.jsx)(_aqlSearchInput.AqlSearchInput, {
      value: (_initialSearchData$aq = initialSearchData.aql) !== null && _initialSearchData$aq !== void 0 ? _initialSearchData$aq : DEFAULT_AQL_QUERY,
      workspaceId: workspaceId,
      isSearching: isSearching
    })));
  });
};
