// @atlaskit/smart-card/dist/cjs/view/FlexibleCard/components/blocks/title-block/resolved/TitleBlockResolvedViewOld.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));
var _react = _interopRequireDefault(require("react"));
var _react2 = require("@emotion/react");
var _constants = require("../../../../../../constants");
var _elements = require("../../../elements");
var _block = _interopRequireDefault(require("../../block"));
var _elementGroup = _interopRequireDefault(require("../../element-group"));
var _utils = require("../../utils");
var _excluded = ["actionGroup", "metadata", "position", "subtitle", "testId", "text", "icon", "title", "metadataPosition", "hideIcon"]; // eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766
/**
 * This renders a fully resolved TitleBlock.
 * This should render when a Smart Link returns a valid response.
 * @see TitleBlock
 */
var TitleBlockResolvedViewOld = function TitleBlockResolvedViewOld(_ref) {
  var actionGroup = _ref.actionGroup,
    _ref$metadata = _ref.metadata,
    metadata = _ref$metadata === void 0 ? [] : _ref$metadata,
    position = _ref.position,
    _ref$subtitle = _ref.subtitle,
    subtitle = _ref$subtitle === void 0 ? [] : _ref$subtitle,
    testId = _ref.testId,
    text = _ref.text,
    icon = _ref.icon,
    title = _ref.title,
    metadataPosition = _ref.metadataPosition,
    hideIcon = _ref.hideIcon,
    blockProps = (0, _objectWithoutProperties2.default)(_ref, _excluded);
  var metadataElements = (0, _utils.renderElementItems)(metadata);
  var subtitleElements = (0, _utils.renderElementItems)(subtitle);
  return /*#__PURE__*/_react.default.createElement(_block.default, (0, _extends2.default)({}, blockProps, {
    testId: "".concat(testId, "-resolved-view")
  }), !hideIcon && /*#__PURE__*/_react.default.createElement(_elements.LinkIcon, {
    overrideIcon: icon,
    position: position
  }), /*#__PURE__*/_react.default.createElement(_elementGroup.default, {
    direction: _constants.SmartLinkDirection.Vertical,
    width: _constants.SmartLinkWidth.Flexible,
    overrideCss: (0, _react2.css)({
      gap: "var(--ds-space-050, 0.25rem)"
    })
  }, title, subtitleElements && /*#__PURE__*/_react.default.createElement(_elementGroup.default, {
    direction: _constants.SmartLinkDirection.Horizontal
  }, subtitleElements)), metadataElements && /*#__PURE__*/_react.default.createElement(_elementGroup.default, {
    direction: _constants.SmartLinkDirection.Horizontal,
    align: _constants.SmartLinkAlignment.Right,
    position: metadataPosition !== null && metadataPosition !== void 0 ? metadataPosition : _constants.SmartLinkPosition.Center
  }, metadataElements), actionGroup);
};
var _default = exports.default = TitleBlockResolvedViewOld;
