// @atlaskit/smart-card/dist/es2019/view/FlexibleCard/components/actions/automation-action/automation-manual-triggers/manual-triggers-modal/sub-components/skeleton/SkeletonOld.js
/**
 * @jsxRuntime classic
 * @jsx jsx
 */
// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766
import { css, jsx, keyframes } from '@emotion/react';
import { N20, N30 } from '@atlaskit/theme/colors';
const AutomationSkeletonContainer = css({
  paddingBlock: "var(--ds-space-025, 2px)"
});
const SkeletonContainer = css({
  marginBlock: "var(--ds-space-100, 8px)",
  marginInline: "var(--ds-space-250, 20px)",
  display: 'flex'
});
const SkeletonIcon = css({
  width: "var(--ds-space-200, 16px)",
  height: "var(--ds-space-200, 16px)",
  display: 'inline-block',
  borderRadius: "var(--ds-space-025, 2px)",
  backgroundColor: 'currentColor',
  border: `${"var(--ds-space-025, 2px)"} solid transparent`,
  opacity: 0.15
});
const MenuIcon = css({
  width: '24px',
  height: '24px',
  display: 'flex',
  justifyContent: 'center',
  alignItems: 'center'
});
const shimmer = keyframes({
  '0%': {
    backgroundPosition: '-300px 0'
  },
  '100%': {
    backgroundPosition: '2000px 0'
  }
});
const SkeletonText = css({
  width: '85%',
  height: "var(--ds-space-200, 16px)",
  marginTop: "var(--ds-space-050, 4px)",
  marginLeft: "var(--ds-space-150, 12px)",
  animationDuration: '1s',
  animationFillMode: 'forwards',
  animationIterationCount: 'infinite',
  animationName: shimmer,
  animationTimingFunction: 'linear',
  backgroundColor: `var(--ds-skeleton, ${N30})`,
  backgroundImage: `linear-gradient(to right, ${`var(--ds-skeleton, ${N30})`} 10%, ${`var(--ds-skeleton-subtle, ${N20})`} 30%, ${`var(--ds-skeleton, ${N30})`} 50%)`,
  backgroundRepeat: 'no-repeat'
});
const SkeletonLine = () => jsx("div", {
  css: SkeletonContainer
}, jsx("div", {
  css: MenuIcon
}, jsx("div", {
  css: SkeletonIcon
})), jsx("div", {
  css: SkeletonText
}));
const getSkeletonLines = lineCount => {
  const skeletonLines = [];
  for (let i = 0; i < lineCount; i++) {
    skeletonLines.push(jsx(SkeletonLine, {
      key: `skeleton-line-${i}`
    }));
  }
  return skeletonLines;
};
export const SkeletonOld = props => {
  const {
    lineCount = 5
  } = props;
  return (
    // eslint-disable-next-line @atlaskit/design-system/use-primitives
    jsx("div", {
      css: AutomationSkeletonContainer
    }, getSkeletonLines(lineCount))
  );
};
