// @atlaskit/renderer/dist/cjs/react/marks/breakout.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = Breakout;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _react = require("@emotion/react");
var _ui = require("@atlaskit/editor-common/ui");
var _utils = require("@atlaskit/editor-common/utils");
var _editorSharedStyles = require("@atlaskit/editor-shared-styles");
var _experiments = require("@atlaskit/tmp-editor-statsig/experiments");
/**
 * @jsxRuntime classic
 * @jsx jsx
 */
// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766

var wrapperStyles = (0, _react.css)({
  // eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values, @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
  margin: "".concat(_editorSharedStyles.blockNodesVerticalMargin, " 0"),
  // eslint-disable-next-line @atlaskit/design-system/use-tokens-space
  marginLeft: '50%',
  transform: 'translateX(-50%)'
});
function Breakout(props) {
  return (0, _react.jsx)(_ui.WidthConsumer, null, function (_ref) {
    var width = _ref.width;
    return (0, _react.jsx)("div", (0, _extends2.default)({
      css: wrapperStyles,
      "data-mode": props.mode
      // Ignored via go/ees005
      // eslint-disable-next-line react/jsx-props-no-spreading
    }, (0, _experiments.editorExperiment)('advanced_layouts', true) && {
      'data-has-width': !!props.width,
      'data-width': props.width
    }, {
      style: {
        width: (0, _experiments.editorExperiment)('advanced_layouts', true) ?
        // eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values -- Ignored via go/DSP-18766
        (0, _utils.calcBreakoutWithCustomWidth)(props.mode, 'width' in props ? props.width : null, width) :
        // eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values -- Ignored via go/DSP-18766
        (0, _utils.calcBreakoutWidth)(props.mode, width)
      }
      // eslint-disable-next-line @atlaskit/ui-styling-standard/no-classname-prop -- Ignored via go/DSP-18766
      ,
      className: "fabric-editor-breakout-mark fabric-editor-block-mark"
    }), props.children);
  });
}
