// @atlaskit/renderer/dist/cjs/react/marks/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
var _typeof = require("@babel/runtime/helpers/typeof");
Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "Annotation", {
  enumerable: true,
  get: function get() {
    return _annotation.default;
  }
});
Object.defineProperty(exports, "BackgroundColor", {
  enumerable: true,
  get: function get() {
    return _backgroundColor.default;
  }
});
Object.defineProperty(exports, "Border", {
  enumerable: true,
  get: function get() {
    return _border.default;
  }
});
Object.defineProperty(exports, "Breakout", {
  enumerable: true,
  get: function get() {
    return _breakout.default;
  }
});
Object.defineProperty(exports, "Code", {
  enumerable: true,
  get: function get() {
    return _code.default;
  }
});
Object.defineProperty(exports, "DataConsumer", {
  enumerable: true,
  get: function get() {
    return _dataConsumer.default;
  }
});
Object.defineProperty(exports, "Em", {
  enumerable: true,
  get: function get() {
    return _em.default;
  }
});
Object.defineProperty(exports, "FragmentMark", {
  enumerable: true,
  get: function get() {
    return _fragment.default;
  }
});
Object.defineProperty(exports, "Link", {
  enumerable: true,
  get: function get() {
    return _link.default;
  }
});
Object.defineProperty(exports, "Strike", {
  enumerable: true,
  get: function get() {
    return _strike.default;
  }
});
Object.defineProperty(exports, "Strong", {
  enumerable: true,
  get: function get() {
    return _strong.default;
  }
});
Object.defineProperty(exports, "Subsup", {
  enumerable: true,
  get: function get() {
    return _subsup.default;
  }
});
Object.defineProperty(exports, "TextColor", {
  enumerable: true,
  get: function get() {
    return _textColor.default;
  }
});
Object.defineProperty(exports, "Underline", {
  enumerable: true,
  get: function get() {
    return _underline.default;
  }
});
Object.defineProperty(exports, "UnsupportedMark", {
  enumerable: true,
  get: function get() {
    return _unsupportedMark.default;
  }
});
Object.defineProperty(exports, "UnsupportedNodeAttribute", {
  enumerable: true,
  get: function get() {
    return _unsupportedNodeAttribute.default;
  }
});
Object.defineProperty(exports, "isAnnotationMark", {
  enumerable: true,
  get: function get() {
    return _annotation.isAnnotationMark;
  }
});
exports.toReact = exports.markToReact = void 0;
var _code = _interopRequireDefault(require("./code"));
var _em = _interopRequireDefault(require("./em"));
var _link = _interopRequireDefault(require("./link"));
var _strike = _interopRequireDefault(require("./strike"));
var _strong = _interopRequireDefault(require("./strong"));
var _subsup = _interopRequireDefault(require("./subsup"));
var _textColor = _interopRequireDefault(require("./textColor"));
var _backgroundColor = _interopRequireDefault(require("./backgroundColor"));
var _underline = _interopRequireDefault(require("./underline"));
var _breakout = _interopRequireDefault(require("./breakout"));
var _alignment = _interopRequireDefault(require("./alignment"));
var _indentation = _interopRequireDefault(require("./indentation"));
var _unsupportedMark = _interopRequireDefault(require("./unsupportedMark"));
var _unsupportedNodeAttribute = _interopRequireDefault(require("./unsupportedNodeAttribute"));
var _dataConsumer = _interopRequireDefault(require("./data-consumer"));
var _fragment = _interopRequireDefault(require("./fragment"));
var _annotation = _interopRequireWildcard(require("./annotation"));
var _border = _interopRequireDefault(require("./border"));
var _confluenceInlineComment = _interopRequireDefault(require("./confluence-inline-comment"));
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
/* eslint-disable @atlaskit/editor/no-re-export */
// Mapping file

// Stage0

var markToReact = exports.markToReact = {
  code: _code.default,
  em: _em.default,
  link: _link.default,
  strike: _strike.default,
  strong: _strong.default,
  subsup: _subsup.default,
  textColor: _textColor.default,
  backgroundColor: _backgroundColor.default,
  underline: _underline.default,
  annotation: _annotation.default,
  border: _border.default,
  // Stage0
  confluenceInlineComment: _confluenceInlineComment.default,
  breakout: _breakout.default,
  alignment: _alignment.default,
  indentation: _indentation.default,
  unsupportedMark: _unsupportedMark.default,
  unsupportedNodeAttribute: _unsupportedNodeAttribute.default,
  dataConsumer: _dataConsumer.default,
  fragment: _fragment.default
};

// Ignored via go/ees005
// eslint-disable-next-line @typescript-eslint/no-explicit-any
var toReact = exports.toReact = function toReact(mark) {
  return markToReact[mark.type.name];
};
