// @atlaskit/smart-card/dist/cjs/view/InlineCard/ErroredView/InlineCardErroredViewOld.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.InlineCardErroredViewOld = void 0;
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));
var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));
var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _react = _interopRequireDefault(require("react"));
var _reactIntlNext = require("react-intl-next");
var _button = _interopRequireDefault(require("@atlaskit/button"));
var _error = _interopRequireDefault(require("@atlaskit/icon/utility/migration/error"));
var _platformFeatureFlags = require("@atlaskit/platform-feature-flags");
var _primitives = require("@atlaskit/primitives");
var _colors = require("@atlaskit/theme/colors");
var _messages = require("../../../messages");
var _HoverCard = require("../../HoverCard");
var _Frame = require("../Frame");
var _Icon = require("../Icon");
var _IconEmotion = require("../Icon-emotion");
var _IconAndTitleLayout = require("../IconAndTitleLayout");
var _styled = require("../styled");
var _styledEmotion = require("../styled-emotion");
var _withFrameStyleControl = _interopRequireDefault(require("../utils/withFrameStyleControl"));
function _callSuper(t, o, e) { return o = (0, _getPrototypeOf2.default)(o), (0, _possibleConstructorReturn2.default)(t, _isNativeReflectConstruct() ? Reflect.construct(o, e || [], (0, _getPrototypeOf2.default)(t).constructor) : o.apply(t, e)); }
function _isNativeReflectConstruct() { try { var t = !Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); } catch (t) {} return (_isNativeReflectConstruct = function _isNativeReflectConstruct() { return !!t; })(); }
var iconWrapperStyles = (0, _primitives.xcss)({
  marginRight: 'space.negative.025'
});
var InlineCardErroredViewOld = exports.InlineCardErroredViewOld = /*#__PURE__*/function (_React$Component) {
  function InlineCardErroredViewOld() {
    var _this;
    (0, _classCallCheck2.default)(this, InlineCardErroredViewOld);
    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }
    _this = _callSuper(this, InlineCardErroredViewOld, [].concat(args));
    (0, _defineProperty2.default)(_this, "frameRef", /*#__PURE__*/_react.default.createRef());
    (0, _defineProperty2.default)(_this, "handleRetry", function (event) {
      var onRetry = _this.props.onRetry;
      if (onRetry) {
        event.preventDefault();
        event.stopPropagation();
        onRetry();
      }
    });
    (0, _defineProperty2.default)(_this, "renderActionButton", function () {
      var onRetry = _this.props.onRetry;
      var Button = (0, _withFrameStyleControl.default)(_button.default, _this.frameRef);
      return onRetry && /*#__PURE__*/_react.default.createElement(Button, {
        spacing: "none",
        component: (0, _platformFeatureFlags.fg)('bandicoots-compiled-migration-smartcard') ? _styled.IconStyledButtonOldVisualRefresh : _styledEmotion.IconStyledButton,
        onClick: _this.handleRetry,
        role: "button"
      }, /*#__PURE__*/_react.default.createElement(_reactIntlNext.FormattedMessage, _messages.messages.try_again));
    });
    return _this;
  }
  (0, _inherits2.default)(InlineCardErroredViewOld, _React$Component);
  return (0, _createClass2.default)(InlineCardErroredViewOld, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
        url = _this$props.url,
        onClick = _this$props.onClick,
        isSelected = _this$props.isSelected,
        _this$props$testId = _this$props.testId,
        testId = _this$props$testId === void 0 ? 'inline-card-errored-view' : _this$props$testId,
        icon = _this$props.icon,
        message = _this$props.message,
        truncateInline = _this$props.truncateInline;
      var Wrapper = (0, _platformFeatureFlags.fg)('bandicoots-compiled-migration-smartcard') ? _Icon.AKIconWrapper : _IconEmotion.AKIconWrapper;
      var content = /*#__PURE__*/_react.default.createElement(_Frame.Frame, {
        testId: testId,
        isSelected: isSelected,
        ref: this.frameRef,
        truncateInline: truncateInline
      }, /*#__PURE__*/_react.default.createElement(_IconAndTitleLayout.IconAndTitleLayout, {
        icon: icon || ((0, _platformFeatureFlags.fg)('platform-smart-card-icon-migration') ? /*#__PURE__*/_react.default.createElement(_primitives.Box, {
          as: "span",
          xcss: iconWrapperStyles
        }, /*#__PURE__*/_react.default.createElement(_error.default, {
          color: "var(--ds-icon-danger, #C9372C)",
          label: "error",
          LEGACY_size: "small",
          testId: "errored-view-default-icon"
        })) : /*#__PURE__*/_react.default.createElement(Wrapper, null, /*#__PURE__*/_react.default.createElement(_error.default, {
          label: "error",
          LEGACY_size: "small",
          color: "var(--ds-icon-danger, ".concat(_colors.R300, ")"),
          testId: "errored-view-default-icon"
        }))),
        link: url,
        title: url,
        onClick: onClick,
        rightSide: message
      }), this.renderActionButton());
      if (this.props.showHoverPreview) {
        return /*#__PURE__*/_react.default.createElement(_HoverCard.HoverCard, {
          url: url
        }, content);
      }
      return content;
    }
  }]);
}(_react.default.Component);
