// @atlaskit/renderer/dist/cjs/react/nodes/mediaGroup.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
var _typeof = require("@babel/runtime/helpers/typeof");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));
var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));
var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _react = _interopRequireWildcard(require("react"));
var _mediaCard = require("@atlaskit/media-card");
var _mediaFilmstrip = require("@atlaskit/media-filmstrip");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { (0, _defineProperty2.default)(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _callSuper(t, o, e) { return o = (0, _getPrototypeOf2.default)(o), (0, _possibleConstructorReturn2.default)(t, _isNativeReflectConstruct() ? Reflect.construct(o, e || [], (0, _getPrototypeOf2.default)(t).constructor) : o.apply(t, e)); }
function _isNativeReflectConstruct() { try { var t = !Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); } catch (t) {} return (_isNativeReflectConstruct = function _isNativeReflectConstruct() { return !!t; })(); }
// Ignored via go/ees005
// eslint-disable-next-line @repo/internal/react/no-class-components
var MediaGroup = exports.default = /*#__PURE__*/function (_PureComponent) {
  function MediaGroup() {
    var _this;
    (0, _classCallCheck2.default)(this, MediaGroup);
    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }
    _this = _callSuper(this, MediaGroup, [].concat(args));
    (0, _defineProperty2.default)(_this, "state", {
      animate: false,
      offset: 0
    });
    (0, _defineProperty2.default)(_this, "handleSize", function (_ref) {
      var offset = _ref.offset;
      return _this.setState({
        offset: offset
      });
    });
    (0, _defineProperty2.default)(_this, "handleScroll", function (_ref2) {
      var animate = _ref2.animate,
        offset = _ref2.offset;
      return _this.setState({
        animate: animate,
        offset: offset
      });
    });
    (0, _defineProperty2.default)(_this, "onMediaClick", function (cardClickHandler, child, surroundingItems) {
      return (
        // Ignored via go/ees005
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
        function (event, analyticsEvent) {
          var surroundings = {
            // Ignored via go/ees005
            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
            collectionName: child.props.collection,
            list: surroundingItems
          };
          cardClickHandler(event, surroundings, analyticsEvent);
        }
      );
    });
    return _this;
  }
  (0, _inherits2.default)(MediaGroup, _PureComponent);
  return (0, _createClass2.default)(MediaGroup, [{
    key: "render",
    value: function render() {
      var numChildren = _react.default.Children.count(this.props.children);
      var content;
      if (numChildren === 1) {
        // Ignored via go/ees005
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
        var card = _react.default.Children.toArray(this.props.children)[0];
        switch (card.props.type) {
          case 'file':
            content = this.renderSingleFile(card);
            break;
          case 'link':
            content = null;
            break;
          default:
            content = this.renderSingleLink(card);
        }
      } else {
        content = this.renderStrip();
      }
      // eslint-disable-next-line @atlaskit/ui-styling-standard/no-classname-prop -- Ignored via go/DSP-18766
      return /*#__PURE__*/_react.default.createElement("div", {
        className: "MediaGroup"
      }, content);
    }
  }, {
    key: "renderSingleFile",
    value: function renderSingleFile(child) {
      return /*#__PURE__*/_react.default.cloneElement(child, {
        resizeMode: 'stretchy-fit',
        cardDimensions: _mediaCard.defaultImageCardDimensions,
        useInlinePlayer: false,
        featureFlags: this.props.featureFlags,
        enableDownloadButton: this.props.enableDownloadButton
      });
    }
  }, {
    key: "renderSingleLink",
    value: function renderSingleLink(child) {
      return /*#__PURE__*/_react.default.cloneElement(child, {
        appearance: 'auto',
        featureFlags: this.props.featureFlags
      });
    }
  }, {
    key: "cloneFileCard",
    value: function cloneFileCard(child, surroundingItems) {
      var cardClickHandler = this.props && this.props.eventHandlers && this.props.eventHandlers.media && this.props.eventHandlers.media.onClick;
      var onClick = cardClickHandler ? this.onMediaClick(cardClickHandler, child, surroundingItems) : undefined;
      return /*#__PURE__*/_react.default.cloneElement(child, {
        useInlinePlayer: false,
        eventHandlers: _objectSpread(_objectSpread({}, child.props.eventHandlers), {}, {
          media: {
            onClick: onClick
          }
        }),
        featureFlags: this.props.featureFlags,
        enableDownloadButton: this.props.enableDownloadButton
      });
    }
  }, {
    key: "renderStrip",
    value: function renderStrip() {
      var _this2 = this;
      var children = this.props.children;
      var _this$state = this.state,
        animate = _this$state.animate,
        offset = _this$state.offset;
      var childIdentifiers = _react.default.Children.map(children, function (child) {
        if ( /*#__PURE__*/_react.default.isValidElement(child)) {
          return _this2.mapMediaPropsToIdentifier(child.props);
        }
      });
      var surroundingItems = (childIdentifiers || []).filter(function (identifier) {
        return !!identifier;
      });
      return /*#__PURE__*/_react.default.createElement(_mediaFilmstrip.FilmstripView, {
        animate: animate,
        offset: offset,
        onSize: this.handleSize,
        onScroll: this.handleScroll
      }, _react.default.Children.map(children, function (rawChild) {
        var child = rawChild;
        switch (child.props.type) {
          case 'file':
            return _this2.cloneFileCard(child, surroundingItems);
          case 'link':
            return null;
          default:
            return /*#__PURE__*/_react.default.cloneElement(child, {
              featureFlags: _this2.props.featureFlags,
              enableDownloadButton: _this2.props.enableDownloadButton
            });
        }
      }));
    }
  }, {
    key: "mapMediaPropsToIdentifier",
    value: function mapMediaPropsToIdentifier(_ref3) {
      var id = _ref3.id,
        type = _ref3.type,
        occurrenceKey = _ref3.occurrenceKey,
        collection = _ref3.collection;
      switch (type) {
        case 'file':
          return {
            // Ignored via go/ees005
            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
            id: id,
            mediaItemType: type,
            occurrenceKey: occurrenceKey,
            collectionName: collection
          };
        case 'link':
          return undefined;
        case 'external':
          return {
            // Ignored via go/ees005
            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
            id: id,
            mediaItemType: 'file',
            occurrenceKey: occurrenceKey,
            collectionName: collection
          };
      }
    }
  }]);
}(_react.PureComponent);
