// @atlaskit/renderer/dist/cjs/react/nodes/mention.js
"use strict";

var _typeof = require("@babel/runtime/helpers/typeof");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = _interopRequireWildcard(require("react"));
var _mention = require("@atlaskit/editor-common/mention");
var _platformFeatureFlags = require("@atlaskit/platform-feature-flags");
var _useInlineAnnotationProps = require("../../ui/annotations/element/useInlineAnnotationProps");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
var _default = exports.default = /*#__PURE__*/(0, _react.memo)(function MentionItem(props) {
  var eventHandlers = props.eventHandlers,
    id = props.id,
    providers = props.providers,
    text = props.text,
    accessLevel = props.accessLevel,
    localId = props.localId;
  var inlineAnnotationProps = (0, _useInlineAnnotationProps.useInlineAnnotationProps)(props);
  if ((0, _platformFeatureFlags.fg)('editor_inline_comments_on_inline_nodes')) {
    return (
      /*#__PURE__*/
      // Ignored via go/ees005
      // eslint-disable-next-line react/jsx-props-no-spreading
      _react.default.createElement("span", inlineAnnotationProps, /*#__PURE__*/_react.default.createElement(_mention.Mention, {
        id: id,
        text: text,
        accessLevel: accessLevel,
        providers: providers,
        localId: localId,
        eventHandlers: eventHandlers && eventHandlers.mention
      }))
    );
  }
  return /*#__PURE__*/_react.default.createElement(_mention.Mention, {
    id: id,
    text: text,
    accessLevel: accessLevel,
    providers: providers,
    localId: localId,
    eventHandlers: eventHandlers && eventHandlers.mention
  });
});
