// @atlaskit/renderer/dist/cjs/react/nodes/taskItem.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
var _typeof = require("@babel/runtime/helpers/typeof");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));
var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));
var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _react = _interopRequireWildcard(require("react"));
var _providerFactory = require("@atlaskit/editor-common/provider-factory");
var _taskItemWithProviders = _interopRequireDefault(require("./task-item-with-providers"));
var _analyticsNamespacedContext = require("@atlaskit/analytics-namespaced-context");
var _TaskItemsFormatContext = require("../../ui/TaskItemsFormatContext/TaskItemsFormatContext");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _callSuper(t, o, e) { return o = (0, _getPrototypeOf2.default)(o), (0, _possibleConstructorReturn2.default)(t, _isNativeReflectConstruct() ? Reflect.construct(o, e || [], (0, _getPrototypeOf2.default)(t).constructor) : o.apply(t, e)); }
function _isNativeReflectConstruct() { try { var t = !Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); } catch (t) {} return (_isNativeReflectConstruct = function _isNativeReflectConstruct() { return !!t; })(); }
// Ignored via go/ees005
// eslint-disable-next-line @repo/internal/react/no-class-components
var TaskItem = exports.default = /*#__PURE__*/function (_PureComponent) {
  function TaskItem(props) {
    var _this;
    (0, _classCallCheck2.default)(this, TaskItem);
    _this = _callSuper(this, TaskItem, [props]);
    // Ignored via go/ees005
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    (0, _defineProperty2.default)(_this, "renderWithProvider", function (providers) {
      var taskDecisionProvider = providers.taskDecisionProvider,
        contextIdentifierProvider = providers.contextIdentifierProvider;
      var _this$props = _this.props,
        children = _this$props.children,
        localId = _this$props.localId,
        state = _this$props.state,
        rendererContext = _this$props.rendererContext,
        disabled = _this$props.disabled,
        dataAttributes = _this$props.dataAttributes;
      var objectAri = '';
      if (rendererContext) {
        objectAri = rendererContext.objectAri || '';
      }
      return /*#__PURE__*/_react.default.createElement(_analyticsNamespacedContext.FabricElementsAnalyticsContext, {
        data: {
          userContext: 'document'
        }
      }, /*#__PURE__*/_react.default.createElement(_TaskItemsFormatContext.TaskItemsFormatProvider, null, /*#__PURE__*/_react.default.createElement(_TaskItemsFormatContext.TaskItemsFormatConsumer, null, function (_ref) {
        var _ref2 = (0, _slicedToArray2.default)(_ref, 2),
          dispatch = _ref2[1];
        return /*#__PURE__*/_react.default.createElement(_taskItemWithProviders.default, {
          objectAri: objectAri,
          taskId: localId,
          isDone: state === 'DONE',
          isRenderer: true,
          disabled: disabled,
          taskDecisionProvider: taskDecisionProvider,
          contextIdentifierProvider: contextIdentifierProvider,
          dataAttributes: dataAttributes,
          onChange: function onChange(_, isChecked) {
            dispatch(isChecked);
          }
        }, children);
      })));
    });
    _this.providerFactory = props.providers || new _providerFactory.ProviderFactory();
    return _this;
  }
  (0, _inherits2.default)(TaskItem, _PureComponent);
  return (0, _createClass2.default)(TaskItem, [{
    key: "componentWillUnmount",
    value: function componentWillUnmount() {
      if (!this.props.providers) {
        // new ProviderFactory is created if no `providers` has been set
        // in this case when component is unmounted it's safe to destroy this providerFactory
        this.providerFactory.destroy();
      }
    }
  }, {
    key: "render",
    value: function render() {
      return /*#__PURE__*/_react.default.createElement(_providerFactory.WithProviders, {
        providers: ['taskDecisionProvider', 'contextIdentifierProvider'],
        providerFactory: this.providerFactory,
        renderNode: this.renderWithProvider
      });
    }
  }]);
}(_react.PureComponent);
