// @atlaskit/smart-card/dist/cjs/view/HoverCard/components/views/resolving/styled.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.titleBlockStyles = exports.skeletonContainer = exports.loadingViewContainer = exports.getTitleStyles = void 0;
var _react = require("@emotion/react");
// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766

/**
 * @deprecated remove on FF clean up bandicoots-compiled-migration-smartcard
 */
// eslint-disable-next-line @atlaskit/ui-styling-standard/no-exported-styles -- Ignored via go/DSP-18766
var loadingViewContainer = exports.loadingViewContainer = (0, _react.css)({
  display: 'flex',
  flexDirection: 'column',
  padding: "var(--ds-space-200, 1rem)"
});

/**
 * @deprecated remove on FF clean up bandicoots-compiled-migration-smartcard
 */
// eslint-disable-next-line @atlaskit/ui-styling-standard/no-exported-styles -- Ignored via go/DSP-18766
var skeletonContainer = exports.skeletonContainer = (0, _react.css)({
  display: 'flex',
  flexDirection: 'column',
  // eslint-disable-next-line @atlaskit/design-system/use-tokens-space -- needs manual remediation
  gap: '0.625rem',
  alignItems: 'center'
});

/**
 * @deprecated remove on FF clean up bandicoots-compiled-migration-smartcard
 */
var getTitleStyles = exports.getTitleStyles = function getTitleStyles(height) {
  return (0, _react.css)({
    flex: '1 0 auto',
    // eslint-disable-next-line @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
    height: "".concat(height, "rem"),
    // eslint-disable-next-line @atlaskit/ui-styling-standard/no-nested-selectors -- Ignored via go/DSP-18766
    span: {
      width: '100%'
    }
  });
};

/**
 * @deprecated remove on FF clean up bandicoots-compiled-migration-smartcard
 */
// eslint-disable-next-line @atlaskit/ui-styling-standard/no-exported-styles -- Ignored via go/DSP-18766
var titleBlockStyles = exports.titleBlockStyles = (0, _react.css)({
  width: '100%',
  gap: "var(--ds-space-100, 0.5rem)"
});
