// @atlaskit/smart-card/dist/cjs/view/FlexibleCard/components/elements/snippet-element/SnippetElementOld.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));
var _react = require("react");
var _react2 = require("@emotion/react");
var _flexibleUiContext = require("../../../../../state/flexible-ui-context");
var _text = _interopRequireDefault(require("../text"));
var _excluded = ["content", "maxLines", "overrideCss"];
/**
 * @jsxRuntime classic
 * @jsx jsx
 */
// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766
var SNIPPET_DEFAULT_MAX_LINES = 3;
var getSnippetStyles = function getSnippetStyles(overrideCss) {
  return (0, _react2.css)({
    color: "var(--ds-text, #172B4D)",
    WebkitUserSelect: 'text',
    MozUserSelect: 'text',
    MsUserSelect: 'text',
    userSelect: 'text'
  },
  // eslint-disable-next-line @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
  overrideCss);
};
var SnippetElementOld = function SnippetElementOld() {
  var _ref = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  var overrideContent = _ref.content,
    _ref$maxLines = _ref.maxLines,
    maxLines = _ref$maxLines === void 0 ? SNIPPET_DEFAULT_MAX_LINES : _ref$maxLines,
    overrideCss = _ref.overrideCss,
    props = (0, _objectWithoutProperties2.default)(_ref, _excluded);
  var context = (0, _react.useContext)(_flexibleUiContext.FlexibleUiContext);
  return (0, _react2.jsx)(_text.default, (0, _extends2.default)({
    content: overrideContent !== null && overrideContent !== void 0 ? overrideContent : context === null || context === void 0 ? void 0 : context.snippet,
    maxLines: maxLines,
    overrideCss: getSnippetStyles(overrideCss)
  }, props));
};
var _default = exports.default = SnippetElementOld;
