// @atlaskit/smart-card/dist/cjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "ActionName", {
  enumerable: true,
  get: function get() {
    return _constants.ActionName;
  }
});
Object.defineProperty(exports, "Card", {
  enumerable: true,
  get: function get() {
    return _Card.Card;
  }
});
Object.defineProperty(exports, "CardAction", {
  enumerable: true,
  get: function get() {
    return _types.CardAction;
  }
});
Object.defineProperty(exports, "CustomBlock", {
  enumerable: true,
  get: function get() {
    return _blocks.CustomBlock;
  }
});
Object.defineProperty(exports, "EditorCardProvider", {
  enumerable: true,
  get: function get() {
    return _linkProvider.EditorCardProvider;
  }
});
Object.defineProperty(exports, "ElementName", {
  enumerable: true,
  get: function get() {
    return _constants.ElementName;
  }
});
Object.defineProperty(exports, "EmbedResizeMessageListener", {
  enumerable: true,
  get: function get() {
    return _EmbedResizeMessageListener.EmbedResizeMessageListener;
  }
});
Object.defineProperty(exports, "ExpandedFrame", {
  enumerable: true,
  get: function get() {
    return _ExpandedFrame.ExpandedFrame;
  }
});
Object.defineProperty(exports, "FooterBlock", {
  enumerable: true,
  get: function get() {
    return _blocks.FooterBlock;
  }
});
Object.defineProperty(exports, "MediaPlacement", {
  enumerable: true,
  get: function get() {
    return _constants.MediaPlacement;
  }
});
Object.defineProperty(exports, "MetadataBlock", {
  enumerable: true,
  get: function get() {
    return _blocks.MetadataBlock;
  }
});
Object.defineProperty(exports, "PreviewBlock", {
  enumerable: true,
  get: function get() {
    return _blocks.PreviewBlock;
  }
});
Object.defineProperty(exports, "SmartLinkDirection", {
  enumerable: true,
  get: function get() {
    return _constants.SmartLinkDirection;
  }
});
Object.defineProperty(exports, "SmartLinkEvents", {
  enumerable: true,
  get: function get() {
    return _analytics.SmartLinkEvents;
  }
});
Object.defineProperty(exports, "SmartLinkPosition", {
  enumerable: true,
  get: function get() {
    return _constants.SmartLinkPosition;
  }
});
Object.defineProperty(exports, "SmartLinkSize", {
  enumerable: true,
  get: function get() {
    return _constants.SmartLinkSize;
  }
});
Object.defineProperty(exports, "SmartLinkTheme", {
  enumerable: true,
  get: function get() {
    return _constants.SmartLinkTheme;
  }
});
Object.defineProperty(exports, "SnippetBlock", {
  enumerable: true,
  get: function get() {
    return _blocks.SnippetBlock;
  }
});
Object.defineProperty(exports, "TitleBlock", {
  enumerable: true,
  get: function get() {
    return _blocks.TitleBlock;
  }
});
Object.defineProperty(exports, "blockCardContentClassName", {
  enumerable: true,
  get: function get() {
    return _classNames.blockCardContentClassName;
  }
});
Object.defineProperty(exports, "blockCardContentHeaderClassName", {
  enumerable: true,
  get: function get() {
    return _classNames.blockCardContentHeaderClassName;
  }
});
Object.defineProperty(exports, "blockCardContentHeaderNameClassName", {
  enumerable: true,
  get: function get() {
    return _classNames.blockCardContentHeaderNameClassName;
  }
});
Object.defineProperty(exports, "blockCardErroredViewClassName", {
  enumerable: true,
  get: function get() {
    return _classNames.blockCardErroredViewClassName;
  }
});
Object.defineProperty(exports, "blockCardForbiddenViewClassName", {
  enumerable: true,
  get: function get() {
    return _classNames.blockCardForbiddenViewClassName;
  }
});
Object.defineProperty(exports, "blockCardForbiddenViewLinkClassName", {
  enumerable: true,
  get: function get() {
    return _classNames.blockCardForbiddenViewLinkClassName;
  }
});
Object.defineProperty(exports, "blockCardIconImageClassName", {
  enumerable: true,
  get: function get() {
    return _classNames.blockCardIconImageClassName;
  }
});
Object.defineProperty(exports, "blockCardNotFoundViewClassName", {
  enumerable: true,
  get: function get() {
    return _classNames.blockCardNotFoundViewClassName;
  }
});
Object.defineProperty(exports, "blockCardResolvedViewByClassName", {
  enumerable: true,
  get: function get() {
    return _classNames.blockCardResolvedViewByClassName;
  }
});
Object.defineProperty(exports, "blockCardResolvedViewClassName", {
  enumerable: true,
  get: function get() {
    return _classNames.blockCardResolvedViewClassName;
  }
});
Object.defineProperty(exports, "blockCardResolvingViewClassName", {
  enumerable: true,
  get: function get() {
    return _classNames.blockCardResolvingViewClassName;
  }
});
Object.defineProperty(exports, "contentFooterClassName", {
  enumerable: true,
  get: function get() {
    return _classNames.contentFooterClassName;
  }
});
Object.defineProperty(exports, "editorCardProvider", {
  enumerable: true,
  get: function get() {
    return _linkProvider.editorCardProvider;
  }
});
Object.defineProperty(exports, "embedHeaderHeight", {
  enumerable: true,
  get: function get() {
    return _styled.embedHeaderHeight;
  }
});
Object.defineProperty(exports, "loadingPlaceholderClassName", {
  enumerable: true,
  get: function get() {
    return _LazyFallback.loadingPlaceholderClassName;
  }
});
Object.defineProperty(exports, "metadataListClassName", {
  enumerable: true,
  get: function get() {
    return _classNames.metadataListClassName;
  }
});
Object.defineProperty(exports, "useSmartLinkEvents", {
  enumerable: true,
  get: function get() {
    return _useSmartLinkEvents.useSmartLinkEvents;
  }
});
var _Card = require("./view/Card");
var _types = require("./view/Card/types");
var _EmbedResizeMessageListener = require("./view/EmbedCard/EmbedResizeMessageListener");
var _ExpandedFrame = require("./view/EmbedCard/components/ExpandedFrame");
var _styled = require("./view/EmbedCard/components/styled");
var _analytics = require("./utils/analytics/analytics");
var _useSmartLinkEvents = require("./view/SmartLinkEvents/useSmartLinkEvents");
var _classNames = require("./classNames");
var _LazyFallback = require("./view/CardWithUrl/component-lazy/LazyFallback");
var _constants = require("./constants");
var _blocks = require("./view/FlexibleCard/components/blocks");
var _linkProvider = require("@atlaskit/link-provider");
