// @atlaskit/link-datasource/dist/es2019/ui/issue-like-table/shared-components/avatar/avatar-old.js
import React from 'react';
import { FormattedMessage, useIntl } from 'react-intl-next';
import Avatar from '@atlaskit/avatar';
import { Box, xcss } from '@atlaskit/primitives';
import Tooltip from '@atlaskit/tooltip';
import { USER_TYPE_TEST_ID } from '../../render-type/user';
import { userTypeMessages } from '../../render-type/user/messages';
const userWrapperStyles = xcss({
  display: 'flex',
  alignItems: 'center',
  font: 'font.body'
});
const avatarWrapperStyles = xcss({
  marginRight: 'space.100'
});
const emptyAvatarWrapperStyles = xcss({
  paddingBlock: 'space.050',
  paddingInline: 'space.100',
  opacity: 0,
  ':hover': {
    opacity: 1
  }
});
export const EmptyAvatarOld = () => {
  const intl = useIntl();
  return /*#__PURE__*/React.createElement(Tooltip, {
    content: intl.formatMessage(userTypeMessages.userDefaultdisplayNameValue)
  }, /*#__PURE__*/React.createElement(Box, {
    xcss: [userWrapperStyles, emptyAvatarWrapperStyles],
    testId: USER_TYPE_TEST_ID
  }, /*#__PURE__*/React.createElement(Box, {
    xcss: avatarWrapperStyles
  }, /*#__PURE__*/React.createElement(Avatar, {
    appearance: "circle",
    size: 'small',
    testId: `${USER_TYPE_TEST_ID}--avatar`
  })), /*#__PURE__*/React.createElement(FormattedMessage, userTypeMessages.userDefaultdisplayNameValue)));
};
