// @atlaskit/smart-card/dist/cjs/view/InlineCard/ResolvingView/InlineCardResolvingViewOld.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.InlineCardResolvingViewOld = void 0;
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));
var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));
var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));
var _react = _interopRequireDefault(require("react"));
var _platformFeatureFlags = require("@atlaskit/platform-feature-flags");
var _spinner = _interopRequireDefault(require("@atlaskit/spinner"));
var _Frame = require("../Frame");
var _IconAndTitleLayout = require("../IconAndTitleLayout");
var _styled = require("../IconAndTitleLayout/styled");
var _styledEmotion = require("../IconAndTitleLayout/styled-emotion");
var _styled2 = require("./styled");
var _styledEmotion2 = require("./styled-emotion");
function _callSuper(t, o, e) { return o = (0, _getPrototypeOf2.default)(o), (0, _possibleConstructorReturn2.default)(t, _isNativeReflectConstruct() ? Reflect.construct(o, e || [], (0, _getPrototypeOf2.default)(t).constructor) : o.apply(t, e)); }
function _isNativeReflectConstruct() { try { var t = !Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); } catch (t) {} return (_isNativeReflectConstruct = function _isNativeReflectConstruct() { return !!t; })(); }
var InlineCardResolvingViewOld = exports.InlineCardResolvingViewOld = /*#__PURE__*/function (_React$Component) {
  function InlineCardResolvingViewOld() {
    (0, _classCallCheck2.default)(this, InlineCardResolvingViewOld);
    return _callSuper(this, InlineCardResolvingViewOld, arguments);
  }
  (0, _inherits2.default)(InlineCardResolvingViewOld, _React$Component);
  return (0, _createClass2.default)(InlineCardResolvingViewOld, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
        url = _this$props.url,
        onClick = _this$props.onClick,
        isSelected = _this$props.isSelected,
        inlinePreloaderStyle = _this$props.inlinePreloaderStyle,
        _this$props$testId = _this$props.testId,
        testId = _this$props$testId === void 0 ? 'inline-card-resolving-view' : _this$props$testId,
        titleTextColor = _this$props.titleTextColor,
        resolvingPlaceholder = _this$props.resolvingPlaceholder,
        truncateInline = _this$props.truncateInline;
      var SpinnerWrapperComponent = (0, _platformFeatureFlags.fg)('bandicoots-compiled-migration-smartcard') ? _styled2.SpinnerWrapperOldVisualRefresh : _styledEmotion2.SpinnerWrapper;
      if (inlinePreloaderStyle === 'on-right-without-skeleton') {
        if ((0, _platformFeatureFlags.fg)('bandicoots-compiled-migration-smartcard')) {
          return /*#__PURE__*/_react.default.createElement(_Frame.Frame, {
            withoutBackground: true,
            testId: testId,
            onClick: onClick,
            isSelected: isSelected,
            truncateInline: truncateInline
          }, /*#__PURE__*/_react.default.createElement(_IconAndTitleLayout.IconTitleWrapper, null, url, /*#__PURE__*/_react.default.createElement(_styled.RightIconPositionWrapper, null, (0, _platformFeatureFlags.fg)('platform-linking-visual-refresh-v1') ? /*#__PURE__*/_react.default.createElement(_spinner.default, {
            size: 14
          }) : /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(SpinnerWrapperComponent, {
            className: "inline-resolving-spinner"
          }, /*#__PURE__*/_react.default.createElement(_spinner.default, {
            size: 14
          }))))));
        } else {
          return /*#__PURE__*/_react.default.createElement(_Frame.Frame, {
            withoutBackground: true,
            testId: testId,
            onClick: onClick,
            isSelected: isSelected,
            truncateInline: truncateInline
          }, /*#__PURE__*/_react.default.createElement(_styledEmotion.IconTitleWrapper, null, url, /*#__PURE__*/_react.default.createElement(_styledEmotion.RightIconPositionWrapper, null, /*#__PURE__*/_react.default.createElement(SpinnerWrapperComponent, {
            className: "inline-resolving-spinner"
          }, /*#__PURE__*/_react.default.createElement(_spinner.default, {
            size: 14
          })))));
        }
      } else {
        return /*#__PURE__*/_react.default.createElement(_Frame.Frame, {
          testId: testId,
          onClick: onClick,
          isSelected: isSelected,
          link: url,
          truncateInline: truncateInline
        }, /*#__PURE__*/_react.default.createElement(_IconAndTitleLayout.IconAndTitleLayout, {
          title: resolvingPlaceholder !== null && resolvingPlaceholder !== void 0 ? resolvingPlaceholder : url,
          titleTextColor: titleTextColor
        }, (0, _platformFeatureFlags.fg)('platform-linking-visual-refresh-v1') ? /*#__PURE__*/_react.default.createElement(_spinner.default, {
          size: 14
        }) : /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(SpinnerWrapperComponent, {
          className: "inline-resolving-spinner"
        }, /*#__PURE__*/_react.default.createElement(_spinner.default, {
          size: 14
        })))));
      }
    }
  }]);
}(_react.default.Component);
