// @forge/react/out/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.useForm = exports.replaceUnsupportedDocumentNodes = exports.I18nProvider = exports.useTranslation = exports.useIssueProperty = exports.useSpaceProperty = exports.useContentProperty = exports.default = exports.useConfig = exports.useProductContext = void 0;
const tslib_1 = require("tslib");
var useProductContext_1 = require("./hooks/useProductContext");
Object.defineProperty(exports, "useProductContext", { enumerable: true, get: function () { return useProductContext_1.useProductContext; } });
var useConfig_1 = require("./hooks/useConfig");
Object.defineProperty(exports, "useConfig", { enumerable: true, get: function () { return useConfig_1.useConfig; } });
var reconciler_1 = require("./reconciler");
Object.defineProperty(exports, "default", { enumerable: true, get: function () { return reconciler_1.ForgeReconciler; } });
tslib_1.__exportStar(require("./components"), exports);
var useContentProperty_1 = require("./hooks/useContentProperty");
Object.defineProperty(exports, "useContentProperty", { enumerable: true, get: function () { return useContentProperty_1.useContentProperty; } });
var useSpaceProperty_1 = require("./hooks/useSpaceProperty");
Object.defineProperty(exports, "useSpaceProperty", { enumerable: true, get: function () { return useSpaceProperty_1.useSpaceProperty; } });
var useIssueProperty_1 = require("./hooks/useIssueProperty");
Object.defineProperty(exports, "useIssueProperty", { enumerable: true, get: function () { return useIssueProperty_1.useIssueProperty; } });
var useTranslation_1 = require("./hooks/useTranslation");
Object.defineProperty(exports, "useTranslation", { enumerable: true, get: function () { return useTranslation_1.useTranslation; } });
Object.defineProperty(exports, "I18nProvider", { enumerable: true, get: function () { return useTranslation_1.I18nProvider; } });
var replaceUnsupportedDocumentNodes_1 = require("./components/utils/replaceUnsupportedDocumentNodes");
Object.defineProperty(exports, "replaceUnsupportedDocumentNodes", { enumerable: true, get: function () { return replaceUnsupportedDocumentNodes_1.replaceUnsupportedDocumentNodes; } });
var useForm_1 = require("./hooks/useForm");
Object.defineProperty(exports, "useForm", { enumerable: true, get: function () { return useForm_1.useForm; } });
tslib_1.__exportStar(require("./package-types"), exports);
