// @atlaskit/link-datasource/dist/cjs/ui/issue-like-table/column-picker/concatenated-menu-list-old.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SELECT_ITEMS_MAXIMUM_THRESHOLD = exports.ConcatenatedMenuListOld = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));
var _react = require("@emotion/react");
var _reactIntlNext = require("react-intl-next");
var _select = require("@atlaskit/select");
var _typography = require("@atlaskit/theme/typography");
var _messages = require("./messages");
var _excluded = ["children"];
/**
 * @jsxRuntime classic
 * @jsx jsx
 */
// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766
var SELECT_ITEMS_MAXIMUM_THRESHOLD = exports.SELECT_ITEMS_MAXIMUM_THRESHOLD = 200;
var messageStyles = (0, _react.css)({
  color: "var(--ds-text-subtle, #44546F)",
  // eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values -- Ignored via go/DSP-18766
  font: "var(--ds-font-body-small, ".concat(_typography.fontFallback.body.small, ")"),
  fontWeight: "var(--ds-font-weight-regular, 400)"
});
var ConcatenatedMenuListOld = exports.ConcatenatedMenuListOld = function ConcatenatedMenuListOld(_ref) {
  var children = _ref.children,
    props = (0, _objectWithoutProperties2.default)(_ref, _excluded);
  var shouldUseDefaultMenuList = !children || !Array.isArray(children) || children.length <= SELECT_ITEMS_MAXIMUM_THRESHOLD;
  if (shouldUseDefaultMenuList) {
    return (0, _react.jsx)(_select.components.MenuList, props, children);
  }
  var optionStyle = {
    padding: "var(--ds-space-050, 4px)".concat(" ", "var(--ds-space-200, 16px)"),
    height: 'auto'
  };
  var maximumLimitReachedMessage =
  // eslint-disable-next-line @atlaskit/ui-styling-standard/enforce-style-prop -- Ignored via go/DSP-18766
  (0, _react.jsx)("div", {
    css: messageStyles,
    style: optionStyle
  }, (0, _react.jsx)(_reactIntlNext.FormattedMessage, (0, _extends2.default)({
    tagName: 'div'
  }, _messages.columnPickerMessages.maximumItemsShownLine1)), (0, _react.jsx)(_reactIntlNext.FormattedMessage, (0, _extends2.default)({
    tagName: 'div'
  }, _messages.columnPickerMessages.maximumItemsShownLine2)));
  return (0, _react.jsx)(_select.components.MenuList, props, children.slice(0, SELECT_ITEMS_MAXIMUM_THRESHOLD), maximumLimitReachedMessage);
};
