// @atlaskit/smart-card/dist/cjs/view/FlexibleCard/components/container/indexOld.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
var _typeof = require("@babel/runtime/helpers/typeof");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getContainerStyles = exports.getChildrenOptions = exports.default = void 0;
var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));
var _react = _interopRequireWildcard(require("react"));
var _react2 = require("@emotion/react");
var _reactMagneticDi = require("react-magnetic-di");
var _colors = require("@atlaskit/theme/colors");
var _constants = require("../../../../constants");
var _flexibleUiContext = require("../../../../state/flexible-ui-context");
var _utils = require("../../../../state/flexible-ui-context/utils");
var _flexible = require("../../../../utils/flexible");
var _hoverCardControl = _interopRequireDefault(require("./hover-card-control"));
var _layeredLink = _interopRequireDefault(require("./layered-link"));
var _templateObject;
/**
 * @jsxRuntime classic
 * @jsx jsx
 */
// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
var elevationStyles = (0, _react2.css)({
  border: "1px solid ".concat("var(--ds-border, ".concat(_colors.N40, ")")),
  borderRadius: "var(--ds-border-radius-200, 8px)",
  margin: "var(--ds-space-025, 2px)"
});
var clickableContainerStyles = (0, _react2.css)({
  // eslint-disable-next-line @atlaskit/ui-styling-standard/no-nested-selectors -- Ignored via go/DSP-18766
  'a, button, .has-action': {
    position: 'relative',
    zIndex: 1
  }
});
var getGap = function getGap(size) {
  switch (size) {
    case _constants.SmartLinkSize.XLarge:
      return '1.25rem';
    case _constants.SmartLinkSize.Large:
      return '1rem';
    case _constants.SmartLinkSize.Medium:
      return '.5rem';
    case _constants.SmartLinkSize.Small:
    default:
      return '.25rem';
  }
};

/**
 * Get container padding based on smart link size
 * Equivalent version for DS primitives space token is getPrimitivesPaddingSpaceBySize()
 * at view/FlexibleCard/components/utils.tsx
 */
var getPadding = function getPadding(size) {
  switch (size) {
    case _constants.SmartLinkSize.XLarge:
      return '1.5rem';
    case _constants.SmartLinkSize.Large:
      return '1.25rem';
    case _constants.SmartLinkSize.Medium:
      return '1rem';
    case _constants.SmartLinkSize.Small:
    default:
      return '.5rem';
  }
};
var getContainerPaddingStyles = function getContainerPaddingStyles(size, hidePadding, childrenOptions) {
  var padding = hidePadding ? '0rem' : getPadding(size);
  var gap = getGap(size);
  var previewOnLeft = childrenOptions.previewOnLeft,
    previewOnRight = childrenOptions.previewOnRight;
  return (0, _react2.css)({
    // eslint-disable-next-line @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
    '--container-padding': padding,
    // eslint-disable-next-line @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
    '--container-gap-left': previewOnLeft ? gap : padding,
    // eslint-disable-next-line @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
    '--container-gap-right': previewOnRight ? gap : padding,
    '--preview-block-width': '30%',
    // eslint-disable-next-line @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
    padding: padding
  },
  // eslint-disable-next-line @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
  previewOnLeft ? "padding-left: calc(var(--preview-block-width) + ".concat(gap, ");") : '',
  // eslint-disable-next-line @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
  previewOnRight ? "padding-right: calc(var(--preview-block-width) + ".concat(gap, ");") : '');
};

/**
 * @deprecated consider removing on FF bandicoots-compiled-migration-smartcard
 */
var getContainerStyles = exports.getContainerStyles = function getContainerStyles(size, hideBackground, hideElevation, hidePadding, clickableContainer, childrenOptions) {
  var paddingCss = getContainerPaddingStyles(size, hidePadding, childrenOptions);

  // eslint-disable-next-line @atlaskit/design-system/no-css-tagged-template-expression -- needs manual remediation
  return (0, _react2.css)(_templateObject || (_templateObject = (0, _taggedTemplateLiteral2.default)(["\n\t\tdisplay: flex;\n\t\tgap: ", " 0;\n\t\tflex-direction: column;\n\t\tmin-width: 0;\n\t\toverflow-x: hidden;\n\t\tposition: relative;\n\t\t", "\n\t\t", "\n    ", "\n    ", "\n    &:hover ~ .actions-button-group {\n\t\t\topacity: 1;\n\t\t}\n\t\ta:focus,\n\t\t.has-action:focus {\n\t\t\toutline-offset: -2px;\n\t\t}\n\t"])), getGap(size), hideBackground ? '' : "background-color: ".concat("var(--ds-surface-raised, #FFFFFF)", ";"), paddingCss, hideElevation ? '' : elevationStyles, clickableContainer ? clickableContainerStyles : '');
};
var getLayeredLink = function getLayeredLink(testId, context, children, onClick) {
  var _ref = context || {},
    title = _ref.title,
    _ref$url = _ref.url,
    url = _ref$url === void 0 ? '' : _ref$url;
  var _ref2 = getTitleBlockProps(children) || {},
    target = _ref2.anchorTarget,
    text = _ref2.text;
  return (0, _react2.jsx)(_layeredLink.default, {
    onClick: onClick,
    target: target,
    testId: testId,
    text: text || title,
    url: url
  });
};
var getTitleBlockProps = function getTitleBlockProps(children) {
  var block = _react.default.Children.toArray(children).find(function (child) {
    return (0, _flexible.isFlexibleUiTitleBlock)(child);
  });
  if ( /*#__PURE__*/_react.default.isValidElement(block)) {
    return block.props;
  }
};
var getChildrenOptions = exports.getChildrenOptions = function getChildrenOptions(children, context) {
  var options = {};
  if ((0, _utils.isFlexUiPreviewPresent)(context)) {
    _react.default.Children.map(children, function (child) {
      if ( /*#__PURE__*/_react.default.isValidElement(child)) {
        if ((0, _flexible.isFlexibleUiPreviewBlock)(child)) {
          var placement = child.props.placement;
          if (placement === _constants.MediaPlacement.Left) {
            options.previewOnLeft = true;
          }
          if (placement === _constants.MediaPlacement.Right) {
            options.previewOnRight = true;
          }
        }
      }
    });
  }
  return options;
};
var renderChildren = function renderChildren(children, containerSize, containerTheme, status, retry, onClick) {
  return _react.default.Children.map(children, function (child) {
    // TODO: EDM-6468: Use useFlexibleUiOptionContext for rendering options inside block/element instead
    if ( /*#__PURE__*/_react.default.isValidElement(child) && (0, _flexible.isFlexibleUiBlock)(child)) {
      var blockSize = child.props.size;
      var size = blockSize || containerSize;
      if ((0, _flexible.isFlexibleUiTitleBlock)(child)) {
        return /*#__PURE__*/_react.default.cloneElement(child, {
          // @ts-expect-error
          onClick: onClick,
          retry: retry,
          size: size,
          status: status,
          theme: containerTheme
        });
      }
      // @ts-expect-error
      return /*#__PURE__*/_react.default.cloneElement(child, {
        size: size,
        status: status
      });
    }
  });
};

/**
 * A container is a hidden component that build the Flexible Smart Link.
 * All the Flexible UI components are wrapped inside the container.
 * It inherits the ui props from Card component and applies the custom styling
 * accordingly.
 * @internal
 * @see Block
 */
var ContainerOld = function ContainerOld(_ref3) {
  var children = _ref3.children,
    _ref3$clickableContai = _ref3.clickableContainer,
    clickableContainer = _ref3$clickableContai === void 0 ? false : _ref3$clickableContai,
    _ref3$hideBackground = _ref3.hideBackground,
    hideBackground = _ref3$hideBackground === void 0 ? false : _ref3$hideBackground,
    _ref3$hideElevation = _ref3.hideElevation,
    hideElevation = _ref3$hideElevation === void 0 ? false : _ref3$hideElevation,
    _ref3$hidePadding = _ref3.hidePadding,
    hidePadding = _ref3$hidePadding === void 0 ? false : _ref3$hidePadding,
    onClick = _ref3.onClick,
    retry = _ref3.retry,
    _ref3$showHoverPrevie = _ref3.showHoverPreview,
    showHoverPreview = _ref3$showHoverPrevie === void 0 ? false : _ref3$showHoverPrevie,
    hoverPreviewOptions = _ref3.hoverPreviewOptions,
    actionOptions = _ref3.actionOptions,
    _ref3$size = _ref3.size,
    size = _ref3$size === void 0 ? _constants.SmartLinkSize.Medium : _ref3$size,
    status = _ref3.status,
    _ref3$testId = _ref3.testId,
    testId = _ref3$testId === void 0 ? 'smart-links-container' : _ref3$testId,
    _ref3$theme = _ref3.theme,
    theme = _ref3$theme === void 0 ? _constants.SmartLinkTheme.Link : _ref3$theme;
  var context = (0, _react.useContext)(_flexibleUiContext.FlexibleUiContext);
  var childrenOptions = getChildrenOptions(children, context);
  var canShowHoverPreview = showHoverPreview && status === 'resolved';
  // `retry` object contains action that can be performed on
  // unresolved link (unauthorized, forbidden, not found, etc.)
  var canShowAuthTooltip = showHoverPreview && status === 'unauthorized' && retry !== undefined;
  var container = (0, _react2.jsx)("div", {
    // eslint-disable-next-line @atlaskit/design-system/consistent-css-prop-usage -- Ignored via go/DSP-18766
    css: getContainerStyles(size, hideBackground, hideElevation, hidePadding, clickableContainer, childrenOptions),
    "data-smart-link-container": true,
    "data-testid": testId
  }, clickableContainer ? getLayeredLink(testId, context, children, onClick) : null, renderChildren(children, size, theme, status, retry, onClick));
  if (context !== null && context !== void 0 && context.url && (canShowHoverPreview || canShowAuthTooltip)) {
    return (0, _react2.jsx)(_hoverCardControl.default, {
      isHoverPreview: canShowHoverPreview,
      isAuthTooltip: canShowAuthTooltip,
      actionOptions: actionOptions,
      testId: testId,
      url: context.url,
      hoverPreviewOptions: hoverPreviewOptions
    }, container);
  }
  return container;
};
var _default = exports.default = ContainerOld;
