// @atlaskit/smart-card/dist/es2019/view/HoverCard/components/ImagePreviewOld.js
import React, { useCallback, useRef, useState } from 'react';

// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766

import extractPreview from '../../../extractors/flexible/extract-preview';
import { PreviewBlock } from '../../FlexibleCard/components/blocks';
import { getPreviewBlockStyles, getTransitionStyles } from '../styled';
const ImagePreviewOld = ({
  data,
  fallbackElementHeight
}) => {
  const transitionStarted = useRef(false);
  const previewBlockRef = useRef(null);
  const [showPreview, setShowPreview] = useState(true);
  const [previewCss, setPreviewCss] = useState();

  // Set Preview to a fixed height to enable transitions
  const onPreviewRender = useCallback(() => {
    var _previewBlockRef$curr;
    previewBlockRef.current && setPreviewCss(getPreviewBlockStyles((_previewBlockRef$curr = previewBlockRef.current) === null || _previewBlockRef$curr === void 0 ? void 0 : _previewBlockRef$curr.getBoundingClientRect().height));
  }, []);

  // On error set Preview to Fallback height with transition
  const onPreviewError = useCallback(() => {
    if (transitionStarted.current === false) {
      setPreviewCss(getTransitionStyles(fallbackElementHeight));
      transitionStarted.current = true;
    }
  }, [fallbackElementHeight]);
  const onPreviewTransitionEnd = useCallback(() => {
    setShowPreview(false);
  }, []);
  return showPreview && data && extractPreview(data) ? /*#__PURE__*/React.createElement(PreviewBlock, {
    onError: onPreviewError,
    ignoreContainerPadding: true,
    overrideCss: previewCss,
    onTransitionEnd: onPreviewTransitionEnd,
    blockRef: previewBlockRef,
    onRender: onPreviewRender
  }) : null;
};
export default ImagePreviewOld;
