// @atlaskit/logo/dist/cjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.AssetsLogo = exports.AssetsIcon = exports.AnalyticsLogo = exports.AnalyticsIcon = exports.AlignLogo = exports.AlignIcon = exports.AdminLogo = exports.AdminIcon = void 0;
Object.defineProperty(exports, "AtlasIcon", {
  enumerable: true,
  get: function get() {
    return _atlas.AtlasIcon;
  }
});
Object.defineProperty(exports, "AtlasLogo", {
  enumerable: true,
  get: function get() {
    return _atlas.AtlasLogo;
  }
});
Object.defineProperty(exports, "AtlassianAccessIcon", {
  enumerable: true,
  get: function get() {
    return _atlassianAccess.AtlassianAccessIcon;
  }
});
Object.defineProperty(exports, "AtlassianAccessLogo", {
  enumerable: true,
  get: function get() {
    return _atlassianAccess.AtlassianAccessLogo;
  }
});
Object.defineProperty(exports, "AtlassianAdminIcon", {
  enumerable: true,
  get: function get() {
    return _atlassianAdmin.AtlassianAdminIcon;
  }
});
Object.defineProperty(exports, "AtlassianAdminLogo", {
  enumerable: true,
  get: function get() {
    return _atlassianAdmin.AtlassianAdminLogo;
  }
});
Object.defineProperty(exports, "AtlassianAdministrationIcon", {
  enumerable: true,
  get: function get() {
    return _atlassianAdministration.AtlassianAdministrationIcon;
  }
});
Object.defineProperty(exports, "AtlassianAdministrationLogo", {
  enumerable: true,
  get: function get() {
    return _atlassianAdministration.AtlassianAdministrationLogo;
  }
});
Object.defineProperty(exports, "AtlassianAnalyticsIcon", {
  enumerable: true,
  get: function get() {
    return _atlassianAnalytics.AtlassianAnalyticsIcon;
  }
});
Object.defineProperty(exports, "AtlassianAnalyticsLogo", {
  enumerable: true,
  get: function get() {
    return _atlassianAnalytics.AtlassianAnalyticsLogo;
  }
});
Object.defineProperty(exports, "AtlassianIcon", {
  enumerable: true,
  get: function get() {
    return _atlassian.AtlassianIcon;
  }
});
Object.defineProperty(exports, "AtlassianLogo", {
  enumerable: true,
  get: function get() {
    return _atlassian.AtlassianLogo;
  }
});
Object.defineProperty(exports, "AtlassianMarketplaceIcon", {
  enumerable: true,
  get: function get() {
    return _atlassianMarketplace.AtlassianMarketplaceIcon;
  }
});
Object.defineProperty(exports, "AtlassianMarketplaceLogo", {
  enumerable: true,
  get: function get() {
    return _atlassianMarketplace.AtlassianMarketplaceLogo;
  }
});
exports.HomeLogo = exports.HomeIcon = exports.GuardLogo = exports.GuardIcon = exports.GuardDetectLogo = exports.GuardDetectIcon = exports.GoalsLogo = exports.GoalsIcon = exports.FocusLogo = exports.FocusIcon = exports.ConfluenceLogo = exports.ConfluenceIcon = exports.CompassLogo = exports.CompassIcon = exports.CompanyHubLogo = exports.CompanyHubIcon = exports.ChatLogo = exports.ChatIcon = exports.BitbucketLogo = exports.BitbucketIcon = void 0;
Object.defineProperty(exports, "JiraAlignIcon", {
  enumerable: true,
  get: function get() {
    return _jiraAlign.JiraAlignIcon;
  }
});
Object.defineProperty(exports, "JiraAlignLogo", {
  enumerable: true,
  get: function get() {
    return _jiraAlign.JiraAlignLogo;
  }
});
exports.JiraServiceManagementLogo = exports.JiraServiceManagementIcon = exports.JiraProductDiscoveryLogo = exports.JiraProductDiscoveryIcon = exports.JiraLogo = exports.JiraIcon = void 0;
Object.defineProperty(exports, "JiraSoftwareIcon", {
  enumerable: true,
  get: function get() {
    return _jiraSoftware.JiraSoftwareIcon;
  }
});
Object.defineProperty(exports, "JiraSoftwareLogo", {
  enumerable: true,
  get: function get() {
    return _jiraSoftware.JiraSoftwareLogo;
  }
});
Object.defineProperty(exports, "JiraWorkManagementIcon", {
  enumerable: true,
  get: function get() {
    return _jiraWorkManagement.JiraWorkManagementIcon;
  }
});
Object.defineProperty(exports, "JiraWorkManagementLogo", {
  enumerable: true,
  get: function get() {
    return _jiraWorkManagement.JiraWorkManagementLogo;
  }
});
Object.defineProperty(exports, "LoomAttributionIcon", {
  enumerable: true,
  get: function get() {
    return _loomAttribution.LoomAttributionIcon;
  }
});
Object.defineProperty(exports, "LoomAttributionLogo", {
  enumerable: true,
  get: function get() {
    return _loomAttribution.LoomAttributionLogo;
  }
});
exports.TrelloLogo = exports.TrelloIcon = exports.TeamsLogo = exports.TeamsIcon = exports.TalentLogo = exports.TalentIcon = exports.StudioLogo = exports.StudioIcon = exports.StatuspageLogo = exports.StatuspageIcon = exports.SearchLogo = exports.SearchIcon = exports.RovoLogo = exports.RovoIcon = exports.ProjectsLogo = exports.ProjectsIcon = exports.OpsgenieLogo = exports.OpsgenieIcon = exports.LoomLogo = exports.LoomIcon = void 0;
var _admin = require("@atlaskit/temp-nav-app-icons/admin");
var _align = require("@atlaskit/temp-nav-app-icons/align");
var _analytics = require("@atlaskit/temp-nav-app-icons/analytics");
var _assets = require("@atlaskit/temp-nav-app-icons/assets");
var _bitbucket = require("@atlaskit/temp-nav-app-icons/bitbucket");
var _chat = require("@atlaskit/temp-nav-app-icons/chat");
var _companyHub = require("@atlaskit/temp-nav-app-icons/company-hub");
var _compass = require("@atlaskit/temp-nav-app-icons/compass");
var _confluence = require("@atlaskit/temp-nav-app-icons/confluence");
var _focus = require("@atlaskit/temp-nav-app-icons/focus");
var _goals = require("@atlaskit/temp-nav-app-icons/goals");
var _guard = require("@atlaskit/temp-nav-app-icons/guard");
var _guardDetect = require("@atlaskit/temp-nav-app-icons/guard-detect");
var _home = require("@atlaskit/temp-nav-app-icons/home");
var _jira = require("@atlaskit/temp-nav-app-icons/jira");
var _jiraProductDiscovery = require("@atlaskit/temp-nav-app-icons/jira-product-discovery");
var _jiraServiceManagement = require("@atlaskit/temp-nav-app-icons/jira-service-management");
var _loom = require("@atlaskit/temp-nav-app-icons/loom");
var _opsgenie = require("@atlaskit/temp-nav-app-icons/opsgenie");
var _projects = require("@atlaskit/temp-nav-app-icons/projects");
var _rovo = require("@atlaskit/temp-nav-app-icons/rovo");
var _search = require("@atlaskit/temp-nav-app-icons/search");
var _statuspage = require("@atlaskit/temp-nav-app-icons/statuspage");
var _studio = require("@atlaskit/temp-nav-app-icons/studio");
var _talent = require("@atlaskit/temp-nav-app-icons/talent");
var _teams = require("@atlaskit/temp-nav-app-icons/teams");
var _trello = require("@atlaskit/temp-nav-app-icons/trello");
var _bitbucket2 = require("./legacy-logos/bitbucket");
var _compass2 = require("./legacy-logos/compass");
var _confluence2 = require("./legacy-logos/confluence");
var _focus2 = require("./legacy-logos/focus");
var _guard2 = require("./legacy-logos/guard");
var _jira2 = require("./legacy-logos/jira");
var _jiraProductDiscovery2 = require("./legacy-logos/jira-product-discovery");
var _jiraServiceManagement2 = require("./legacy-logos/jira-service-management");
var _loom2 = require("./legacy-logos/loom");
var _opsgenie2 = require("./legacy-logos/opsgenie");
var _rovo2 = require("./legacy-logos/rovo");
var _statuspage2 = require("./legacy-logos/statuspage");
var _trello2 = require("./legacy-logos/trello");
var _atlassianAnalytics = require("./legacy-logos/atlassian-analytics");
var _atlas = require("./legacy-logos/atlas");
var _atlassianMarketplace = require("./legacy-logos/atlassian-marketplace");
var _atlassianAccess = require("./legacy-logos/atlassian-access");
var _atlassianAdministration = require("./legacy-logos/atlassian-administration");
var _atlassianAdmin = require("./legacy-logos/atlassian-admin");
var _jiraAlign = require("./legacy-logos/jira-align");
var _jiraWorkManagement = require("./legacy-logos/jira-work-management");
var _loomAttribution = require("./legacy-logos/loom-attribution");
var _atlassian = require("./legacy-logos/atlassian");
var _jiraSoftware = require("./legacy-logos/jira-software");
var _logoConfig = require("./logo-config");
/**
 * __Admin icon__
 *
 * An admin icon without an accompanying wordmark.
 *
 * - [Examples](https://atlassian.design/components/{packageName}/examples)
 * - [Code](https://atlassian.design/components/{packageName}/code)
 * - [Usage](https://atlassian.design/components/{packageName}/usage)
 */
var AdminIcon = exports.AdminIcon = (0, _logoConfig.tempSizeWrapper)(_admin.AdminIcon);
/**
 * __Admin logo__
 *
 * An admin logo with both the wordmark and the icon combined.
 *
 * - [Examples](https://atlassian.design/components/{packageName}/examples)
 * - [Code](https://atlassian.design/components/{packageName}/code)
 * - [Usage](https://atlassian.design/components/{packageName}/usage)
 */
var AdminLogo = exports.AdminLogo = (0, _logoConfig.tempSizeWrapper)(_admin.AdminLogo);
/**
 * __Assets icon__
 *
 * An assets icon without an accompanying wordmark.
 *
 * - [Examples](https://atlassian.design/components/{packageName}/examples)
 * - [Code](https://atlassian.design/components/{packageName}/code)
 * - [Usage](https://atlassian.design/components/{packageName}/usage)
 */
var AssetsIcon = exports.AssetsIcon = (0, _logoConfig.tempSizeWrapper)(_assets.AssetsIcon);
/**
 * __Assets logo__
 *
 * An assets logo with both the wordmark and the icon combined.
 *
 * - [Examples](https://atlassian.design/components/{packageName}/examples)
 * - [Code](https://atlassian.design/components/{packageName}/code)
 * - [Usage](https://atlassian.design/components/{packageName}/usage)
 */
var AssetsLogo = exports.AssetsLogo = (0, _logoConfig.tempSizeWrapper)(_assets.AssetsLogo);
/**
 * __Analytics icon__
 *
 * An analytics icon without an accompanying wordmark.
 *
 * - [Examples](https://atlassian.design/components/{packageName}/examples)
 * - [Code](https://atlassian.design/components/{packageName}/code)
 * - [Usage](https://atlassian.design/components/{packageName}/usage)
 */
var AnalyticsIcon = exports.AnalyticsIcon = (0, _logoConfig.tempSizeWrapper)(_analytics.AnalyticsIcon);
/**
 * __Analytics logo__
 *
 * An analytics logo with both the wordmark and the icon combined.
 *
 * - [Examples](https://atlassian.design/components/{packageName}/examples)
 * - [Code](https://atlassian.design/components/{packageName}/code)
 * - [Usage](https://atlassian.design/components/{packageName}/usage)
 */
var AnalyticsLogo = exports.AnalyticsLogo = (0, _logoConfig.tempSizeWrapper)(_analytics.AnalyticsLogo);
/**
 * __Align icon__
 *
 * An align icon without an accompanying wordmark.
 *
 * - [Examples](https://atlassian.design/components/{packageName}/examples)
 * - [Code](https://atlassian.design/components/{packageName}/code)
 * - [Usage](https://atlassian.design/components/{packageName}/usage)
 */
var AlignIcon = exports.AlignIcon = (0, _logoConfig.tempSizeWrapper)(_align.AlignIcon);
/**
 * __Align logo__
 *
 * An align logo with both the wordmark and the icon combined.
 *
 * - [Examples](https://atlassian.design/components/{packageName}/examples)
 * - [Code](https://atlassian.design/components/{packageName}/code)
 * - [Usage](https://atlassian.design/components/{packageName}/usage)
 */
var AlignLogo = exports.AlignLogo = (0, _logoConfig.tempSizeWrapper)(_align.AlignLogo);
/**
 * __Chat icon__
 *
 * A chat icon without an accompanying wordmark.
 *
 * - [Examples](https://atlassian.design/components/{packageName}/examples)
 * - [Code](https://atlassian.design/components/{packageName}/code)
 * - [Usage](https://atlassian.design/components/{packageName}/usage)
 */
var ChatIcon = exports.ChatIcon = (0, _logoConfig.tempSizeWrapper)(_chat.ChatIcon);
/**
 * __Chat logo__
 *
 * A chat logo with both the wordmark and the icon combined.
 *
 * - [Examples](https://atlassian.design/components/{packageName}/examples)
 * - [Code](https://atlassian.design/components/{packageName}/code)
 * - [Usage](https://atlassian.design/components/{packageName}/usage)
 */
var ChatLogo = exports.ChatLogo = (0, _logoConfig.tempSizeWrapper)(_chat.ChatLogo);
/**
 * __Company hub icon__
 *
 * A company hub icon without an accompanying wordmark.
 *
 * - [Examples](https://atlassian.design/components/{packageName}/examples)
 * - [Code](https://atlassian.design/components/{packageName}/code)
 * - [Usage](https://atlassian.design/components/{packageName}/usage)
 */
var CompanyHubIcon = exports.CompanyHubIcon = (0, _logoConfig.tempSizeWrapper)(_companyHub.CompanyHubIcon);
/**
 * __Company hub logo__
 *
 * A company hub logo with both the wordmark and the icon combined.
 *
 * - [Examples](https://atlassian.design/components/{packageName}/examples)
 * - [Code](https://atlassian.design/components/{packageName}/code)
 * - [Usage](https://atlassian.design/components/{packageName}/usage)
 */
var CompanyHubLogo = exports.CompanyHubLogo = (0, _logoConfig.tempSizeWrapper)(_companyHub.CompanyHubLogo);
/**
 * __Goals icon__
 *
 * A goals icon without an accompanying wordmark.
 *
 * - [Examples](https://atlassian.design/components/{packageName}/examples)
 * - [Code](https://atlassian.design/components/{packageName}/code)
 * - [Usage](https://atlassian.design/components/{packageName}/usage)
 */
var GoalsIcon = exports.GoalsIcon = (0, _logoConfig.tempSizeWrapper)(_goals.GoalsIcon);
/**
 * __Goals logo__
 *
 * A goals logo with both the wordmark and the icon combined.
 *
 * - [Examples](https://atlassian.design/components/{packageName}/examples)
 * - [Code](https://atlassian.design/components/{packageName}/code)
 * - [Usage](https://atlassian.design/components/{packageName}/usage)
 */
var GoalsLogo = exports.GoalsLogo = (0, _logoConfig.tempSizeWrapper)(_goals.GoalsLogo);
/**
 * __Guard detect icon__
 *
 * A guard detect icon without an accompanying wordmark.
 *
 * - [Examples](https://atlassian.design/components/{packageName}/examples)
 * - [Code](https://atlassian.design/components/{packageName}/code)
 * - [Usage](https://atlassian.design/components/{packageName}/usage)
 */
var GuardDetectIcon = exports.GuardDetectIcon = (0, _logoConfig.tempSizeWrapper)(_guardDetect.GuardDetectIcon);
/**
 * __Guard detect logo__
 *
 * A guard detect logo with both the wordmark and the icon combined.
 *
 * - [Examples](https://atlassian.design/components/{packageName}/examples)
 * - [Code](https://atlassian.design/components/{packageName}/code)
 * - [Usage](https://atlassian.design/components/{packageName}/usage)
 */
var GuardDetectLogo = exports.GuardDetectLogo = (0, _logoConfig.tempSizeWrapper)(_guardDetect.GuardDetectLogo);
/**
 * __Home icon__
 *
 * A home icon without an accompanying wordmark.
 *
 * - [Examples](https://atlassian.design/components/{packageName}/examples)
 * - [Code](https://atlassian.design/components/{packageName}/code)
 * - [Usage](https://atlassian.design/components/{packageName}/usage)
 */
var HomeIcon = exports.HomeIcon = (0, _logoConfig.tempSizeWrapper)(_home.HomeIcon);
/**
 * __Home logo__
 *
 * A home logo with both the wordmark and the icon combined.
 *
 * - [Examples](https://atlassian.design/components/{packageName}/examples)
 * - [Code](https://atlassian.design/components/{packageName}/code)
 * - [Usage](https://atlassian.design/components/{packageName}/usage)
 */
var HomeLogo = exports.HomeLogo = (0, _logoConfig.tempSizeWrapper)(_home.HomeLogo);
/**
 * __Projects icon__
 *
 * A projects icon without an accompanying wordmark.
 *
 * - [Examples](https://atlassian.design/components/{packageName}/examples)
 * - [Code](https://atlassian.design/components/{packageName}/code)
 * - [Usage](https://atlassian.design/components/{packageName}/usage)
 */
var ProjectsIcon = exports.ProjectsIcon = (0, _logoConfig.tempSizeWrapper)(_projects.ProjectsIcon);
/**
 * __Projects logo__
 *
 * A projects logo with both the wordmark and the icon combined.
 *
 * - [Examples](https://atlassian.design/components/{packageName}/examples)
 * - [Code](https://atlassian.design/components/{packageName}/code)
 * - [Usage](https://atlassian.design/components/{packageName}/usage)
 */
var ProjectsLogo = exports.ProjectsLogo = (0, _logoConfig.tempSizeWrapper)(_projects.ProjectsLogo);
/**
 * __Search icon__
 *
 * A search icon without an accompanying wordmark.
 *
 * - [Examples](https://atlassian.design/components/{packageName}/examples)
 * - [Code](https://atlassian.design/components/{packageName}/code)
 * - [Usage](https://atlassian.design/components/{packageName}/usage)
 */
var SearchIcon = exports.SearchIcon = (0, _logoConfig.tempSizeWrapper)(_search.SearchIcon);
/**
 * __Search logo__
 *
 * A search logo with both the wordmark and the icon combined.
 *
 * - [Examples](https://atlassian.design/components/{packageName}/examples)
 * - [Code](https://atlassian.design/components/{packageName}/code)
 * - [Usage](https://atlassian.design/components/{packageName}/usage)
 */
var SearchLogo = exports.SearchLogo = (0, _logoConfig.tempSizeWrapper)(_search.SearchLogo);
/**
 * __Studio icon__
 *
 * A studio icon without an accompanying wordmark.
 *
 * - [Examples](https://atlassian.design/components/{packageName}/examples)
 * - [Code](https://atlassian.design/components/{packageName}/code)
 * - [Usage](https://atlassian.design/components/{packageName}/usage)
 */
var StudioIcon = exports.StudioIcon = (0, _logoConfig.tempSizeWrapper)(_studio.StudioIcon);
/**
 * __Studio logo__
 *
 * A studio logo with both the wordmark and the icon combined.
 *
 * - [Examples](https://atlassian.design/components/{packageName}/examples)
 * - [Code](https://atlassian.design/components/{packageName}/code)
 * - [Usage](https://atlassian.design/components/{packageName}/usage)
 */
var StudioLogo = exports.StudioLogo = (0, _logoConfig.tempSizeWrapper)(_studio.StudioLogo);
/**
 * __Talent icon__
 *
 * A talent icon without an accompanying wordmark.
 *
 * - [Examples](https://atlassian.design/components/{packageName}/examples)
 * - [Code](https://atlassian.design/components/{packageName}/code)
 * - [Usage](https://atlassian.design/components/{packageName}/usage)
 */
var TalentIcon = exports.TalentIcon = (0, _logoConfig.tempSizeWrapper)(_talent.TalentIcon);
/**
 * __Talent logo__
 *
 * A talent logo with both the wordmark and the icon combined.
 *
 * - [Examples](https://atlassian.design/components/{packageName}/examples)
 * - [Code](https://atlassian.design/components/{packageName}/code)
 * - [Usage](https://atlassian.design/components/{packageName}/usage)
 */
var TalentLogo = exports.TalentLogo = (0, _logoConfig.tempSizeWrapper)(_talent.TalentLogo);
/**
 * __Teams icon__
 *
 * A teams icon without an accompanying wordmark.
 *
 * - [Examples](https://atlassian.design/components/{packageName}/examples)
 * - [Code](https://atlassian.design/components/{packageName}/code)
 * - [Usage](https://atlassian.design/components/{packageName}/usage)
 */
var TeamsIcon = exports.TeamsIcon = (0, _logoConfig.tempSizeWrapper)(_teams.TeamsIcon);
/**
 * __Teams logo__
 *
 * A teams logo with both the wordmark and the icon combined.
 *
 * - [Examples](https://atlassian.design/components/{packageName}/examples)
 * - [Code](https://atlassian.design/components/{packageName}/code)
 * - [Usage](https://atlassian.design/components/{packageName}/usage)
 */
var TeamsLogo = exports.TeamsLogo = (0, _logoConfig.tempSizeWrapper)(_teams.TeamsLogo);

/**
 * __Bitbucket icon__
 *
 * The Bitbucket icon without an accompanying wordmark.
 *
 * - [Examples](https://atlassian.design/components/logo/examples)
 * - [Code](https://atlassian.design/components/logo/code)
 * - [Usage](https://atlassian.design/components/logo/usage)
 */
var BitbucketIcon = exports.BitbucketIcon = (0, _logoConfig.createFeatureFlaggedComponent)(_bitbucket2.BitbucketIcon, _bitbucket.BitbucketIcon);
/**
 * __Bitbucket logo__
 *
 * The Bitbucket logo with both the wordmark and the icon combined.
 *
 * - [Examples](https://atlassian.design/components/logo/examples)
 * - [Code](https://atlassian.design/components/logo/code)
 * - [Usage](https://atlassian.design/components/logo/usage)
 */
var BitbucketLogo = exports.BitbucketLogo = (0, _logoConfig.createFeatureFlaggedComponent)(_bitbucket2.BitbucketLogo, _bitbucket.BitbucketLogo);
/**
 * __Compass icon__
 *
 * The Compass icon without an accompanying wordmark.
 *
 * - [Examples](https://atlassian.design/components/logo/examples)
 * - [Code](https://atlassian.design/components/logo/code)
 * - [Usage](https://atlassian.design/components/logo/usage)
 */
var CompassIcon = exports.CompassIcon = (0, _logoConfig.createFeatureFlaggedComponent)(_compass2.CompassIcon, _compass.CompassIcon);
/**
 * __Compass logo__
 *
 * The Compass logo with both the wordmark and the icon combined.
 *
 * - [Examples](https://atlassian.design/components/logo/examples)
 * - [Code](https://atlassian.design/components/logo/code)
 * - [Usage](https://atlassian.design/components/logo/usage)
 */
var CompassLogo = exports.CompassLogo = (0, _logoConfig.createFeatureFlaggedComponent)(_compass2.CompassLogo, _compass.CompassLogo);

/**
 * __Confluence icon__
 *
 * The Confluence icon without an accompanying wordmark.
 *
 * - [Examples](https://atlassian.design/components/logo/examples)
 * - [Code](https://atlassian.design/components/logo/code)
 * - [Usage](https://atlassian.design/components/logo/usage)
 */
var ConfluenceIcon = exports.ConfluenceIcon = (0, _logoConfig.createFeatureFlaggedComponent)(_confluence2.ConfluenceIcon, _confluence.ConfluenceIcon);
/**
 * __Confluence logo__
 *
 * The Confluence logo with both the wordmark and the icon combined.
 *
 * - [Examples](https://atlassian.design/components/logo/examples)
 * - [Code](https://atlassian.design/components/logo/code)
 * - [Usage](https://atlassian.design/components/logo/usage)
 */
var ConfluenceLogo = exports.ConfluenceLogo = (0, _logoConfig.createFeatureFlaggedComponent)(_confluence2.ConfluenceLogo, _confluence.ConfluenceLogo);
/**
 * __Jira icon__
 *
 * The Jira icon without an accompanying wordmark.
 *
 * - [Examples](https://atlassian.design/components/logo/examples)
 * - [Code](https://atlassian.design/components/logo/code)
 * - [Usage](https://atlassian.design/components/logo/usage)
 */
var JiraIcon = exports.JiraIcon = (0, _logoConfig.createFeatureFlaggedComponent)(_jira2.JiraIcon, _jira.JiraIcon);
/**
 * __Jira logo__
 *
 * The Jira logo with both the wordmark and the icon combined.
 *
 * - [Examples](https://atlassian.design/components/logo/examples)
 * - [Code](https://atlassian.design/components/logo/code)
 * - [Usage](https://atlassian.design/components/logo/usage)
 */
var JiraLogo = exports.JiraLogo = (0, _logoConfig.createFeatureFlaggedComponent)(_jira2.JiraLogo, _jira.JiraLogo);
/**
 * __Jira product discovery icon__
 *
 * The Jira Product Discovery icon without an accompanying wordmark.
 *
 * - [Examples](https://atlassian.design/components/logo/examples)
 * - [Code](https://atlassian.design/components/logo/code)
 * - [Usage](https://atlassian.design/components/logo/usage)
 */
var JiraProductDiscoveryIcon = exports.JiraProductDiscoveryIcon = (0, _logoConfig.createFeatureFlaggedComponent)(_jiraProductDiscovery2.JiraProductDiscoveryIcon, _jiraProductDiscovery.JiraProductDiscoveryIcon);

/**
 * __Jira product discovery logo__
 *
 * The Jira Product Discovery logo with both the wordmark and the icon combined.
 *
 * - [Examples](https://atlassian.design/components/logo/examples)
 * - [Code](https://atlassian.design/components/logo/code)
 * - [Usage](https://atlassian.design/components/logo/usage)
 */
var JiraProductDiscoveryLogo = exports.JiraProductDiscoveryLogo = (0, _logoConfig.createFeatureFlaggedComponent)(_jiraProductDiscovery2.JiraProductDiscoveryLogo, _jiraProductDiscovery.JiraProductDiscoveryLogo);

/**
 * __Jira service management icon__
 *
 * The Jira Service Management icon without an accompanying wordmark.
 *
 * - [Examples](https://atlassian.design/components/logo/examples)
 * - [Code](https://atlassian.design/components/logo/code)
 * - [Usage](https://atlassian.design/components/logo/usage)
 */
var JiraServiceManagementIcon = exports.JiraServiceManagementIcon = (0, _logoConfig.createFeatureFlaggedComponent)(_jiraServiceManagement2.JiraServiceManagementIcon, _jiraServiceManagement.JiraServiceManagementIcon);

/**
 * __Jira service management logo__
 *
 * The Jira Service Management logo with both the wordmark and the icon combined.
 *
 * - [Examples](https://atlassian.design/components/logo/examples)
 * - [Code](https://atlassian.design/components/logo/code)
 * - [Usage](https://atlassian.design/components/logo/usage)
 */
var JiraServiceManagementLogo = exports.JiraServiceManagementLogo = (0, _logoConfig.createFeatureFlaggedComponent)(_jiraServiceManagement2.JiraServiceManagementLogo, _jiraServiceManagement.JiraServiceManagementLogo);

/**
 * __Loom icon__
 *
 * A loom icon {description}.
 *
 * - [Examples](https://atlassian.design/components/logo/examples)
 * - [Code](https://atlassian.design/components/logo/code)
 * - [Usage](https://atlassian.design/components/logo/usage)
 */
var LoomIcon = exports.LoomIcon = (0, _logoConfig.createFeatureFlaggedComponent)(_loom2.LoomIcon, _loom.LoomIcon);

/**
 * __Loom logo__
 *
 * The Loom logo with both the wordmark and the icon combined.
 *
 * - [Examples](https://atlassian.design/components/logo/examples)
 * - [Code](https://atlassian.design/components/logo/code)
 * - [Usage](https://atlassian.design/components/logo/usage)
 */
var LoomLogo = exports.LoomLogo = (0, _logoConfig.createFeatureFlaggedComponent)(_loom2.LoomLogo, _loom.LoomLogo);

/**
 * __Opsgenie icon__
 *
 * The Opsgenie icon without an accompanying wordmark.
 *
 * - [Examples](https://atlassian.design/components/logo/examples)
 * - [Code](https://atlassian.design/components/logo/code)
 * - [Usage](https://atlassian.design/components/logo/usage)
 */
var OpsgenieIcon = exports.OpsgenieIcon = (0, _logoConfig.createFeatureFlaggedComponent)(_opsgenie2.OpsgenieIcon, _opsgenie.OpsgenieIcon);
/**
 * __Opsgenie logo__
 *
 * The Opsgenie logo with both the wordmark and the icon combined.
 *
 * - [Examples](https://atlassian.design/components/logo/examples)
 * - [Code](https://atlassian.design/components/logo/code)
 * - [Usage](https://atlassian.design/components/logo/usage)
 */
var OpsgenieLogo = exports.OpsgenieLogo = (0, _logoConfig.createFeatureFlaggedComponent)(_opsgenie2.OpsgenieLogo, _opsgenie.OpsgenieLogo);

/**
 * __Statuspage icon__
 *
 * The Statuspage icon without an accompanying wordmark.
 *
 * - [Examples](https://atlassian.design/components/logo/examples)
 * - [Code](https://atlassian.design/components/logo/code)
 * - [Usage](https://atlassian.design/components/logo/usage)
 */
var StatuspageIcon = exports.StatuspageIcon = (0, _logoConfig.createFeatureFlaggedComponent)(_statuspage2.StatuspageIcon, _statuspage.StatuspageIcon);
/**
 * __Statuspage logo__
 *
 * The Statuspage logo with both the wordmark and the icon combined.
 *
 * - [Examples](https://atlassian.design/components/logo/examples)
 * - [Code](https://atlassian.design/components/logo/code)
 * - [Usage](https://atlassian.design/components/logo/usage)
 */
var StatuspageLogo = exports.StatuspageLogo = (0, _logoConfig.createFeatureFlaggedComponent)(_statuspage2.StatuspageLogo, _statuspage.StatuspageLogo);
/**
 * __Trello icon__
 *
 * The Trello icon without an accompanying wordmark.
 *
 * - [Examples](https://atlassian.design/components/logo/examples)
 * - [Code](https://atlassian.design/components/logo/code)
 * - [Usage](https://atlassian.design/components/logo/usage)
 */
var TrelloIcon = exports.TrelloIcon = (0, _logoConfig.createFeatureFlaggedComponent)(_trello2.TrelloIcon, _trello.TrelloIcon);
/**
 * __Trello logo__
 *
 * The Trello logo with both the wordmark and the icon combined.
 *
 * - [Examples](https://atlassian.design/components/logo/examples)
 * - [Code](https://atlassian.design/components/logo/code)
 * - [Usage](https://atlassian.design/components/logo/usage)
 */
var TrelloLogo = exports.TrelloLogo = (0, _logoConfig.createFeatureFlaggedComponent)(_trello2.TrelloLogo, _trello.TrelloLogo);
/**
 * __Rovo icon__
 *
 * The Rovo icon without an accompanying wordmark.
 *
 * - [Examples](https://atlassian.design/components/logo/examples)
 * - [Code](https://atlassian.design/components/logo/code)
 * - [Usage](https://atlassian.design/components/logo/usage)
 */
var RovoIcon = exports.RovoIcon = (0, _logoConfig.createFeatureFlaggedComponent)(_rovo2.RovoIcon, _rovo.RovoIcon);
/**
 * __Rovo logo__
 *
 * The Rovo logo with both the wordmark and the icon combined.
 *
 * - [Examples](https://atlassian.design/components/logo/examples)
 * - [Code](https://atlassian.design/components/logo/code)
 * - [Usage](https://atlassian.design/components/logo/usage)
 */
var RovoLogo = exports.RovoLogo = (0, _logoConfig.createFeatureFlaggedComponent)(_rovo2.RovoLogo, _rovo.RovoLogo);
/**
 * __Guard icon__
 *
 * The Guard icon without an accompanying wordmark.
 *
 * - [Examples](https://atlassian.design/components/logo/examples)
 * - [Code](https://atlassian.design/components/logo/code)
 * - [Usage](https://atlassian.design/components/logo/usage)
 */
var GuardIcon = exports.GuardIcon = (0, _logoConfig.createFeatureFlaggedComponent)(_guard2.GuardIcon, _guard.GuardIcon);
/**
 * __Guard logo__
 *
 * The Guard logo with both the wordmark and the icon combined.
 *
 * - [Examples](https://atlassian.design/components/logo/examples)
 * - [Code](https://atlassian.design/components/logo/code)
 * - [Usage](https://atlassian.design/components/logo/usage)
 */
var GuardLogo = exports.GuardLogo = (0, _logoConfig.createFeatureFlaggedComponent)(_guard2.GuardLogo, _guard.GuardLogo);
/**
 * __Focus icon__
 *
 * The Focus icon without an accompanying wordmark.
 *
 * - [Examples](https://atlassian.design/components/logo/examples)
 * - [Code](https://atlassian.design/components/logo/code)
 * - [Usage](https://atlassian.design/components/logo/usage)
 */
var FocusIcon = exports.FocusIcon = (0, _logoConfig.createFeatureFlaggedComponent)(_focus2.FocusIcon, _focus.FocusIcon);
/**
 * __Focus logo__
 *
 * The Focus logo with both the wordmark and the icon combined.
 *
 * - [Examples](https://atlassian.design/components/logo/examples)
 * - [Code](https://atlassian.design/components/logo/code)
 * - [Usage](https://atlassian.design/components/logo/usage)
 */
var FocusLogo = exports.FocusLogo = (0, _logoConfig.createFeatureFlaggedComponent)(_focus2.FocusLogo, _focus.FocusLogo);
