// @atlaskit/link-extractors/dist/es2019/entity/index.js
import React from 'react';
import { ConfluenceIcon } from '@atlaskit/logo/confluence-icon';
import { JiraIcon } from '@atlaskit/logo/jira-icon';
import { CONFLUENCE_GENERATOR_ID, JIRA_GENERATOR_ID } from '../common/constants';
export const isEntityPresent = response => Boolean(response === null || response === void 0 ? void 0 : response.nounData);
export const extractEntity = response => response === null || response === void 0 ? void 0 : response.nounData;
export const extractEntityEmbedUrl = response => {
  const entity = extractEntity(response);
  if (entity && instanceOfDesignEntity(entity)) {
    return entity['atlassian:design'].liveEmbedUrl;
  }
};
export function instanceOfDesignEntity(object) {
  return typeof object === 'object' && object !== null && 'atlassian:design' in object;
}
export const extractEntityProvider = response => {
  var _response$meta;
  if (!(response !== null && response !== void 0 && (_response$meta = response.meta) !== null && _response$meta !== void 0 && _response$meta.generator)) {
    return undefined;
  }
  const {
    icon,
    id,
    image,
    name
  } = response.meta.generator;
  if (!name || !icon) {
    throw Error('Link.generator requires a name and icon.');
  }
  let providerIcon;
  switch (id) {
    case CONFLUENCE_GENERATOR_ID:
      providerIcon = /*#__PURE__*/React.createElement(ConfluenceIcon, {
        appearance: "brand",
        size: "xxsmall"
      });
      break;
    case JIRA_GENERATOR_ID:
      providerIcon = /*#__PURE__*/React.createElement(JiraIcon, {
        appearance: "brand",
        size: "xxsmall"
      });
      break;
    default:
      providerIcon = icon.url;
  }
  return {
    text: name,
    icon: providerIcon,
    id,
    image: image ? image : icon.url
  };
};
export const extractEntityIcon = response => {
  const entity = extractEntity(response);
  let url;
  if (instanceOfDesignEntity(entity)) {
    url = entity === null || entity === void 0 ? void 0 : entity['atlassian:design'].iconUrl;
  }
  return {
    url,
    label: entity === null || entity === void 0 ? void 0 : entity.displayName
  };
};
