// @atlaskit/link-datasource/dist/cjs/ui/common/modal/count-view-smart-link/count-view-smart-link-old.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SmartLinkOld = exports.SmartCardPlaceholderOld = void 0;
var _react = _interopRequireDefault(require("react"));
var _reactIntlNext = require("react-intl-next");
var _primitives = require("@atlaskit/primitives");
var _link = _interopRequireDefault(require("../../../issue-like-table/render-type/link"));
var placeholderSmartLinkStyles = (0, _primitives.xcss)({
  backgroundColor: 'elevation.surface.raised',
  borderRadius: "var(--ds-border-radius-200, 3px)",
  color: 'color.text.brand',
  paddingTop: 'space.0',
  paddingBottom: 'space.0',
  paddingLeft: 'space.025',
  paddingRight: 'space.025',
  boxShadow: 'elevation.shadow.raised'
});
var smartLinkContainerStyles = (0, _primitives.xcss)({
  paddingLeft: 'space.025'
});
var SmartCardPlaceholderOld = exports.SmartCardPlaceholderOld = function SmartCardPlaceholderOld(_ref) {
  var placeholderText = _ref.placeholderText;
  return /*#__PURE__*/_react.default.createElement(_primitives.Box, {
    xcss: smartLinkContainerStyles
  }, /*#__PURE__*/_react.default.createElement(_primitives.Box, {
    as: "span",
    testId: "datasource-modal--smart-card-placeholder",
    xcss: placeholderSmartLinkStyles
  }, /*#__PURE__*/_react.default.createElement(_reactIntlNext.FormattedMessage, placeholderText)));
};
var SmartLinkOld = exports.SmartLinkOld = function SmartLinkOld(_ref2) {
  var url = _ref2.url;
  return /*#__PURE__*/_react.default.createElement(_primitives.Box, {
    xcss: smartLinkContainerStyles
  }, /*#__PURE__*/_react.default.createElement(_link.default, {
    url: url
  }));
};
