// @atlaskit/link-datasource/dist/es2019/ui/common/modal/count-view-smart-link/count-view-smart-link-old.js
import React from 'react';
import { FormattedMessage } from 'react-intl-next';
import { Box, xcss } from '@atlaskit/primitives';
import LinkRenderType from '../../../issue-like-table/render-type/link';
const placeholderSmartLinkStyles = xcss({
  backgroundColor: 'elevation.surface.raised',
  borderRadius: "var(--ds-border-radius-200, 3px)",
  color: 'color.text.brand',
  paddingTop: 'space.0',
  paddingBottom: 'space.0',
  paddingLeft: 'space.025',
  paddingRight: 'space.025',
  boxShadow: 'elevation.shadow.raised'
});
const smartLinkContainerStyles = xcss({
  paddingLeft: 'space.025'
});
export const SmartCardPlaceholderOld = ({
  placeholderText
}) => /*#__PURE__*/React.createElement(Box, {
  xcss: smartLinkContainerStyles
}, /*#__PURE__*/React.createElement(Box, {
  as: "span",
  testId: "datasource-modal--smart-card-placeholder",
  xcss: placeholderSmartLinkStyles
}, /*#__PURE__*/React.createElement(FormattedMessage, placeholderText)));
export const SmartLinkOld = ({
  url
}) => /*#__PURE__*/React.createElement(Box, {
  xcss: smartLinkContainerStyles
}, /*#__PURE__*/React.createElement(LinkRenderType, {
  url: url
}));
