// @atlaskit/link-datasource/dist/cjs/ui/issue-like-table/shared-components/avatar/avatar-old.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.EmptyAvatarOld = void 0;
var _react = _interopRequireDefault(require("react"));
var _reactIntlNext = require("react-intl-next");
var _avatar = _interopRequireDefault(require("@atlaskit/avatar"));
var _primitives = require("@atlaskit/primitives");
var _tooltip = _interopRequireDefault(require("@atlaskit/tooltip"));
var _user = require("../../render-type/user");
var _messages = require("../../render-type/user/messages");
var userWrapperStyles = (0, _primitives.xcss)({
  display: 'flex',
  alignItems: 'center',
  font: 'font.body'
});
var avatarWrapperStyles = (0, _primitives.xcss)({
  marginRight: 'space.100'
});
var emptyAvatarWrapperStyles = (0, _primitives.xcss)({
  paddingBlock: 'space.050',
  paddingInline: 'space.100',
  opacity: 0,
  ':hover': {
    opacity: 1
  }
});
var EmptyAvatarOld = exports.EmptyAvatarOld = function EmptyAvatarOld() {
  var intl = (0, _reactIntlNext.useIntl)();
  return /*#__PURE__*/_react.default.createElement(_tooltip.default, {
    content: intl.formatMessage(_messages.userTypeMessages.userDefaultdisplayNameValue)
  }, /*#__PURE__*/_react.default.createElement(_primitives.Box, {
    xcss: [userWrapperStyles, emptyAvatarWrapperStyles],
    testId: _user.USER_TYPE_TEST_ID
  }, /*#__PURE__*/_react.default.createElement(_primitives.Box, {
    xcss: avatarWrapperStyles
  }, /*#__PURE__*/_react.default.createElement(_avatar.default, {
    appearance: "circle",
    size: 'small',
    testId: "".concat(_user.USER_TYPE_TEST_ID, "--avatar")
  })), /*#__PURE__*/_react.default.createElement(_reactIntlNext.FormattedMessage, _messages.userTypeMessages.userDefaultdisplayNameValue)));
};
