// @atlaskit/link-datasource/dist/cjs/ui/issue-like-table/shared-components/avatar/index.js
/* index.tsx generated by @compiled/babel-plugin v0.36.1 */
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.EmptyAvatarNew = exports.EmptyAvatar = void 0;
require("./index.compiled.css");
var _runtime = require("@compiled/react/runtime");
var _react = _interopRequireDefault(require("react"));
var _reactIntlNext = require("react-intl-next");
var _avatar = _interopRequireDefault(require("@atlaskit/avatar"));
var _platformFeatureFlags = require("@atlaskit/platform-feature-flags");
var _compiled = require("@atlaskit/primitives/compiled");
var _tooltip = _interopRequireDefault(require("@atlaskit/tooltip"));
var _user = require("../../render-type/user");
var _messages = require("../../render-type/user/messages");
var _avatarOld = require("./avatar-old");
var styles = {
  avatarWrapperStyles: "_2hwxu2gc",
  emptyAvatarWrapperStyles: "_11c82smr _1rjc1b66 _18zru2gc _1e0c1txw _4cvr1h6o _tzy4idpf _16l4kb7n"
};
var EmptyAvatarNew = exports.EmptyAvatarNew = function EmptyAvatarNew() {
  var intl = (0, _reactIntlNext.useIntl)();
  return /*#__PURE__*/_react.default.createElement(_tooltip.default, {
    content: intl.formatMessage(_messages.userTypeMessages.userDefaultdisplayNameValue)
  }, /*#__PURE__*/_react.default.createElement(_compiled.Box, {
    xcss: styles.emptyAvatarWrapperStyles,
    testId: _user.USER_TYPE_TEST_ID
  }, /*#__PURE__*/_react.default.createElement(_compiled.Box, {
    xcss: styles.avatarWrapperStyles
  }, /*#__PURE__*/_react.default.createElement(_avatar.default, {
    appearance: "circle",
    size: 'small',
    testId: "".concat(_user.USER_TYPE_TEST_ID, "--avatar")
  })), /*#__PURE__*/_react.default.createElement(_reactIntlNext.FormattedMessage, _messages.userTypeMessages.userDefaultdisplayNameValue)));
};
var EmptyAvatar = exports.EmptyAvatar = function EmptyAvatar() {
  if ((0, _platformFeatureFlags.fg)('bandicoots-compiled-migration-link-datasource')) {
    return /*#__PURE__*/_react.default.createElement(EmptyAvatarNew, null);
  } else {
    return /*#__PURE__*/_react.default.createElement(_avatarOld.EmptyAvatarOld, null);
  }
};
