// @atlaskit/link-datasource/dist/cjs/ui/common/initial-state-view/index.js
/* index.tsx generated by @compiled/babel-plugin v0.36.1 */
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
var _typeof = require("@babel/runtime/helpers/typeof");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.InitialStateViewNew = exports.InitialStateView = void 0;
require("./index.compiled.css");
var React = _interopRequireWildcard(require("react"));
var _runtime = require("@compiled/react/runtime");
var _reactIntlNext = require("react-intl-next");
var _lozenge = _interopRequireDefault(require("@atlaskit/lozenge"));
var _platformFeatureFlags = require("@atlaskit/platform-feature-flags");
var _colors = require("@atlaskit/theme/colors");
var _typography = require("@atlaskit/theme/typography");
var _initialStateViewOld = require("./initial-state-view-old");
var _messages = require("./messages");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
var initialStateViewContainerStyles = null;
var svgAndTextsWrapperStyles = null;
var betaTagStyles = null;
var searchTitleStyles = null;
var mainTextStyles = null;
var learnMoreLinkStyles = null;
var InitialStateViewNew = exports.InitialStateViewNew = function InitialStateViewNew(_ref) {
  var icon = _ref.icon,
    _ref$showBeta = _ref.showBeta,
    showBeta = _ref$showBeta === void 0 ? false : _ref$showBeta,
    title = _ref.title,
    description = _ref.description,
    learnMoreLink = _ref.learnMoreLink;
  var _useIntl = (0, _reactIntlNext.useIntl)(),
    formatMessage = _useIntl.formatMessage;
  return /*#__PURE__*/React.createElement("div", {
    "data-testid": "datasource-modal--initial-state-view",
    className: (0, _runtime.ax)(["_1e0c1txw _1bah1h6o _4t3i1osq"])
  }, /*#__PURE__*/React.createElement("div", {
    className: (0, _runtime.ax)(["_y3gn1h6o _1wpz1h6o _ca0qoahv _n3tdoahv"])
  }, icon, /*#__PURE__*/React.createElement("div", {
    className: (0, _runtime.ax)(["_zulpu2gc _syazg0pi _ca0qpxbi _n3tdu2gc _1e0c1txw _1bah1h6o _4cvr1h6o _wtt4ykg7 _pfzw85h9 _zzcc1umn _x3fx1hgc _v2gl1yp8 _1y0bdipb _z7oy1jq3"]),
    style: {
      "--_1dzgjqk": (0, _runtime.ix)("var(--ds-text-subtlest, ".concat(_colors.N300, ")"))
    }
  }, showBeta && /*#__PURE__*/React.createElement("div", {
    className: (0, _runtime.ax)(["_1e0c1txw"])
  }, /*#__PURE__*/React.createElement(_lozenge.default, {
    appearance: "new"
  }, /*#__PURE__*/React.createElement(_reactIntlNext.FormattedMessage, _messages.initialStateViewMessages.beta))), formatMessage(title)), /*#__PURE__*/React.createElement("div", {
    className: (0, _runtime.ax)(["_syazg0pi"]),
    style: {
      "--_1dzgjqk": (0, _runtime.ix)("var(--ds-text-subtlest, ".concat(_colors.N300, ")"))
    }
  }, formatMessage(description)), learnMoreLink && /*#__PURE__*/React.createElement("a", {
    href: learnMoreLink.href,
    target: "_blank",
    className: (0, _runtime.ax)(["_ca0qpxbi _1e0c1o8l"])
  }, formatMessage(learnMoreLink.text))));
};
var InitialStateView = exports.InitialStateView = function InitialStateView(props) {
  if ((0, _platformFeatureFlags.fg)('bandicoots-compiled-migration-link-datasource')) {
    return /*#__PURE__*/React.createElement(InitialStateViewNew, props);
  } else {
    return /*#__PURE__*/React.createElement(_initialStateViewOld.InitialStateViewOld, props);
  }
};
