// @atlaskit/smart-card/dist/es2019/view/RelatedLinksModal/components/RelatedLinksBaseModalOld.js
/**
 * @jsxRuntime classic
 * @jsx jsx
 */

import { useCallback, useRef } from 'react';

// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766
import { jsx } from '@emotion/react';
import { FormattedMessage } from 'react-intl-next';
import Button from '@atlaskit/button/new';
import Modal, { ModalBody, ModalFooter, ModalHeader, ModalTitle, ModalTransition } from '@atlaskit/modal-dialog';
import { Box, xcss } from '@atlaskit/primitives';
import { useAnalyticsEvents } from '../../../common/analytics/generated/use-analytics-events';
import { messages } from '../../../messages';
const fixedWidth = 'small'; // pre-defined 400px by Atlaskit

const boxStyles = xcss({
  height: '396px' // Specified by the designer as this will display 5 incoming links and 2 outgoing links
});
const RelatedLinksBaseModal = ({
  onClose,
  showModal,
  children
}) => {
  const {
    fireEvent
  } = useAnalyticsEvents();
  const modalOpenTimeRef = useRef(Date.now());
  const openCompleteHandler = useCallback(el => {
    el.focus();
    modalOpenTimeRef.current = Date.now();
    fireEvent('ui.modal.opened.relatedLinks', {});
  }, [fireEvent]);
  const closeHandler = useCallback(() => {
    fireEvent('ui.modal.closed.relatedLinks', {
      dwellTime: Date.now() - modalOpenTimeRef.current
    });
    onClose === null || onClose === void 0 ? void 0 : onClose();
  }, [fireEvent, onClose]);
  return jsx(ModalTransition, null, showModal && jsx(Modal, {
    testId: "related-links-modal",
    onClose: closeHandler,
    width: fixedWidth,
    autoFocus: false,
    shouldReturnFocus: false,
    onOpenComplete: openCompleteHandler
  }, jsx(ModalHeader, null, jsx(ModalTitle, null, jsx(FormattedMessage, messages.related_links_modal_title))), jsx(ModalBody, null, jsx(Box, {
    xcss: boxStyles
  }, children)), jsx(ModalFooter, null, jsx(Button, {
    appearance: "primary",
    onClick: closeHandler
  }, jsx(FormattedMessage, messages.close)))));
};
export default RelatedLinksBaseModal;
