// @atlaskit/link-datasource/dist/es2019/ui/common/modal/basic-search-input/basic-search-input-old.js
/**
 * @jsxRuntime classic
 * @jsx jsx
 */

// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766
import { css, jsx } from '@emotion/react';
import { useIntl } from 'react-intl-next';
import { LoadingButton } from '@atlaskit/button';
import SearchIcon from '@atlaskit/icon/core/migration/search--editor-search';
import { fg } from '@atlaskit/platform-feature-flags';
import Textfield from '@atlaskit/textfield';
import { useDatasourceAnalyticsEvents } from '../../../../analytics';
import { basicSearchInputMessages } from './messages';
const searchButtonStyles = css({
  marginRight: "var(--ds-space-075, 6px)"
});
const formStyles = css({
  flex: 1
});
const formWithMaxWidthStyles = css({
  flex: 1,
  maxWidth: 250
});
export const BasicSearchInputOld = ({
  isDisabled,
  isSearching,
  onChange,
  onSearch,
  searchTerm,
  fullWidth,
  testId,
  placeholder
}) => {
  const {
    formatMessage
  } = useIntl();
  const {
    fireEvent
  } = useDatasourceAnalyticsEvents();
  const handleSearchWithAnalytics = () => {
    fireEvent('ui.form.submitted.basicSearch', {});
    onSearch(searchTerm);
  };
  const handleFormSubmit = e => {
    e.preventDefault();
    handleSearchWithAnalytics();
  };
  return jsx("form", {
    css: fullWidth ? formStyles : formWithMaxWidthStyles,
    onSubmit: handleFormSubmit
  }, jsx(Textfield, {
    elemAfterInput: jsx(LoadingButton, {
      appearance: "primary",
      css: searchButtonStyles,
      iconBefore: jsx(SearchIcon, {
        label: formatMessage(fg('confluence-issue-terminology-refresh') ? basicSearchInputMessages.basicTextSearchLabelIssueTermRefresh : basicSearchInputMessages.basicTextSearchLabel),
        LEGACY_size: "medium",
        color: "currentColor",
        spacing: "spacious"
      }),
      isDisabled: isDisabled,
      isLoading: isSearching,
      onClick: handleSearchWithAnalytics,
      spacing: "none",
      testId: `${testId}--basic-search-button`
    }),
    autoFocus: true,
    onChange: onChange,
    placeholder: formatMessage(placeholder),
    testId: `${testId}--basic-search-input`,
    value: searchTerm,
    "aria-label": formatMessage(placeholder)
  }));
};
